{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "H37lmilO9cDG"
   },
   "source": [
    "# Network-based association analysis to infer disease-gene relationship using Neo4J\n",
    "\n",
    "\n",
    "## Introduction and Objective\n",
    "\n",
    "The main objective of human genetic research is to identify genomic sequence variations that may play a role in the development of diseases. Thus information is critical to generate insights into the biology of health and disease that can support clinical translation. However the complexity of some diseases that implies the activity of several genes / proteins challenges our understanding about these pathologies. The technological boost in molecular biology that occured in the last decades prompted scientists to develop and establish several experimental approaches to identify disease-gene associations (i.e, genome-wide association studies, RNA interference screens, and linkage studies). As a result, many databases of diseases and genes have been developed and made available to the community. Consequently computational methods are now critical tools to retrieve and analyze these data for a better understanding of pathological mechanisms. Herein we are going to present and demonstrate how to harness Neo4j, a grah database tool that sits on top of a query language called Cypher to assess the relations between genes and diseases. The main objective will be to illustrate how easy and flexible this can be to quickly query data and perform simple analysis to investigate relation between genetic variants and diseases. In this regards we are going to use DisGeNET that is one of the largest available repositories of genes and variants involved in human diseases.\n",
    "\n",
    "## Before Starting\n",
    "\n",
    "If anyone wants to go through this notebook we strongly advise you to first install the [Neo4J](https://neo4j.com/) desktop app available on their website. Once done, we are going to use a python package called [py2neo](https://py2neo.org/v4/) that provides an API to direclty interact with our graph database. To be able to interact with your database you need to run its server locally using the Neo4J app. The `Graph()` class from *py2neo* provides an interface to directly connect to Neo4J database server and run the expected queries. \n",
    "\n",
    "To check if you have all needed packages you can run the cell below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "execution": {
     "iopub.execute_input": "2020-07-04T11:35:22.228Z",
     "iopub.status.busy": "2020-07-04T11:35:22.205Z",
     "iopub.status.idle": "2020-07-04T11:35:25.608Z",
     "shell.execute_reply": "2020-07-04T11:35:25.625Z"
    },
    "id": "mabK8W6L9cDH",
    "outputId": "5e44f9f1-6d34-47d0-a7b9-0701c4765af7",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "import subprocess\n",
    "import pkg_resources\n",
    "\n",
    "required = {'pandas', 'py2neo', 'matplotlib','neo4jupyter'}\n",
    "installed = {pkg.key for pkg in pkg_resources.working_set}\n",
    "missing = required - installed\n",
    "\n",
    "if missing:\n",
    "    python = sys.executable\n",
    "    subprocess.check_call([python, '-m', 'pip', 'install', *missing], stdout=subprocess.DEVNULL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "execution": {
     "iopub.execute_input": "2020-07-04T11:35:37.420Z",
     "iopub.status.busy": "2020-07-04T11:35:37.403Z",
     "iopub.status.idle": "2020-07-04T11:35:37.451Z",
     "shell.execute_reply": "2020-07-04T11:35:37.468Z"
    },
    "id": "HmBaduq39cDM"
   },
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.style.use('fivethirtyeight')\n",
    "import pandas as pd\n",
    "import neo4jupyter\n",
    "from py2neo import Graph"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "pPlsFIcz9cDP"
   },
   "source": [
    "## Modelization of the data\n",
    "\n",
    "Graphs are mathematical objects that provide a very convenient way of modeling and analyzing data. The objects that make up a graph are called nodes or vertices and the links between them are known as relationships, links, or edges. Here we are going to investigate the link between genetic variants and pathologies. Genetic variants can be refered to mutations or more specifically **single nucleotide polymorphisms** for a same gene. Nucleotides are the building blocks of our genomes. We have four different nucleotides (commonly labelled as A, T, G, C) that generate long DNA sequences which encode our genes. Several diseases or pathological conditions stem from genetic variations or mutations and identifying the  genetic variant involved in these diseases is of critical importance in medical sciences. \n",
    "\n",
    "In this regard we are going to interrogate 3 different data repositories (`curated_variant_disease_associations.tsv`, `variant_to_gene_mappings.tsv`, `disease_mappings_to_attributes.tsv`) to generate the following data model.\n",
    "\n",
    "![data model](data_model_2.svg)\n",
    "\n",
    "<br></br>\n",
    "\n",
    "Consequently in this part we are going to see and describe the type of queries we need in order to generate this data model in Neo4j. First let's connect to our database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "DZQ_2pl39cDQ"
   },
   "outputs": [],
   "source": [
    "graph = Graph(\"bolt://localhost:7687\", auth = ('neo4j', 'faouzi'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "QtDCCN519cDT"
   },
   "source": [
    "Next let's have a quick look at our data sets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "L4qMVxff9cDU",
    "outputId": "2a00de47-6fe4-44bd-ebb4-9002f9bcb989"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'line': {'diseaseSemanticType': 'Laboratory Procedure',\n",
       "   'EI': '1',\n",
       "   'chromosome': '21',\n",
       "   'source': 'GWASCAT',\n",
       "   'YearFinal': '2016',\n",
       "   'diseaseType': 'phenotype',\n",
       "   'YearInitial': '2016',\n",
       "   'NofPmids': '  1',\n",
       "   'diseaseName': 'Eosinophil count procedure',\n",
       "   'score': '0.70',\n",
       "   'diseaseClass': None,\n",
       "   'snpId': 'rs1000005',\n",
       "   'position': ' 33060745',\n",
       "   'DSI': None,\n",
       "   'DPI': None,\n",
       "   'diseaseId': 'C0200638'}},\n",
       " {'line': {'diseaseSemanticType': 'Neoplastic Process',\n",
       "   'EI': '1',\n",
       "   'chromosome': '4',\n",
       "   'source': 'GWASCAT',\n",
       "   'YearFinal': '2017',\n",
       "   'diseaseType': 'disease',\n",
       "   'YearInitial': '2017',\n",
       "   'NofPmids': '  1',\n",
       "   'diseaseName': 'Leukemia, Myelocytic, Acute',\n",
       "   'score': '0.70',\n",
       "   'diseaseClass': 'C04',\n",
       "   'snpId': 'rs10000770',\n",
       "   'position': '142693109',\n",
       "   'DSI': '1',\n",
       "   'DPI': '0.04',\n",
       "   'diseaseId': 'C0023467'}}]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# for curated_variant_disease_associations.tsv\n",
    "\n",
    "query_01 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///curated_variant_disease_associations.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "RETURN line LIMIT 2\n",
    "\"\"\"\n",
    "graph.run(query_01).data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "t3vjha549cDY",
    "outputId": "48596ac0-ef93-4fc3-b14d-6098e40559a4"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'line': {'sourceId': 'VEP',\n",
       "   'geneId': '2',\n",
       "   'snpId': 'rs7980288',\n",
       "   'geneSymbol': 'A2M'}},\n",
       " {'line': {'sourceId': 'DBSNP',\n",
       "   'geneId': '2',\n",
       "   'snpId': 'rs7980288',\n",
       "   'geneSymbol': 'A2M'}}]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# for variant_to_gene_mappings.tsv\n",
    "\n",
    "query_02 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///variant_to_gene_mappings.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "RETURN line LIMIT 2\n",
    "\"\"\"\n",
    "graph.run(query_02).data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "Nb9ovR-39cDb",
    "outputId": "af391aa6-9540-416f-e120-8ac16e9a2369"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'line': {'hpoClassName': None,\n",
       "   'doClassName': None,\n",
       "   'diseaseClassMSH': 'C23',\n",
       "   'umlsSemanticTypeId': 'T184',\n",
       "   'name': 'Abdomen, Acute',\n",
       "   'diseaseClassNameMSH': 'Pathological Conditions, Signs and Symptoms',\n",
       "   'umlsSemanticTypeName': 'Sign or Symptom',\n",
       "   'type': 'phenotype',\n",
       "   'hpoClassId': None,\n",
       "   'diseaseId': 'C0000727',\n",
       "   'doClassId': None}},\n",
       " {'line': {'hpoClassName': None,\n",
       "   'doClassName': None,\n",
       "   'diseaseClassMSH': 'C16',\n",
       "   'umlsSemanticTypeId': 'T184',\n",
       "   'name': 'Abdominal Cramps',\n",
       "   'diseaseClassNameMSH': 'Congenital, Hereditary, and Neonatal Diseases and Abnormalities',\n",
       "   'umlsSemanticTypeName': 'Sign or Symptom',\n",
       "   'type': 'phenotype',\n",
       "   'hpoClassId': None,\n",
       "   'diseaseId': 'C0000729',\n",
       "   'doClassId': None}}]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# for disease_mappings_to_attributes.tsv\n",
    "\n",
    "query_03 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///disease_mappings_to_attributes.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "RETURN line LIMIT 2\n",
    "\"\"\"\n",
    "graph.run(query_03).data()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "OAXGnguo9cDe"
   },
   "source": [
    "1.  **The `curated_variant_disease_associations.tsv` file**\n",
    "- We can see that Neo4J returns the data as a kind of JSON data structure. The keys would correspond to the headers of a classical table, in other words the column names. Here we can focus on some interesting entries given the data model we want to achieve. \n",
    "- The key `chromosome` has a string value. In our example both values are `'21'` and `'4'`. This is the way we name chromosomes. We have 23 pairs of chromosomes from 1 to 22 and a pair of sexual chromosomes named 'X' and 'Y'. But is it all ? For someone with no deep knowledge in biology we could actually miss something here. Let's check that we did not forget anything. We can for example ask Neo4j to return us the distinct values hold by the key `chromosome`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "B5CIMGvE9cDf",
    "outputId": "ee23cab2-5d5c-4a59-a3fd-bb99a46a2b8c"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table><tr><th>chromosome_name</th></tr><tr><td style=\"text-align:left\">1</td></tr><tr><td style=\"text-align:left\">2</td></tr><tr><td style=\"text-align:left\">3</td></tr><tr><td style=\"text-align:left\">4</td></tr><tr><td style=\"text-align:left\">5</td></tr><tr><td style=\"text-align:left\">6</td></tr><tr><td style=\"text-align:left\">7</td></tr><tr><td style=\"text-align:left\">8</td></tr><tr><td style=\"text-align:left\">9</td></tr><tr><td style=\"text-align:left\">10</td></tr><tr><td style=\"text-align:left\">11</td></tr><tr><td style=\"text-align:left\">12</td></tr><tr><td style=\"text-align:left\">13</td></tr><tr><td style=\"text-align:left\">14</td></tr><tr><td style=\"text-align:left\">15</td></tr><tr><td style=\"text-align:left\">16</td></tr><tr><td style=\"text-align:left\">17</td></tr><tr><td style=\"text-align:left\">18</td></tr><tr><td style=\"text-align:left\">19</td></tr><tr><td style=\"text-align:left\">20</td></tr><tr><td style=\"text-align:left\">21</td></tr><tr><td style=\"text-align:left\">22</td></tr><tr><td style=\"text-align:left\">X</td></tr><tr><td style=\"text-align:left\">Y</td></tr><tr><td style=\"text-align:left\">MT</td></tr></table>"
      ],
      "text/plain": [
       " chromosome_name \n",
       "-----------------\n",
       " 1               \n",
       " 2               \n",
       " 3               \n",
       " 4               \n",
       " 5               \n",
       " 6               \n",
       " 7               \n",
       " 8               \n",
       " 9               \n",
       " 10              \n",
       " 11              \n",
       " 12              \n",
       " 13              \n",
       " 14              \n",
       " 15              \n",
       " 16              \n",
       " 17              \n",
       " 18              \n",
       " 19              \n",
       " 20              \n",
       " 21              \n",
       " 22              \n",
       " X               \n",
       " Y               \n",
       " MT              "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_04 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///curated_variant_disease_associations.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "RETURN DISTINCT line.chromosome AS chromosome_name\n",
    "ORDER BY toInteger(line.chromosome)\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_04).to_table()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "guICOsFJ9cDj"
   },
   "source": [
    ">A quick side note here: The *py2neo* provides a very smart and interesting way to interact with Neo4J server databases. The `graph.run(your_query)` expression returns an instance of `Cursor()`. This is actually a pointer that can be used to navigate through the stream of data generated by our Cypher request in Neo4j. With *py2neo* We can invoke a set of methods on this `Cursor` object to read data returned by our Cypher queries. Among these methods we have `data()`, `to_data_frame()`, `to_table()`. These are very convenient to format and convert data in a tabular and more readable format.\n",
    "\n",
    "- Interestingly we can notice that we missed one type of chromosome named `'MT'` in our dataset. `'MT'` is for mitochondrial. Indeed the 23 pairs of chromosomes are packed in the nucleus of our cells. But a small fraction of our total DNA is also packed into our mitochondria. Mitochondria are the cellular components responsible for the production of cellular energy. \n",
    "- Other entries of interest in the `curated_variant_disease_associations.tsv` are `snpId` and `diseaseId`. `snpId` is the identification reference to the genetic variant. `diseaseId` is also an identification reference to a disease from the Medical Subject Headings thesaurus. No information can be found about genes here.\n",
    "\n",
    "2. **The variant_to_gene_mappings.tsv file**\n",
    "\n",
    "- The data is more contained here. We have two entries of interest: `snpId` and `geneSymbol`. The latter holds a string value that is an acronym of the gene name. These acronyms are commonly used in the scientific community to name genes. These two entries will be of interest to generate our `Variant` node and the respective properties (see our data model above).\n",
    "\n",
    "3. **The disease_mappings_to_attributes.tsv file**\n",
    "\n",
    "- The last file contains data that will permit us to map `diseaseId` data found in our first file and map this information to the `name`, `type` and `diseaseClassNameMSH` necessary to create our `Disease` and `Phenotype` nodes. We will come back later on these nodes with more explanations.\n",
    "\n",
    "## Creation of our Data Model\n",
    "\n",
    "### Chromosomes, Variants and Genes\n",
    "\n",
    "We are going to create our `Variant` and `Chromosome` nodes. First we’ll create constraints to ensure we don’t create duplicate chromosomes or variants."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "ae21toU19cDj",
    "outputId": "3bb827cb-7454-4fe3-9d09-8835dd7835f0"
   },
   "outputs": [],
   "source": [
    "query_05 = \"CREATE CONSTRAINT ON (v:Variant) ASSERT v.variantId IS UNIQUE\"\n",
    "graph.run(query_05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "9BpbEXCT9cDm",
    "outputId": "cfbddb72-046c-4580-89df-e6fbf2b59f55"
   },
   "outputs": [],
   "source": [
    "query_06 = \"CREATE CONSTRAINT ON (c:Chromosome) ASSERT c.name IS UNIQUE\"\n",
    "graph.run(query_06)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "VFMEP-hR9cDp"
   },
   "source": [
    "Next we are going to set some of the properties and check the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "jBPX4HnX9cDq",
    "outputId": "ed2e4c5d-28b8-4cee-9c6a-ace9587731ac"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7fee837acdd0>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_07 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///curated_variant_disease_associations.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "MERGE (v:Variant {variantId: line.snpId})\n",
    "MERGE (c:Chromosome {name: line.chromosome})\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_07)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "jJ60jSL79cDt"
   },
   "source": [
    "Let's add the property `variantOf` to our node `Variant`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "5TlrZ9DA9cDu",
    "outputId": "127a68bd-eeea-4b97-d396-b44df5d957b9"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f014b50>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_08 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///variant_to_gene_mappings.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "MATCH (v:Variant) WHERE v.variantId = line.snpId\n",
    "SET v.variantOf = line.geneSymbol\n",
    "\"\"\"\n",
    "graph.run(query_08)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "U0Myp1-99cDx"
   },
   "source": [
    "For sanity check let's be sure that the data provided by our `variant_to_gene_mappings.tsv` covered all the variants we have in our database. If true, we should have all nodes with a non-null string entry in its `variantOf` property."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "hHXddDv_9cDx",
    "outputId": "00d59168-417e-425a-893a-cc56408dc99d"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'COUNT(v)': 23744}]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_09 = \"\"\"\n",
    "MATCH (v:Variant)\n",
    "WHERE v.variantOf is NULL\n",
    "RETURN COUNT(v)\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_09).data()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "lNdHWOsQ9cD1"
   },
   "source": [
    "Interesting. We have 23744 nodes that do not have a mapping for a gene. What does this mean? Fortunately there is a biological explanation to it. Indeed there are several genetic variants that are not genes. Indeed genetic sequences that encode genes represent a very small fraction of our DNA (around 5%). The majority of our DNA is composed of non-coding genes. Non coding sequences impact and regulate the expression of genes and are often refered as a kind of *dark matter*. The exact role and the mechanisms by which non-coding DNA affect the expression of our genes is under investigation and fall under the umbrella of epigenetic. For the sake of our analysis and based on our data model we are going to focus on the variants that are encoded by genes. We can just suppress the nodes that do not map to a gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "fF0qljpL9cD1",
    "outputId": "09d434e1-cba1-44a8-ed6f-7f95aa0ecb8b"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'COUNT(v)': 168051}]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Before suppression\n",
    "\n",
    "query_10 = \"\"\"\n",
    "MATCH (v:Variant) \n",
    "RETURN COUNT(v)\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_10).data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "7mckesKB9cD4",
    "outputId": "422fe9cf-b33c-4678-aa01-0b17ca1a50f5"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f01ded0>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Suppresion\n",
    "\n",
    "query_11 = \"\"\"\n",
    "MATCH (v:Variant) \n",
    "WHERE v.variantOf is NULL\n",
    "DELETE v\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_11)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "4fiGlJyk9cD7",
    "outputId": "bc3c5078-0e2c-4654-b135-61ecafdaa005"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'COUNT(v)': 144307}]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count after suppression\n",
    "\n",
    "query_12 = \"\"\"\n",
    "MATCH (v:Variant) \n",
    "RETURN COUNT(v)\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_12).data()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "2cfCXugM9cD-"
   },
   "source": [
    "Finally we can create relationships between the variants and their chromosomic locations. Notice how pattern matching greatly simplifies the process. Cypher's synthax is almost visual and very intuitive. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "Qrb33LFy9cD-",
    "outputId": "0874e869-b264-4f4a-816c-087a8bec1b97"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f022c90>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_13 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///curated_variant_disease_associations.tsv\"AS line FIELDTERMINATOR \"\\t\"\n",
    "MATCH (v:Variant {variantId: line.snpId})\n",
    "MATCH (c:Chromosome {name: line.chromosome})\n",
    "MERGE (c)<-[:LOCATED_IN]-(v)\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_13)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "uJIF7p6f9cEB"
   },
   "source": [
    "Let's visualize our data model up to this point. Running the following statement `CALL db.schema.visualization()` in the Neo4J app will return the following graph.\n",
    "\n",
    "![data_model_first](data_model_first.svg)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "CQ06FK3c9cEC"
   },
   "source": [
    "### Diseases and Phenotypes\n",
    "\n",
    "We are going now to create the other two nodes we need: `Disease` and `Phenotype`. A bit of definition. The term *phenotype* refers to an observable and quantifyable properties of an organism; these include the organism's appearance, development, and behavior. Clinical signs of diseases are some kind of phenotypes. For example a sudden increase of liver enzymes is a phenotype. However this can be found in the diagnosis of many diseases. Consequently we want here to really distinguish between diseases and phenotypes. It is also interesting to link which variants and *a fortiori* which genes are associated to such phenotypes as they can give a quick information about the why. Indeed some pehnotypes can be the results of some genetic mutations.This is particularly the case in monogeneic diseases. To create these nodes we can first generate our constraints to avoid any redundancy in our data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "pRW-0zvX9cEC",
    "outputId": "93b6d838-e55c-4af7-c076-f206017004e2"
   },
   "outputs": [],
   "source": [
    "query_14 = \"\"\"\n",
    "CREATE CONSTRAINT ON (d:Disease) ASSERT d.diseaseId IS UNIQUE\n",
    "\"\"\"\n",
    "query_15 = \"\"\"\n",
    "CREATE CONSTRAINT ON (p:Phenotype) ASSERT p.phenotypeId IS UNIQUE\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_14)\n",
    "graph.run(query_15)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "cSzNLGkQ9cEF"
   },
   "source": [
    "From our data model we wish to build two nodes:\n",
    "- A `Disease` node with the following properties: `diseaseId`, `diseaseName` and `diseaseGroup`\n",
    "- A `Phenotype` node with the following properties: `phenotypeId`, `phenotypeName` and `phenotypeGroup`\n",
    "\n",
    "To achieve this goal we are going to have a deeper look to the `disease_mappings_to_attributes.tsv` file and focus on the following entries: `diseaseId`, `name`, `type` (disease or phenotype) and `diseaseClassNameMSH` (for group). Let's have a first look at the diversity of the values we can find in each entry of this dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "jMC3P7Yc9cEG",
    "outputId": "449a00a3-56a8-458c-935f-99027fa7a2ee"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table><tr><th>diseaseOrPhenotypeName</th><th>diseaseOrPhenotypeId</th><th>diseaseOrPhenotypeGroup</th><th>Type</th></tr><tr><td style=\"text-align:right\">30293</td><td style=\"text-align:right\">30293</td><td style=\"text-align:right\">1108</td><td style=\"text-align:right\">3</td></tr></table>"
      ],
      "text/plain": [
       " diseaseOrPhenotypeName | diseaseOrPhenotypeId | diseaseOrPhenotypeGroup | Type \n",
       "------------------------|----------------------|-------------------------|------\n",
       "                  30293 |                30293 |                    1108 |    3 "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_16 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///disease_mappings_to_attributes.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "RETURN COUNT(DISTINCT line.name) AS diseaseOrPhenotypeName, \n",
    "       COUNT(DISTINCT line.diseaseId) AS diseaseOrPhenotypeId, \n",
    "       COUNT(DISTINCT line.diseaseClassNameMSH) AS diseaseOrPhenotypeGroup,\n",
    "       COUNT(DISTINCT line.type) AS Type\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_16).to_table()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "M6S04GkG9cEI"
   },
   "source": [
    "This query gives us useful details regarding the data. First we can see that we have the same number of `diseaseOrPhenotypeName` and `diseaseOrPhenotypeId` which makes sense for a data which is supposed to map an id to a name. In addition we can see that our disease can be gather in 1108 different `diseaseOrPhenotypeGroup` of diseases and 3 different `Type`. Next we are going to investigate the values of by the two last variables."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "ZMOxw3tk9cEJ"
   },
   "outputs": [],
   "source": [
    "query_17 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///disease_mappings_to_attributes.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "RETURN DISTINCT line.type AS Type\n",
    "\"\"\"\n",
    "\n",
    "query_18 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///disease_mappings_to_attributes.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "RETURN DISTINCT line.diseaseClassNameMSH as Groups LIMIT 50\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "aVNY_UK29cEL",
    "outputId": "05c1a313-b9b4-4540-bb59-d4ab898b6bd2"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table><tr><th>Type</th></tr><tr><td style=\"text-align:left\">phenotype</td></tr><tr><td style=\"text-align:left\">group</td></tr><tr><td style=\"text-align:left\">disease</td></tr></table>"
      ],
      "text/plain": [
       " Type      \n",
       "-----------\n",
       " phenotype \n",
       " group     \n",
       " disease   "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "graph.run(query_17).to_table()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "y95AWqxO9cEO",
    "outputId": "f1f4d3b0-bf53-4d16-fc80-2084a1476dde"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table><tr><th>Groups</th></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities</td></tr><tr><td style=\"text-align:left\">Digestive System Diseases</td></tr><tr><td style=\"text-align:left\">Neoplasms</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases</td></tr><tr><td style=\"text-align:left\">Female Urogenital Diseases and Pregnancy Complications</td></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms; Infections</td></tr><tr><td style=\"text-align:left\">null</td></tr><tr><td style=\"text-align:left\">Infections; Eye Diseases</td></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms; Skin and Connective Tissue Diseases</td></tr><tr><td style=\"text-align:left\">Skin and Connective Tissue Diseases</td></tr><tr><td style=\"text-align:left\">Digestive System Diseases; Nutritional and Metabolic Diseases</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Musculoskeletal Diseases</td></tr><tr><td style=\"text-align:left\">Nutritional and Metabolic Diseases</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases; Female Urogenital Diseases and Pregnancy Complications; Male Urogenital Diseases</td></tr><tr><td style=\"text-align:left\">Nutritional and Metabolic Diseases; Respiratory Tract Diseases</td></tr><tr><td style=\"text-align:left\">Infections</td></tr><tr><td style=\"text-align:left\">Infections; Immune System Diseases</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Skin and Connective Tissue Diseases</td></tr><tr><td style=\"text-align:left\">Musculoskeletal Diseases; Nervous System Diseases; Endocrine System Diseases</td></tr><tr><td style=\"text-align:left\">Digestive System Diseases; Chemically-Induced Disorders; Mental Disorders</td></tr><tr><td style=\"text-align:left\">Infections; Respiratory Tract Diseases</td></tr><tr><td style=\"text-align:left\">Infections; Nervous System Diseases</td></tr><tr><td style=\"text-align:left\">Stomatognathic Diseases</td></tr><tr><td style=\"text-align:left\">Infections; Respiratory Tract Diseases; Stomatognathic Diseases; Otorhinolaryngologic Diseases</td></tr><tr><td style=\"text-align:left\">Immune System Diseases; Endocrine System Diseases</td></tr><tr><td style=\"text-align:left\">Hemic and Lymphatic Diseases</td></tr><tr><td style=\"text-align:left\">Nutritional and Metabolic Diseases; Skin and Connective Tissue Diseases</td></tr><tr><td style=\"text-align:left\">Mental Disorders</td></tr><tr><td style=\"text-align:left\">Endocrine System Diseases</td></tr><tr><td style=\"text-align:left\">Neoplasms; Endocrine System Diseases</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases; Female Urogenital Diseases and Pregnancy Complications; Male Urogenital Diseases; Endocrine System Diseases</td></tr><tr><td style=\"text-align:left\">Behavior and Behavior Mechanisms</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Hemic and Lymphatic Diseases</td></tr><tr><td style=\"text-align:left\">Infections; Animal Diseases</td></tr><tr><td style=\"text-align:left\">Immune System Diseases; Hemic and Lymphatic Diseases</td></tr><tr><td style=\"text-align:left\">Nutritional and Metabolic Diseases; Musculoskeletal Diseases</td></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms; Nervous System Diseases; Behavior and Behavior Mechanisms</td></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms; Nervous System Diseases</td></tr><tr><td style=\"text-align:left\">Occupational Diseases</td></tr><tr><td style=\"text-align:left\">Infections; Immune System Diseases; Nervous System Diseases; Mental Disorders</td></tr><tr><td style=\"text-align:left\">Respiratory Tract Diseases</td></tr><tr><td style=\"text-align:left\">Nervous System Diseases</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases; Eye Diseases; Skin and Connective Tissue Diseases</td></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms; Female Urogenital Diseases and Pregnancy Complications; Male Urogenital Diseases</td></tr><tr><td style=\"text-align:left\">Chemically-Induced Disorders; Mental Disorders</td></tr><tr><td style=\"text-align:left\">Nervous System Diseases; Chemically-Induced Disorders; Mental Disorders</td></tr><tr><td style=\"text-align:left\">Infections; Eye Diseases; Immune System Diseases; Animal Diseases</td></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms; Nervous System Diseases; Mental Disorders</td></tr><tr><td style=\"text-align:left\">Respiratory Tract Diseases; Immune System Diseases; Otorhinolaryngologic Diseases</td></tr></table>"
      ],
      "text/plain": [
       " Groups                                                                                                                                                                                                           \n",
       "------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
       " Pathological Conditions, Signs and Symptoms                                                                                                                                                                      \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities                                                                                                                                                  \n",
       " Digestive System Diseases                                                                                                                                                                                        \n",
       " Neoplasms                                                                                                                                                                                                        \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases                                                                                                              \n",
       " Female Urogenital Diseases and Pregnancy Complications                                                                                                                                                           \n",
       " Pathological Conditions, Signs and Symptoms; Infections                                                                                                                                                          \n",
       " null                                                                                                                                                                                                             \n",
       " Infections; Eye Diseases                                                                                                                                                                                         \n",
       " Pathological Conditions, Signs and Symptoms; Skin and Connective Tissue Diseases                                                                                                                                 \n",
       " Skin and Connective Tissue Diseases                                                                                                                                                                              \n",
       " Digestive System Diseases; Nutritional and Metabolic Diseases                                                                                                                                                    \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Musculoskeletal Diseases                                                                                                                        \n",
       " Nutritional and Metabolic Diseases                                                                                                                                                                               \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases; Female Urogenital Diseases and Pregnancy Complications; Male Urogenital Diseases                            \n",
       " Nutritional and Metabolic Diseases; Respiratory Tract Diseases                                                                                                                                                   \n",
       " Infections                                                                                                                                                                                                       \n",
       " Infections; Immune System Diseases                                                                                                                                                                               \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Skin and Connective Tissue Diseases                                                                                                             \n",
       " Musculoskeletal Diseases; Nervous System Diseases; Endocrine System Diseases                                                                                                                                     \n",
       " Digestive System Diseases; Chemically-Induced Disorders; Mental Disorders                                                                                                                                        \n",
       " Infections; Respiratory Tract Diseases                                                                                                                                                                           \n",
       " Infections; Nervous System Diseases                                                                                                                                                                              \n",
       " Stomatognathic Diseases                                                                                                                                                                                          \n",
       " Infections; Respiratory Tract Diseases; Stomatognathic Diseases; Otorhinolaryngologic Diseases                                                                                                                   \n",
       " Immune System Diseases; Endocrine System Diseases                                                                                                                                                                \n",
       " Hemic and Lymphatic Diseases                                                                                                                                                                                     \n",
       " Nutritional and Metabolic Diseases; Skin and Connective Tissue Diseases                                                                                                                                          \n",
       " Mental Disorders                                                                                                                                                                                                 \n",
       " Endocrine System Diseases                                                                                                                                                                                        \n",
       " Neoplasms; Endocrine System Diseases                                                                                                                                                                             \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases; Female Urogenital Diseases and Pregnancy Complications; Male Urogenital Diseases; Endocrine System Diseases \n",
       " Behavior and Behavior Mechanisms                                                                                                                                                                                 \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Hemic and Lymphatic Diseases                                                                                                                    \n",
       " Infections; Animal Diseases                                                                                                                                                                                      \n",
       " Immune System Diseases; Hemic and Lymphatic Diseases                                                                                                                                                             \n",
       " Nutritional and Metabolic Diseases; Musculoskeletal Diseases                                                                                                                                                     \n",
       " Pathological Conditions, Signs and Symptoms; Nervous System Diseases; Behavior and Behavior Mechanisms                                                                                                           \n",
       " Pathological Conditions, Signs and Symptoms; Nervous System Diseases                                                                                                                                             \n",
       " Occupational Diseases                                                                                                                                                                                            \n",
       " Infections; Immune System Diseases; Nervous System Diseases; Mental Disorders                                                                                                                                    \n",
       " Respiratory Tract Diseases                                                                                                                                                                                       \n",
       " Nervous System Diseases                                                                                                                                                                                          \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases; Eye Diseases; Skin and Connective Tissue Diseases                                                           \n",
       " Pathological Conditions, Signs and Symptoms; Female Urogenital Diseases and Pregnancy Complications; Male Urogenital Diseases                                                                                    \n",
       " Chemically-Induced Disorders; Mental Disorders                                                                                                                                                                   \n",
       " Nervous System Diseases; Chemically-Induced Disorders; Mental Disorders                                                                                                                                          \n",
       " Infections; Eye Diseases; Immune System Diseases; Animal Diseases                                                                                                                                                \n",
       " Pathological Conditions, Signs and Symptoms; Nervous System Diseases; Mental Disorders                                                                                                                           \n",
       " Respiratory Tract Diseases; Immune System Diseases; Otorhinolaryngologic Diseases                                                                                                                                "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "graph.run(query_18).to_table()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "JdFuIdkD9cER"
   },
   "source": [
    "From these two queries we can first see that the 3 types are the following: `phenotype`, `group` and `disease`. `group` is a particular semantic used to qualify syndromes and complexe diseases. In our database entries with the type `group` represent a very small percentage of our data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "pTpA6Q3F9cER",
    "outputId": "f9a83580-a38d-4fab-9598-1c0df2ac091d"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table><tr><th>Number_of_entries_with_group_type</th></tr><tr><td style=\"text-align:right\">962</td></tr></table>"
      ],
      "text/plain": [
       " Number_of_entries_with_group_type \n",
       "-----------------------------------\n",
       "                               962 "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_19 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///disease_mappings_to_attributes.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "WITH line \n",
    "WHERE line.type = \"group\"\n",
    "RETURN COUNT(line) AS Number_of_entries_with_group_type\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_19).to_table()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "z2YrJUHa9cEU"
   },
   "source": [
    "Because we want to focus on well defined diseases we are going to skip values with the type `group`. Another information that we got from our `query_18` is the fact that `diseaseClassNameMSH` entry contains some `null` values. Here we are not going to suppress them but rather assign a `Not Determined` string value. Having all these informations we can create our nodes \"Disease\" and \"Phenotype\". "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "6gimBck79cEV",
    "outputId": "241753b1-3fcb-438c-f457-f544cd603ae1"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f02dc90>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_20 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///disease_mappings_to_attributes.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "FOREACH(uselessVar IN CASE WHEN line.type = \"disease\" THEN [1] ELSE [] END | \n",
    "    MERGE (d:Disease {diseaseId: line.diseaseId, diseaseName: line.name})\n",
    ")\n",
    "\n",
    "FOREACH(uselessVar IN CASE WHEN line.type = \"phenotype\" THEN [1] ELSE [] END | \n",
    "    MERGE (p:Phenotype {phenotypeId: line.diseaseId, phenotypeName: line.name})\n",
    ")\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "Cojl1kht9cEZ"
   },
   "source": [
    "Another aspect of the data shown from our `query_18` is the fact that some values from `diseaseClassNameMSH` are semi-colon separated. For example we have `Respiratory Tract Diseases` or `Congenital, Hereditary, and Neonatal Diseases and Abnormalities; Nutritional and Metabolic Diseases; Eye Diseases; Skin and Connective Tissue Diseases`. In our second example we can see that we have several class of diseases. For the sake of our analysis we are going to clean these string by splitting them based the presence of the `; ` character and select the last entry of the resulting list. This is an arbitrary choice and from our reading and investigations of the data reflects the fact that most of the time the last entry is a very direct and verbose description of the disease class."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "yBppCdgi9cEa",
    "outputId": "571aecf1-4f6c-4868-873c-1f3aacc85025"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f030450>"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_21 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///disease_mappings_to_attributes.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "MATCH (d:Disease {diseaseId: line.diseaseId})\n",
    "    WITH split(line.diseaseClassNameMSH, \"; \") AS splitted_groups, d\n",
    "    UNWIND splitted_groups as splitted_group\n",
    "    WITH apoc.agg.nth(splitted_group, -1) AS x, d\n",
    "    SET d.diseaseGroup = x\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_21)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "C093eNhy9cEc"
   },
   "source": [
    ">Quick Note: We are using here an **APOC** procedure. **APOC** is a very interesting extension for Neo4J which provides tons of procedures and algorithms to explore, query and analyse you data. Do not forget to install the plugin for your Neo4J database server."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "jGPJGu2n9cEd",
    "outputId": "c707b023-9d97-4f00-85d6-ee5c7616c4f4"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f030bd0>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_22 = \"\"\"\n",
    "MATCH (d:Disease)\n",
    "WHERE d.diseaseGroup IS NULL\n",
    "SET d.diseaseGroup = \"Not determined\"\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_22)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "QxzFq7pl9cEf",
    "outputId": "34aefbb6-4ef5-4901-cf44-da12051d4171"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table><tr><th>disease_group</th></tr><tr><td style=\"text-align:left\">Nutritional and Metabolic Diseases</td></tr><tr><td style=\"text-align:left\">Female Urogenital Diseases and Pregnancy Complications</td></tr><tr><td style=\"text-align:left\">Not determined</td></tr><tr><td style=\"text-align:left\">Eye Diseases</td></tr><tr><td style=\"text-align:left\">Skin and Connective Tissue Diseases</td></tr><tr><td style=\"text-align:left\">Musculoskeletal Diseases</td></tr><tr><td style=\"text-align:left\">Endocrine System Diseases</td></tr><tr><td style=\"text-align:left\">Neoplasms</td></tr><tr><td style=\"text-align:left\">Infections</td></tr><tr><td style=\"text-align:left\">Mental Disorders</td></tr><tr><td style=\"text-align:left\">Digestive System Diseases</td></tr><tr><td style=\"text-align:left\">Respiratory Tract Diseases</td></tr><tr><td style=\"text-align:left\">Nervous System Diseases</td></tr><tr><td style=\"text-align:left\">Stomatognathic Diseases</td></tr><tr><td style=\"text-align:left\">Otorhinolaryngologic Diseases</td></tr><tr><td style=\"text-align:left\">Hemic and Lymphatic Diseases</td></tr><tr><td style=\"text-align:left\">Animal Diseases</td></tr><tr><td style=\"text-align:left\">Behavior and Behavior Mechanisms</td></tr><tr><td style=\"text-align:left\">Immune System Diseases</td></tr><tr><td style=\"text-align:left\">Cardiovascular Diseases</td></tr><tr><td style=\"text-align:left\">Pathological Conditions, Signs and Symptoms</td></tr><tr><td style=\"text-align:left\">Occupational Diseases</td></tr><tr><td style=\"text-align:left\">Wounds and Injuries</td></tr><tr><td style=\"text-align:left\">Congenital, Hereditary, and Neonatal Diseases and Abnormalities</td></tr><tr><td style=\"text-align:left\">Male Urogenital Diseases</td></tr><tr><td style=\"text-align:left\">Chemically-Induced Disorders</td></tr><tr><td style=\"text-align:left\">Disorders of Environmental Origin</td></tr></table>"
      ],
      "text/plain": [
       " disease_group                                                   \n",
       "-----------------------------------------------------------------\n",
       " Nutritional and Metabolic Diseases                              \n",
       " Female Urogenital Diseases and Pregnancy Complications          \n",
       " Not determined                                                  \n",
       " Eye Diseases                                                    \n",
       " Skin and Connective Tissue Diseases                             \n",
       " Musculoskeletal Diseases                                        \n",
       " Endocrine System Diseases                                       \n",
       " Neoplasms                                                       \n",
       " Infections                                                      \n",
       " Mental Disorders                                                \n",
       " Digestive System Diseases                                       \n",
       " Respiratory Tract Diseases                                      \n",
       " Nervous System Diseases                                         \n",
       " Stomatognathic Diseases                                         \n",
       " Otorhinolaryngologic Diseases                                   \n",
       " Hemic and Lymphatic Diseases                                    \n",
       " Animal Diseases                                                 \n",
       " Behavior and Behavior Mechanisms                                \n",
       " Immune System Diseases                                          \n",
       " Cardiovascular Diseases                                         \n",
       " Pathological Conditions, Signs and Symptoms                     \n",
       " Occupational Diseases                                           \n",
       " Wounds and Injuries                                             \n",
       " Congenital, Hereditary, and Neonatal Diseases and Abnormalities \n",
       " Male Urogenital Diseases                                        \n",
       " Chemically-Induced Disorders                                    \n",
       " Disorders of Environmental Origin                               "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_23 = \"\"\"\n",
    "MATCH (d:Disease)\n",
    "RETURN DISTINCT d.diseaseGroup AS disease_group\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_23).to_table()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "AouS65CJ9cEh"
   },
   "source": [
    "We can see from the above query two main things:\n",
    "- First that our cleaning worked. One class contains only one value. No redundancy or artefactual class due to redundant class name.\n",
    "- Secondly and as a consequence of this data cleaning we reduce the number of groups we observed in `query_16` where we had more than 1000 groups / classes of diseases. Here we have 27 (counting `Not determined` values) distinct disease classes / groups. Reducing the number of classes has the trade-off of getting not so specific class of diseases but has the merit to really aggregate in well define groups.\n",
    "\n",
    "Having done that we can now generate the relationships between the `Variant` & `Disease` and `Variant` & `Phenotype`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "OkVdnKik9cEi",
    "outputId": "ed26987b-319a-40f2-c9d1-6615357b80ca"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f030e90>"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_24 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///curated_variant_disease_associations.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "MATCH (d:Disease {diseaseId: line.diseaseId})\n",
    "MATCH (v:Variant {variantId: line.snpId})\n",
    "MERGE (v)-[:ASSOCIATED_WITH]->(d)\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_24)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "beRFfEnD9cEl"
   },
   "source": [
    "Finally we can do the same with our `Phenotype` node and create association with the `Variant` nodes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "2swbEM5l9cEl",
    "outputId": "c528b297-cc4e-48cf-816e-b7e94f70f6cd"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<py2neo.database.Cursor at 0x7f112f025890>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_25 = \"\"\"\n",
    "LOAD CSV WITH HEADERS FROM \"file:///curated_variant_disease_associations.tsv\" AS line FIELDTERMINATOR \"\\t\"\n",
    "MATCH (p:Phenotype {phenotypeId: line.diseaseId})\n",
    "MATCH (v:Variant {variantId: line.snpId})\n",
    "MERGE (v)-[:IMPACTS_on]->(p)\n",
    "\"\"\"\n",
    "\n",
    "graph.run(query_25)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "JzZ635A-9cEo"
   },
   "source": [
    "Let's now have a look to the *schema* or data model of our database. As before just use the following command on your Neo4J app: `CALL db.schema.visualization()`.\n",
    "\n",
    "![data_model_first](data_model_final.svg)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Another dynamic way to display the data is to use the module neo4jupyter.\n",
    "This tool is designed to visualize graph database queries from Neo4j in a Jupyter Notebook."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Installation and import have been made earlier, thus, the next thing we must do is to call the neo4jupyter.init_notebook_mode() to load all the javascript code."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "var link = document.createElement(\"link\");\n",
       "\tlink.ref = \"stylesheet\";\n",
       "\tlink.type = \"text/css\";\n",
       "\tlink.href = \"https://cdnjs.cloudflare.com/ajax/libs/vis/4.8.2/vis.css\";\n",
       "\tdocument.head.appendChild(link);\n",
       "require.config({     paths: {         vis: '//cdnjs.cloudflare.com/ajax/libs/vis/4.8.2/vis.min'     } }); require(['vis'], function(vis) {  window.vis = vis; }); "
      ],
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "neo4jupyter.init_notebook_mode()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then, drawing a graph is as easy as giving the funcion neo4jupyter.draw() the py2neo graph object and the parameters\n",
    "we want to be displayed.\n",
    "Parameters :\n",
    "- options => specifies the nodes we want to visualize. The corresponding relationships are displayed too\n",
    "- physics => allows to \"manipulate\" the display by moving nodes\n",
    "- limit => limits the total number of displayed items (nodes and relationships) as the process can be ressource consumming.\n",
    "By zooming on the graph, we can even read the names of the objects and then focus on a specific chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div id=\"cc00f517-2427-4bf0-b864-fb7d27a5bdd9\" style=\"height: 400px;\"></div>\n",
       "\n",
       "<script type=\"text/javascript\">\n",
       "\n",
       "    var nodes = [{\"id\": 39466, \"label\": \"Post-surgical hypoparathyroidism\", \"group\": \"Disease\", \"title\": \"(_39466:Disease {diseaseGroup: 'Endocrine System Diseases', diseaseId: 'C0342341', diseaseName: 'Post-surgical hypoparathyroidism'})\"}, {\"id\": 79813, \"label\": \"HLA-B\", \"group\": \"Variant\", \"title\": \"(_79813:Variant {variantId: 'rs2524162', variantOf: 'HLA-B'})\"}, {\"id\": 173443, \"label\": \"Corpuscular Hemoglobin Concentration Mean\", \"group\": \"Phenotype\", \"title\": \"(_173443:Phenotype {phenotypeId: 'C4528257', phenotypeName: 'Corpuscular Hemoglobin Concentration Mean'})\"}, {\"id\": 171, \"label\": \"6\", \"group\": \"Chromosome\", \"title\": \"(_171:Chromosome {name: '6'})\"}, {\"id\": 192246, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_192246:Variant {variantId: 'rs71306579'})\"}, {\"id\": 172510, \"label\": \"Ectopic ossification in tendon tissue\", \"group\": \"Phenotype\", \"title\": \"(_172510:Phenotype {phenotypeId: 'C4023095', phenotypeName: 'Ectopic ossification in tendon tissue'})\"}, {\"id\": 191455, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_191455:Variant {variantId: 'rs6797530'})\"}, {\"id\": 133773, \"label\": \"FSTL5\", \"group\": \"Variant\", \"title\": \"(_133773:Variant {variantId: 'rs747628234', variantOf: 'FSTL5'})\"}, {\"id\": 1497, \"label\": \"Colorectal Carcinoma\", \"group\": \"Disease\", \"title\": \"(_1497:Disease {diseaseGroup: 'Neoplasms', diseaseId: 'C0009402', diseaseName: 'Colorectal Carcinoma'})\"}, {\"id\": 3, \"label\": \"4\", \"group\": \"Chromosome\", \"title\": \"(_3:Chromosome {name: '4'})\"}, {\"id\": 191056, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_191056:Variant {variantId: 'rs6656815'})\"}, {\"id\": 163645, \"label\": \"ERBB3\", \"group\": \"Variant\", \"title\": \"(_163645:Variant {variantId: 'rs931676601', variantOf: 'ERBB3'})\"}, {\"id\": 45, \"label\": \"12\", \"group\": \"Chromosome\", \"title\": \"(_45:Chromosome {name: '12'})\"}, {\"id\": 55659, \"label\": \"MIR6797\", \"group\": \"Variant\", \"title\": \"(_55659:Variant {variantId: 'rs1555841356', variantOf: 'MIR6797'})\"}, {\"id\": 74283, \"label\": \"Anemia, Diamond-Blackfan\", \"group\": \"Disease\", \"title\": \"(_74283:Disease {diseaseGroup: 'Hemic and Lymphatic Diseases', diseaseId: 'C1260899', diseaseName: 'Anemia, Diamond-Blackfan'})\"}, {\"id\": 251, \"label\": \"19\", \"group\": \"Chromosome\", \"title\": \"(_251:Chromosome {name: '19'})\"}, {\"id\": 152187, \"label\": \"LRP5\", \"group\": \"Variant\", \"title\": \"(_152187:Variant {variantId: 'rs80358312', variantOf: 'LRP5'})\"}, {\"id\": 162988, \"label\": \"Muscle hypotonia\", \"group\": \"Phenotype\", \"title\": \"(_162988:Phenotype {phenotypeId: 'C0026827', phenotypeName: 'Muscle hypotonia'})\"}, {\"id\": 125815, \"label\": \"EXUDATIVE VITREORETINOPATHY 4, AUTOSOMAL RECESSIVE\", \"group\": \"Disease\", \"title\": \"(_125815:Disease {diseaseGroup: 'Not determined', diseaseId: 'C4016839', diseaseName: 'EXUDATIVE VITREORETINOPATHY 4, AUTOSOMAL RECESSIVE'})\"}, {\"id\": 101922, \"label\": \"EXUDATIVE VITREORETINOPATHY 4 (disorder)\", \"group\": \"Disease\", \"title\": \"(_101922:Disease {diseaseGroup: 'Eye Diseases', diseaseId: 'C1866176', diseaseName: 'EXUDATIVE VITREORETINOPATHY 4 (disorder)'})\"}, {\"id\": 5, \"label\": \"11\", \"group\": \"Chromosome\", \"title\": \"(_5:Chromosome {name: '11'})\"}, {\"id\": 173081, \"label\": \"Increased serum interferon-gamma level\", \"group\": \"Phenotype\", \"title\": \"(_173081:Phenotype {phenotypeId: 'C4072900', phenotypeName: 'Increased serum interferon-gamma level'})\"}, {\"id\": 130143, \"label\": \"MYBPC3\", \"group\": \"Variant\", \"title\": \"(_130143:Variant {variantId: 'rs727504271', variantOf: 'MYBPC3'})\"}, {\"id\": 1120, \"label\": \"Hypertrophic Cardiomyopathy\", \"group\": \"Disease\", \"title\": \"(_1120:Disease {diseaseGroup: 'Cardiovascular Diseases', diseaseId: 'C0007194', diseaseName: 'Hypertrophic Cardiomyopathy'})\"}, {\"id\": 66034, \"label\": \"CLEC16A\", \"group\": \"Variant\", \"title\": \"(_66034:Variant {variantId: 'rs17673553', variantOf: 'CLEC16A'})\"}, {\"id\": 169087, \"label\": \"Allergic Reaction\", \"group\": \"Phenotype\", \"title\": \"(_169087:Phenotype {phenotypeId: 'C1527304', phenotypeName: 'Allergic Reaction'})\"}, {\"id\": 1973, \"label\": \"Diabetes Mellitus, Insulin-Dependent\", \"group\": \"Disease\", \"title\": \"(_1973:Disease {diseaseGroup: 'Endocrine System Diseases', diseaseId: 'C0011854', diseaseName: 'Diabetes Mellitus, Insulin-Dependent'})\"}, {\"id\": 141, \"label\": \"16\", \"group\": \"Chromosome\", \"title\": \"(_141:Chromosome {name: '16'})\"}, {\"id\": 45045, \"label\": \"Autosomal Dominant Emery-Dreifuss Muscular Dystrophy (disorder)\", \"group\": \"Disease\", \"title\": \"(_45045:Disease {diseaseGroup: 'Nervous System Diseases', diseaseId: 'C0410190', diseaseName: 'Autosomal Dominant Emery-Dreifuss Muscular Dystrophy (disorder)'})\"}, {\"id\": 80650, \"label\": \"MIP\", \"group\": \"Variant\", \"title\": \"(_80650:Variant {variantId: 'rs2638319', variantOf: 'MIP'})\"}, {\"id\": 168194, \"label\": \"Serum albumin measurement\", \"group\": \"Phenotype\", \"title\": \"(_168194:Phenotype {phenotypeId: 'C0523465', phenotypeName: 'Serum albumin measurement'})\"}, {\"id\": 24331, \"label\": \"LINC01705\", \"group\": \"Variant\", \"title\": \"(_24331:Variant {variantId: 'rs12125368', variantOf: 'LINC01705'})\"}, {\"id\": 10653, \"label\": \"Leukemia, Myelocytic, Acute\", \"group\": \"Disease\", \"title\": \"(_10653:Disease {diseaseGroup: 'Neoplasms', diseaseId: 'C0023467', diseaseName: 'Leukemia, Myelocytic, Acute'})\"}, {\"id\": 95, \"label\": \"1\", \"group\": \"Chromosome\", \"title\": \"(_95:Chromosome {name: '1'})\"}, {\"id\": 150583, \"label\": \"IL23R\", \"group\": \"Variant\", \"title\": \"(_150583:Variant {variantId: 'rs80174646', variantOf: 'IL23R'})\"}, {\"id\": 1692, \"label\": \"Crohn Disease\", \"group\": \"Disease\", \"title\": \"(_1692:Disease {diseaseGroup: 'Digestive System Diseases', diseaseId: 'C0010346', diseaseName: 'Crohn Disease'})\"}, {\"id\": 1494, \"label\": \"Ulcerative Colitis\", \"group\": \"Disease\", \"title\": \"(_1494:Disease {diseaseGroup: 'Digestive System Diseases', diseaseId: 'C0009324', diseaseName: 'Ulcerative Colitis'})\"}, {\"id\": 12994, \"label\": \"Psoriasis\", \"group\": \"Disease\", \"title\": \"(_12994:Disease {diseaseGroup: 'Skin and Connective Tissue Diseases', diseaseId: 'C0033860', diseaseName: 'Psoriasis'})\"}, {\"id\": 1311, \"label\": \"Cholangitis, Sclerosing\", \"group\": \"Disease\", \"title\": \"(_1311:Disease {diseaseGroup: 'Digestive System Diseases', diseaseId: 'C0008313', diseaseName: 'Cholangitis, Sclerosing'})\"}, {\"id\": 14128, \"label\": \"Ankylosing spondylitis\", \"group\": \"Disease\", \"title\": \"(_14128:Disease {diseaseGroup: 'Musculoskeletal Diseases', diseaseId: 'C0038013', diseaseName: 'Ankylosing spondylitis'})\"}, {\"id\": 31091, \"label\": \"LOC105374478\", \"group\": \"Variant\", \"title\": \"(_31091:Variant {variantId: 'rs12513376', variantOf: 'LOC105374478'})\"}, {\"id\": 164251, \"label\": \"Uric acid measurement (procedure)\", \"group\": \"Phenotype\", \"title\": \"(_164251:Phenotype {phenotypeId: 'C0202239', phenotypeName: 'Uric acid measurement (procedure)'})\"}, {\"id\": 550, \"label\": \"Arthritis, Gouty\", \"group\": \"Disease\", \"title\": \"(_550:Disease {diseaseGroup: 'Musculoskeletal Diseases', diseaseId: 'C0003868', diseaseName: 'Arthritis, Gouty'})\"}, {\"id\": 5148, \"label\": \"Gout\", \"group\": \"Disease\", \"title\": \"(_5148:Disease {diseaseGroup: 'Musculoskeletal Diseases', diseaseId: 'C0018099', diseaseName: 'Gout'})\"}, {\"id\": 152489, \"label\": \"BRCA2\", \"group\": \"Variant\", \"title\": \"(_152489:Variant {variantId: 'rs80359281', variantOf: 'BRCA2'})\"}, {\"id\": 104970, \"label\": \"BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2\", \"group\": \"Disease\", \"title\": \"(_104970:Disease {diseaseGroup: 'Not determined', diseaseId: 'C2675520', diseaseName: 'BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2'})\"}, {\"id\": 413, \"label\": \"13\", \"group\": \"Chromosome\", \"title\": \"(_413:Chromosome {name: '13'})\"}, {\"id\": 96825, \"label\": \"Adrenocortical Carcinoma, Pediatric\", \"group\": \"Disease\", \"title\": \"(_96825:Disease {diseaseGroup: 'Endocrine System Diseases', diseaseId: 'C1859973', diseaseName: 'Adrenocortical Carcinoma, Pediatric'})\"}, {\"id\": 8381, \"label\": \"NF1\", \"group\": \"Variant\", \"title\": \"(_8381:Variant {variantId: 'rs1060500319', variantOf: 'NF1'})\"}, {\"id\": 12071, \"label\": \"Neurofibromatosis 1\", \"group\": \"Disease\", \"title\": \"(_12071:Disease {diseaseGroup: 'Nervous System Diseases', diseaseId: 'C0027831', diseaseName: 'Neurofibromatosis 1'})\"}, {\"id\": 51, \"label\": \"17\", \"group\": \"Chromosome\", \"title\": \"(_51:Chromosome {name: '17'})\"}, {\"id\": 96176, \"label\": \"ORC1\", \"group\": \"Variant\", \"title\": \"(_96176:Variant {variantId: 'rs387906826', variantOf: 'ORC1'})\"}, {\"id\": 147203, \"label\": \"MEIER-GORLIN SYNDROME 1\", \"group\": \"Disease\", \"title\": \"(_147203:Disease {diseaseGroup: 'Otorhinolaryngologic Diseases', diseaseId: 'C4552001', diseaseName: 'MEIER-GORLIN SYNDROME 1'})\"}, {\"id\": 176144, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_176144:Variant {variantId: 'rs114143322'})\"}, {\"id\": 162907, \"label\": \"HLA-DRB1\", \"group\": \"Variant\", \"title\": \"(_162907:Variant {variantId: 'rs9270657', variantOf: 'HLA-DRB1'})\"}, {\"id\": 555, \"label\": \"Rheumatoid Arthritis\", \"group\": \"Disease\", \"title\": \"(_555:Disease {diseaseGroup: 'Immune System Diseases', diseaseId: 'C0003873', diseaseName: 'Rheumatoid Arthritis'})\"}, {\"id\": 13461, \"label\": \"Sarcoidosis\", \"group\": \"Disease\", \"title\": \"(_13461:Disease {diseaseGroup: 'Hemic and Lymphatic Diseases', diseaseId: 'C0036202', diseaseName: 'Sarcoidosis'})\"}, {\"id\": 169760, \"label\": \"Externally rotated hips\", \"group\": \"Phenotype\", \"title\": \"(_169760:Phenotype {phenotypeId: 'C1846339', phenotypeName: 'Externally rotated hips'})\"}, {\"id\": 151075, \"label\": \"DM1-AS\", \"group\": \"Variant\", \"title\": \"(_151075:Variant {variantId: 'rs80356463', variantOf: 'DM1-AS'})\"}, {\"id\": 103460, \"label\": \"Branchiootorenal Syndrome 2\", \"group\": \"Disease\", \"title\": \"(_103460:Disease {diseaseGroup: 'Congenital, Hereditary, and Neonatal Diseases and Abnormalities', diseaseId: 'C1970479', diseaseName: 'Branchiootorenal Syndrome 2'})\"}, {\"id\": 85366, \"label\": \"PAROXYSMAL EXTREME PAIN DISORDER\", \"group\": \"Disease\", \"title\": \"(_85366:Disease {diseaseGroup: 'Pathological Conditions, Signs and Symptoms', diseaseId: 'C1833661', diseaseName: 'PAROXYSMAL EXTREME PAIN DISORDER'})\"}, {\"id\": 57308, \"label\": \"STAT1\", \"group\": \"Variant\", \"title\": \"(_57308:Variant {variantId: 'rs1559011859', variantOf: 'STAT1'})\"}, {\"id\": 124716, \"label\": \"IMMUNODEFICIENCY 31A\", \"group\": \"Disease\", \"title\": \"(_124716:Disease {diseaseGroup: 'Not determined', diseaseId: 'C4013950', diseaseName: 'IMMUNODEFICIENCY 31A'})\"}, {\"id\": 111410, \"label\": \"IMMUNODEFICIENCY 31B\", \"group\": \"Disease\", \"title\": \"(_111410:Disease {diseaseGroup: 'Not determined', diseaseId: 'C3151088', diseaseName: 'IMMUNODEFICIENCY 31B'})\"}, {\"id\": 116640, \"label\": \"CANDIDIASIS, FAMILIAL, 7\", \"group\": \"Disease\", \"title\": \"(_116640:Disease {diseaseGroup: 'Not determined', diseaseId: 'C3279990', diseaseName: 'CANDIDIASIS, FAMILIAL, 7'})\"}, {\"id\": 59, \"label\": \"2\", \"group\": \"Chromosome\", \"title\": \"(_59:Chromosome {name: '2'})\"}, {\"id\": 2404, \"label\": \"SMARCA4\", \"group\": \"Variant\", \"title\": \"(_2404:Variant {variantId: 'rs10423733', variantOf: 'SMARCA4'})\"}, {\"id\": 132704, \"label\": \"MATURITY-ONSET DIABETES OF THE YOUNG, TYPE 14\", \"group\": \"Disease\", \"title\": \"(_132704:Disease {diseaseGroup: 'Not determined', diseaseId: 'C4225299', diseaseName: 'MATURITY-ONSET DIABETES OF THE YOUNG, TYPE 14'})\"}, {\"id\": 107551, \"label\": \"LINC02702\", \"group\": \"Variant\", \"title\": \"(_107551:Variant {variantId: 'rs499790', variantOf: 'LINC02702'})\"}, {\"id\": 167856, \"label\": \"Serum HDL cholesterol measurement\", \"group\": \"Phenotype\", \"title\": \"(_167856:Phenotype {phenotypeId: 'C0428472', phenotypeName: 'Serum HDL cholesterol measurement'})\"}, {\"id\": 164250, \"label\": \"Triglycerides measurement\", \"group\": \"Phenotype\", \"title\": \"(_164250:Phenotype {phenotypeId: 'C0202236', phenotypeName: 'Triglycerides measurement'})\"}, {\"id\": 169056, \"label\": \"Serum total cholesterol measurement\", \"group\": \"Phenotype\", \"title\": \"(_169056:Phenotype {phenotypeId: 'C1445957', phenotypeName: 'Serum total cholesterol measurement'})\"}, {\"id\": 167060, \"label\": \"High density lipoprotein measurement\", \"group\": \"Phenotype\", \"title\": \"(_167060:Phenotype {phenotypeId: 'C0392885', phenotypeName: 'High density lipoprotein measurement'})\"}, {\"id\": 81504, \"label\": \"KIF21A\", \"group\": \"Variant\", \"title\": \"(_81504:Variant {variantId: 'rs267607200', variantOf: 'KIF21A'})\"}, {\"id\": 106799, \"label\": \"FIBROSIS OF EXTRAOCULAR MUSCLES, CONGENITAL, 3B (disorder)\", \"group\": \"Disease\", \"title\": \"(_106799:Disease {diseaseGroup: 'Nervous System Diseases', diseaseId: 'C2751105', diseaseName: 'FIBROSIS OF EXTRAOCULAR MUSCLES, CONGENITAL, 3B (disorder)'})\"}, {\"id\": 132935, \"label\": \"TBX1\", \"group\": \"Variant\", \"title\": \"(_132935:Variant {variantId: 'rs74315522', variantOf: 'TBX1'})\"}, {\"id\": 21002, \"label\": \"Shprintzen syndrome\", \"group\": \"Disease\", \"title\": \"(_21002:Disease {diseaseGroup: 'Cardiovascular Diseases', diseaseId: 'C0220704', diseaseName: 'Shprintzen syndrome'})\"}, {\"id\": 233, \"label\": \"22\", \"group\": \"Chromosome\", \"title\": \"(_233:Chromosome {name: '22'})\"}, {\"id\": 105461, \"label\": \"BPTF\", \"group\": \"Variant\", \"title\": \"(_105461:Variant {variantId: 'rs4790981', variantOf: 'BPTF'})\"}, {\"id\": 168985, \"label\": \"Body mass index\", \"group\": \"Phenotype\", \"title\": \"(_168985:Phenotype {phenotypeId: 'C1305855', phenotypeName: 'Body mass index'})\"}, {\"id\": 26964, \"label\": \"SLC45A2\", \"group\": \"Variant\", \"title\": \"(_26964:Variant {variantId: 'rs121912621', variantOf: 'SLC45A2'})\"}, {\"id\": 88682, \"label\": \"Oculocutaneous Albinism, Type IV\", \"group\": \"Disease\", \"title\": \"(_88682:Disease {diseaseGroup: 'Skin and Connective Tissue Diseases', diseaseId: 'C1847836', diseaseName: 'Oculocutaneous Albinism, Type IV'})\"}, {\"id\": 9, \"label\": \"5\", \"group\": \"Chromosome\", \"title\": \"(_9:Chromosome {name: '5'})\"}, {\"id\": 1411, \"label\": \"DPP10\", \"group\": \"Variant\", \"title\": \"(_1411:Variant {variantId: 'rs10195136', variantOf: 'DPP10'})\"}, {\"id\": 86723, \"label\": \"SCOLIOSIS, ISOLATED, SUSCEPTIBILITY TO, 3\", \"group\": \"Disease\", \"title\": \"(_86723:Disease {diseaseGroup: 'Not determined', diseaseId: 'C1837461', diseaseName: 'SCOLIOSIS, ISOLATED, SUSCEPTIBILITY TO, 3'})\"}, {\"id\": 45183, \"label\": \"Adolescent idiopathic scoliosis\", \"group\": \"Disease\", \"title\": \"(_45183:Disease {diseaseGroup: 'Musculoskeletal Diseases', diseaseId: 'C0410702', diseaseName: 'Adolescent idiopathic scoliosis'})\"}, {\"id\": 134021, \"label\": \"SHPK\", \"group\": \"Variant\", \"title\": \"(_134021:Variant {variantId: 'rs748544120', variantOf: 'SHPK'})\"}, {\"id\": 75881, \"label\": \"Deficiency of sedoheptulokinase\", \"group\": \"Disease\", \"title\": \"(_75881:Disease {diseaseGroup: 'Not determined', diseaseId: 'C1291373', diseaseName: 'Deficiency of sedoheptulokinase'})\"}, {\"id\": 32935, \"label\": \"PPP2R3A\", \"group\": \"Variant\", \"title\": \"(_32935:Variant {variantId: 'rs1278493', variantOf: 'PPP2R3A'})\"}, {\"id\": 164258, \"label\": \"Waist-Hip Ratio\", \"group\": \"Phenotype\", \"title\": \"(_164258:Phenotype {phenotypeId: 'C0205682', phenotypeName: 'Waist-Hip Ratio'})\"}, {\"id\": 637, \"label\": \"Atrial Fibrillation\", \"group\": \"Disease\", \"title\": \"(_637:Disease {diseaseGroup: 'Cardiovascular Diseases', diseaseId: 'C0004238', diseaseName: 'Atrial Fibrillation'})\"}, {\"id\": 72, \"label\": \"3\", \"group\": \"Chromosome\", \"title\": \"(_72:Chromosome {name: '3'})\"}, {\"id\": 70139, \"label\": \"CATSPER1\", \"group\": \"Variant\", \"title\": \"(_70139:Variant {variantId: 'rs193929391', variantOf: 'CATSPER1'})\"}, {\"id\": 107001, \"label\": \"Spermatogenic Failure 7\", \"group\": \"Disease\", \"title\": \"(_107001:Disease {diseaseGroup: 'Male Urogenital Diseases', diseaseId: 'C2751811', diseaseName: 'Spermatogenic Failure 7'})\"}, {\"id\": 38623, \"label\": \"TSC2\", \"group\": \"Variant\", \"title\": \"(_38623:Variant {variantId: 'rs137854175', variantOf: 'TSC2'})\"}, {\"id\": 96862, \"label\": \"TUBEROUS SCLEROSIS 2 (disorder)\", \"group\": \"Disease\", \"title\": \"(_96862:Disease {diseaseGroup: 'Nervous System Diseases', diseaseId: 'C1860707', diseaseName: 'TUBEROUS SCLEROSIS 2 (disorder)'})\"}, {\"id\": 187095, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_187095:Variant {variantId: 'rs4074726'})\"}, {\"id\": 68043, \"label\": \"Brown-Vialetto-Van Laere Syndrome 1\", \"group\": \"Disease\", \"title\": \"(_68043:Disease {diseaseGroup: 'Otorhinolaryngologic Diseases', diseaseId: 'C0796274', diseaseName: 'Brown-Vialetto-Van Laere Syndrome 1'})\"}, {\"id\": 39353, \"label\": \"GJA10\", \"group\": \"Variant\", \"title\": \"(_39353:Variant {variantId: 'rs138866907', variantOf: 'GJA10'})\"}, {\"id\": 168381, \"label\": \"mathematical ability\", \"group\": \"Phenotype\", \"title\": \"(_168381:Phenotype {phenotypeId: 'C0596887', phenotypeName: 'mathematical ability'})\"}, {\"id\": 136618, \"label\": \"MMAB\", \"group\": \"Variant\", \"title\": \"(_136618:Variant {variantId: 'rs756414548', variantOf: 'MMAB'})\"}, {\"id\": 90707, \"label\": \"Methylmalonic aciduria cblB type\", \"group\": \"Disease\", \"title\": \"(_90707:Disease {diseaseGroup: 'Nutritional and Metabolic Diseases', diseaseId: 'C1855102', diseaseName: 'Methylmalonic aciduria cblB type'})\"}, {\"id\": 162041, \"label\": \"LEF1\", \"group\": \"Variant\", \"title\": \"(_162041:Variant {variantId: 'rs898518', variantOf: 'LEF1'})\"}, {\"id\": 68797, \"label\": \"Small Lymphocytic Lymphoma\", \"group\": \"Disease\", \"title\": \"(_68797:Disease {diseaseGroup: 'Hemic and Lymphatic Diseases', diseaseId: 'C0855095', diseaseName: 'Small Lymphocytic Lymphoma'})\"}, {\"id\": 10614, \"label\": \"Chronic Lymphocytic Leukemia\", \"group\": \"Disease\", \"title\": \"(_10614:Disease {diseaseGroup: 'Hemic and Lymphatic Diseases', diseaseId: 'C0023434', diseaseName: 'Chronic Lymphocytic Leukemia'})\"}, {\"id\": 54064, \"label\": \"SPATA22\", \"group\": \"Variant\", \"title\": \"(_54064:Variant {variantId: 'rs1555538148', variantOf: 'SPATA22'})\"}, {\"id\": 20488, \"label\": \"Canavan Disease\", \"group\": \"Disease\", \"title\": \"(_20488:Disease {diseaseGroup: 'Nervous System Diseases', diseaseId: 'C0206307', diseaseName: 'Canavan Disease'})\"}, {\"id\": 154088, \"label\": \"THRB\", \"group\": \"Variant\", \"title\": \"(_154088:Variant {variantId: 'rs826238', variantOf: 'THRB'})\"}, {\"id\": 48509, \"label\": \"APC\", \"group\": \"Variant\", \"title\": \"(_48509:Variant {variantId: 'rs1554076217', variantOf: 'APC'})\"}, {\"id\": 105811, \"label\": \"Polyposis, Adenomatous Intestinal\", \"group\": \"Disease\", \"title\": \"(_105811:Disease {diseaseGroup: 'Neoplasms', diseaseId: 'C2713442', diseaseName: 'Polyposis, Adenomatous Intestinal'})\"}, {\"id\": 92195, \"label\": \"TACR3\", \"group\": \"Variant\", \"title\": \"(_92195:Variant {variantId: 'rs3733632', variantOf: 'TACR3'})\"}, {\"id\": 168994, \"label\": \"Age at menarche\", \"group\": \"Phenotype\", \"title\": \"(_168994:Phenotype {phenotypeId: 'C1314691', phenotypeName: 'Age at menarche'})\"}, {\"id\": 161849, \"label\": \"Body Height\", \"group\": \"Phenotype\", \"title\": \"(_161849:Phenotype {phenotypeId: 'C0005890', phenotypeName: 'Body Height'})\"}, {\"id\": 151648, \"label\": \"BRCA1\", \"group\": \"Variant\", \"title\": \"(_151648:Variant {variantId: 'rs80357571', variantOf: 'BRCA1'})\"}, {\"id\": 105103, \"label\": \"BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 1\", \"group\": \"Disease\", \"title\": \"(_105103:Disease {diseaseGroup: 'Not determined', diseaseId: 'C2676676', diseaseName: 'BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 1'})\"}, {\"id\": 176640, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_176640:Variant {variantId: 'rs11677787'})\"}, {\"id\": 84863, \"label\": \"MIR6891\", \"group\": \"Variant\", \"title\": \"(_84863:Variant {variantId: 'rs2854001', variantOf: 'MIR6891'})\"}, {\"id\": 59177, \"label\": \"MYBPC3\", \"group\": \"Variant\", \"title\": \"(_59177:Variant {variantId: 'rs1565623216', variantOf: 'MYBPC3'})\"}, {\"id\": 24937, \"label\": \"FOXN3\", \"group\": \"Variant\", \"title\": \"(_24937:Variant {variantId: 'rs12147030', variantOf: 'FOXN3'})\"}, {\"id\": 137, \"label\": \"14\", \"group\": \"Chromosome\", \"title\": \"(_137:Chromosome {name: '14'})\"}, {\"id\": 118795, \"label\": \"USP10\", \"group\": \"Variant\", \"title\": \"(_118795:Variant {variantId: 'rs62048856', variantOf: 'USP10'})\"}, {\"id\": 54214, \"label\": \"COL1A1\", \"group\": \"Variant\", \"title\": \"(_54214:Variant {variantId: 'rs1555571755', variantOf: 'COL1A1'})\"}, {\"id\": 10847, \"label\": \"Lobstein Disease\", \"group\": \"Disease\", \"title\": \"(_10847:Disease {diseaseGroup: 'Musculoskeletal Diseases', diseaseId: 'C0023931', diseaseName: 'Lobstein Disease'})\"}, {\"id\": 98750, \"label\": \"B3GALT6\", \"group\": \"Variant\", \"title\": \"(_98750:Variant {variantId: 'rs397514718', variantOf: 'B3GALT6'})\"}, {\"id\": 171981, \"label\": \"SPONDYLOEPIMETAPHYSEAL DYSPLASIA WITH JOINT LAXITY, TYPE 1\", \"group\": \"Phenotype\", \"title\": \"(_171981:Phenotype {phenotypeId: 'C4017377', phenotypeName: 'SPONDYLOEPIMETAPHYSEAL DYSPLASIA WITH JOINT LAXITY, TYPE 1'})\"}, {\"id\": 49225, \"label\": \"Spondyloepimetaphyseal Dysplasia With Joint Laxity\", \"group\": \"Disease\", \"title\": \"(_49225:Disease {diseaseGroup: 'Musculoskeletal Diseases', diseaseId: 'C0432243', diseaseName: 'Spondyloepimetaphyseal Dysplasia With Joint Laxity'})\"}, {\"id\": 133199, \"label\": \"PCCA\", \"group\": \"Variant\", \"title\": \"(_133199:Variant {variantId: 'rs745571507', variantOf: 'PCCA'})\"}, {\"id\": 31535, \"label\": \"Propionic acidemia\", \"group\": \"Disease\", \"title\": \"(_31535:Disease {diseaseGroup: 'Nutritional and Metabolic Diseases', diseaseId: 'C0268579', diseaseName: 'Propionic acidemia'})\"}, {\"id\": 87955, \"label\": \"Pontocerebellar Hypoplasia Type 1\", \"group\": \"Disease\", \"title\": \"(_87955:Disease {diseaseGroup: 'Nervous System Diseases', diseaseId: 'C1843504', diseaseName: 'Pontocerebellar Hypoplasia Type 1'})\"}, {\"id\": 153599, \"label\": \"BRCA2\", \"group\": \"Variant\", \"title\": \"(_153599:Variant {variantId: 'rs81002795', variantOf: 'BRCA2'})\"}, {\"id\": 48433, \"label\": \"HSD17B4\", \"group\": \"Variant\", \"title\": \"(_48433:Variant {variantId: 'rs1554065254', variantOf: 'HSD17B4'})\"}, {\"id\": 166797, \"label\": \"Bifunctional peroxisomal enzyme deficiency\", \"group\": \"Phenotype\", \"title\": \"(_166797:Phenotype {phenotypeId: 'C0342870', phenotypeName: 'Bifunctional peroxisomal enzyme deficiency'})\"}, {\"id\": 37989, \"label\": \"BMPR2\", \"group\": \"Variant\", \"title\": \"(_37989:Variant {variantId: 'rs137852754', variantOf: 'BMPR2'})\"}, {\"id\": 147249, \"label\": \"Pulmonary Hypertension, Primary, 1\", \"group\": \"Disease\", \"title\": \"(_147249:Disease {diseaseGroup: 'Respiratory Tract Diseases', diseaseId: 'C4552070', diseaseName: 'Pulmonary Hypertension, Primary, 1'})\"}, {\"id\": 184220, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_184220:Variant {variantId: 'rs2398227'})\"}, {\"id\": 11743, \"label\": \"EFEMP2\", \"group\": \"Variant\", \"title\": \"(_11743:Variant {variantId: 'rs10896064', variantOf: 'EFEMP2'})\"}, {\"id\": 66478, \"label\": \"Basal Cell Cancer\", \"group\": \"Disease\", \"title\": \"(_66478:Disease {diseaseGroup: 'Neoplasms', diseaseId: 'C0751676', diseaseName: 'Basal Cell Cancer'})\"}, {\"id\": 1061, \"label\": \"Basal cell carcinoma\", \"group\": \"Disease\", \"title\": \"(_1061:Disease {diseaseGroup: 'Neoplasms', diseaseId: 'C0007117', diseaseName: 'Basal cell carcinoma'})\"}, {\"id\": 20839, \"label\": \"Basal Cell Neoplasm\", \"group\": \"Disease\", \"title\": \"(_20839:Disease {diseaseGroup: 'Neoplasms', diseaseId: 'C0206710', diseaseName: 'Basal Cell Neoplasm'})\"}, {\"id\": 83343, \"label\": \"SEMA6D\", \"group\": \"Variant\", \"title\": \"(_83343:Variant {variantId: 'rs281323', variantOf: 'SEMA6D'})\"}, {\"id\": 74640, \"label\": \"Attention deficit hyperactivity disorder\", \"group\": \"Disease\", \"title\": \"(_74640:Disease {diseaseGroup: 'Mental Disorders', diseaseId: 'C1263846', diseaseName: 'Attention deficit hyperactivity disorder'})\"}, {\"id\": 97, \"label\": \"15\", \"group\": \"Chromosome\", \"title\": \"(_97:Chromosome {name: '15'})\"}, {\"id\": 143032, \"label\": \"YARS2\", \"group\": \"Variant\", \"title\": \"(_143032:Variant {variantId: 'rs775256289', variantOf: 'YARS2'})\"}, {\"id\": 146653, \"label\": \"CPT1C\", \"group\": \"Variant\", \"title\": \"(_146653:Variant {variantId: 'rs786204767', variantOf: 'CPT1C'})\"}, {\"id\": 133564, \"label\": \"SPASTIC PARAPLEGIA 73, AUTOSOMAL DOMINANT\", \"group\": \"Disease\", \"title\": \"(_133564:Disease {diseaseGroup: 'Not determined', diseaseId: 'C4225387', diseaseName: 'SPASTIC PARAPLEGIA 73, AUTOSOMAL DOMINANT'})\"}, {\"id\": 75082, \"label\": \"DPP6\", \"group\": \"Variant\", \"title\": \"(_75082:Variant {variantId: 'rs2110267', variantOf: 'DPP6'})\"}, {\"id\": 33, \"label\": \"7\", \"group\": \"Chromosome\", \"title\": \"(_33:Chromosome {name: '7'})\"}, {\"id\": 26420, \"label\": \"ACVRL1\", \"group\": \"Variant\", \"title\": \"(_26420:Variant {variantId: 'rs121909285', variantOf: 'ACVRL1'})\"}, {\"id\": 86870, \"label\": \"OSLER-RENDU-WEBER SYNDROME 2\", \"group\": \"Disease\", \"title\": \"(_86870:Disease {diseaseGroup: 'Cardiovascular Diseases', diseaseId: 'C1838163', diseaseName: 'OSLER-RENDU-WEBER SYNDROME 2'})\"}, {\"id\": 16533, \"label\": \"NF1\", \"group\": \"Variant\", \"title\": \"(_16533:Variant {variantId: 'rs1131691078', variantOf: 'NF1'})\"}, {\"id\": 104645, \"label\": \"TGM5\", \"group\": \"Variant\", \"title\": \"(_104645:Variant {variantId: 'rs471122', variantOf: 'TGM5'})\"}, {\"id\": 164103, \"label\": \"Lymphocyte Count measurement\", \"group\": \"Phenotype\", \"title\": \"(_164103:Phenotype {phenotypeId: 'C0200635', phenotypeName: 'Lymphocyte Count measurement'})\"}, {\"id\": 185969, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_185969:Variant {variantId: 'rs310291'})\"}, {\"id\": 87224, \"label\": \"MAJOR AFFECTIVE DISORDER 2\", \"group\": \"Disease\", \"title\": \"(_87224:Disease {diseaseGroup: 'Mental Disorders', diseaseId: 'C1839839', diseaseName: 'MAJOR AFFECTIVE DISORDER 2'})\"}, {\"id\": 139383, \"label\": \"LINC01619\", \"group\": \"Variant\", \"title\": \"(_139383:Variant {variantId: 'rs76439972', variantOf: 'LINC01619'})\"}, {\"id\": 70289, \"label\": \"RALGAPA1\", \"group\": \"Variant\", \"title\": \"(_70289:Variant {variantId: 'rs1957830', variantOf: 'RALGAPA1'})\"}, {\"id\": 168928, \"label\": \"Finding of Mean Corpuscular Hemoglobin\", \"group\": \"Phenotype\", \"title\": \"(_168928:Phenotype {phenotypeId: 'C1261502', phenotypeName: 'Finding of Mean Corpuscular Hemoglobin'})\"}, {\"id\": 69300, \"label\": \"COL1A2-AS1\", \"group\": \"Variant\", \"title\": \"(_69300:Variant {variantId: 'rs191979398', variantOf: 'COL1A2-AS1'})\"}, {\"id\": 162523, \"label\": \"Forced expiratory volume function\", \"group\": \"Phenotype\", \"title\": \"(_162523:Phenotype {phenotypeId: 'C0016529', phenotypeName: 'Forced expiratory volume function'})\"}, {\"id\": 150499, \"label\": \"HERC4\", \"group\": \"Variant\", \"title\": \"(_150499:Variant {variantId: 'rs80117551', variantOf: 'HERC4'})\"}, {\"id\": 167, \"label\": \"10\", \"group\": \"Chromosome\", \"title\": \"(_167:Chromosome {name: '10'})\"}, {\"id\": 192184, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_192184:Variant {variantId: 'rs7107004'})\"}, {\"id\": 107461, \"label\": \"Nephrotic syndrome with focal glomerulonephritis\", \"group\": \"Disease\", \"title\": \"(_107461:Disease {diseaseGroup: 'Not determined', diseaseId: 'C2902889', diseaseName: 'Nephrotic syndrome with focal glomerulonephritis'})\"}, {\"id\": 173652, \"label\": \"Hypercholanemia\", \"group\": \"Phenotype\", \"title\": \"(_173652:Phenotype {phenotypeId: 'C4747366', phenotypeName: 'Hypercholanemia'})\"}, {\"id\": 153431, \"label\": \"GSDMA\", \"group\": \"Variant\", \"title\": \"(_153431:Variant {variantId: 'rs8076474', variantOf: 'GSDMA'})\"}, {\"id\": 24962, \"label\": \"Childhood asthma\", \"group\": \"Disease\", \"title\": \"(_24962:Disease {diseaseGroup: 'Immune System Diseases', diseaseId: 'C0264408', diseaseName: 'Childhood asthma'})\"}, {\"id\": 85802, \"label\": \"RPL36A-HNRNPH2\", \"group\": \"Variant\", \"title\": \"(_85802:Variant {variantId: 'rs28935196', variantOf: 'RPL36A-HNRNPH2'})\"}, {\"id\": 367, \"label\": \"Fabry Disease\", \"group\": \"Disease\", \"title\": \"(_367:Disease {diseaseGroup: 'Cardiovascular Diseases', diseaseId: 'C0002986', diseaseName: 'Fabry Disease'})\"}, {\"id\": 742, \"label\": \"X\", \"group\": \"Chromosome\", \"title\": \"(_742:Chromosome {name: 'X'})\"}, {\"id\": 117328, \"label\": \"MACROD2\", \"group\": \"Variant\", \"title\": \"(_117328:Variant {variantId: 'rs6110278', variantOf: 'MACROD2'})\"}, {\"id\": 164118, \"label\": \"Platelet mean volume determination (procedure)\", \"group\": \"Phenotype\", \"title\": \"(_164118:Phenotype {phenotypeId: 'C0200665', phenotypeName: 'Platelet mean volume determination (procedure)'})\"}, {\"id\": 69, \"label\": \"20\", \"group\": \"Chromosome\", \"title\": \"(_69:Chromosome {name: '20'})\"}, {\"id\": 58718, \"label\": \"LOC102723566\", \"group\": \"Variant\", \"title\": \"(_58718:Variant {variantId: 'rs1564452075', variantOf: 'LOC102723566'})\"}, {\"id\": 14813, \"label\": \"Hereditary hemorrhagic telangiectasia\", \"group\": \"Disease\", \"title\": \"(_14813:Disease {diseaseGroup: 'Cardiovascular Diseases', diseaseId: 'C0039445', diseaseName: 'Hereditary hemorrhagic telangiectasia'})\"}, {\"id\": 42, \"label\": \"9\", \"group\": \"Chromosome\", \"title\": \"(_42:Chromosome {name: '9'})\"}, {\"id\": 158259, \"label\": \"TSEN15\", \"group\": \"Variant\", \"title\": \"(_158259:Variant {variantId: 'rs879253780', variantOf: 'TSEN15'})\"}, {\"id\": 138268, \"label\": \"PONTOCEREBELLAR HYPOPLASIA, TYPE 2F\", \"group\": \"Disease\", \"title\": \"(_138268:Disease {diseaseGroup: 'Not determined', diseaseId: 'C4310757', diseaseName: 'PONTOCEREBELLAR HYPOPLASIA, TYPE 2F'})\"}, {\"id\": 25918, \"label\": \"SLC7A7\", \"group\": \"Variant\", \"title\": \"(_25918:Variant {variantId: 'rs121908676', variantOf: 'SLC7A7'})\"}, {\"id\": 31615, \"label\": \"Lysinuric Protein Intolerance\", \"group\": \"Disease\", \"title\": \"(_31615:Disease {diseaseGroup: 'Nutritional and Metabolic Diseases', diseaseId: 'C0268647', diseaseName: 'Lysinuric Protein Intolerance'})\"}, {\"id\": 111716, \"label\": \"RAB11FIP5\", \"group\": \"Variant\", \"title\": \"(_111716:Variant {variantId: 'rs58649746', variantOf: 'RAB11FIP5'})\"}, {\"id\": 161891, \"label\": \"Bone Density\", \"group\": \"Phenotype\", \"title\": \"(_161891:Phenotype {phenotypeId: 'C0005938', phenotypeName: 'Bone Density'})\"}, {\"id\": 80707, \"label\": \"TET2-AS1\", \"group\": \"Variant\", \"title\": \"(_80707:Variant {variantId: 'rs2647249', variantOf: 'TET2-AS1'})\"}, {\"id\": 162934, \"label\": \"Intelligence\", \"group\": \"Phenotype\", \"title\": \"(_162934:Phenotype {phenotypeId: 'C0021704', phenotypeName: 'Intelligence'})\"}, {\"id\": 62530, \"label\": \"LOC105375751\", \"group\": \"Variant\", \"title\": \"(_62530:Variant {variantId: 'rs16901549', variantOf: 'LOC105375751'})\"}, {\"id\": 130, \"label\": \"8\", \"group\": \"Chromosome\", \"title\": \"(_130:Chromosome {name: '8'})\"}, {\"id\": 54128, \"label\": \"FANCA\", \"group\": \"Variant\", \"title\": \"(_54128:Variant {variantId: 'rs1555552506', variantOf: 'FANCA'})\"}, {\"id\": 117398, \"label\": \"FANCONI ANEMIA, COMPLEMENTATION GROUP A (disorder)\", \"group\": \"Disease\", \"title\": \"(_117398:Disease {diseaseGroup: 'Hemic and Lymphatic Diseases', diseaseId: 'C3469521', diseaseName: 'FANCONI ANEMIA, COMPLEMENTATION GROUP A (disorder)'})\"}, {\"id\": 97475, \"label\": \"BRCA2\", \"group\": \"Variant\", \"title\": \"(_97475:Variant {variantId: 'rs397507814', variantOf: 'BRCA2'})\"}, {\"id\": 77962, \"label\": \"NIPBL\", \"group\": \"Variant\", \"title\": \"(_77962:Variant {variantId: 'rs2307857', variantOf: 'NIPBL'})\"}, {\"id\": 149131, \"label\": \"RASA1\", \"group\": \"Variant\", \"title\": \"(_149131:Variant {variantId: 'rs797044451', variantOf: 'RASA1'})\"}, {\"id\": 87474, \"label\": \"CAPILLARY MALFORMATION-ARTERIOVENOUS MALFORMATION (disorder)\", \"group\": \"Disease\", \"title\": \"(_87474:Disease {diseaseGroup: 'Cardiovascular Diseases', diseaseId: 'C1842180', diseaseName: 'CAPILLARY MALFORMATION-ARTERIOVENOUS MALFORMATION (disorder)'})\"}, {\"id\": 153207, \"label\": \"CAPILLARY MALFORMATION-ARTERIOVENOUS MALFORMATION 1\", \"group\": \"Disease\", \"title\": \"(_153207:Disease {diseaseGroup: 'Not determined', diseaseId: 'C4747394', diseaseName: 'CAPILLARY MALFORMATION-ARTERIOVENOUS MALFORMATION 1'})\"}, {\"id\": 67349, \"label\": \"LOC112268416\", \"group\": \"Variant\", \"title\": \"(_67349:Variant {variantId: 'rs1802575', variantOf: 'LOC112268416'})\"}, {\"id\": 45624, \"label\": \"Bilateral renal dysplasia\", \"group\": \"Disease\", \"title\": \"(_45624:Disease {diseaseGroup: 'Not determined', diseaseId: 'C0431698', diseaseName: 'Bilateral renal dysplasia'})\"}, {\"id\": 120292, \"label\": \"MSH2\", \"group\": \"Variant\", \"title\": \"(_120292:Variant {variantId: 'rs63750245', variantOf: 'MSH2'})\"}, {\"id\": 78833, \"label\": \"Hereditary Nonpolyposis Colorectal Cancer\", \"group\": \"Disease\", \"title\": \"(_78833:Disease {diseaseGroup: 'Nutritional and Metabolic Diseases', diseaseId: 'C1333990', diseaseName: 'Hereditary Nonpolyposis Colorectal Cancer'})\"}, {\"id\": 67712, \"label\": \"Chromosome 18, tetrasomy 18p\", \"group\": \"Disease\", \"title\": \"(_67712:Disease {diseaseGroup: 'Pathological Conditions, Signs and Symptoms', diseaseId: 'C0795868', diseaseName: 'Chromosome 18, tetrasomy 18p'})\"}, {\"id\": 181517, \"label\": \"\", \"group\": \"Variant\", \"title\": \"(_181517:Variant {variantId: 'rs17084297'})\"}, {\"id\": 6748, \"label\": \"GPR68\", \"group\": \"Variant\", \"title\": \"(_6748:Variant {variantId: 'rs1057517671', variantOf: 'GPR68'})\"}, {\"id\": 137585, \"label\": \"AMELOGENESIS IMPERFECTA, HYPOMATURATION TYPE, IIA6\", \"group\": \"Disease\", \"title\": \"(_137585:Disease {diseaseGroup: 'Not determined', diseaseId: 'C4310665', diseaseName: 'AMELOGENESIS IMPERFECTA, HYPOMATURATION TYPE, IIA6'})\"}, {\"id\": 78923, \"label\": \"Intraurothelial Neoplasia\", \"group\": \"Disease\", \"title\": \"(_78923:Disease {diseaseGroup: 'Not determined', diseaseId: 'C1334266', diseaseName: 'Intraurothelial Neoplasia'})\"}];\n",
       "    var edges = [{\"from\": 79813, \"to\": 173443, \"label\": \"IMPACTS_on\"}, {\"from\": 79813, \"to\": 171, \"label\": \"LOCATED_IN\"}, {\"from\": 133773, \"to\": 1497, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 133773, \"to\": 3, \"label\": \"LOCATED_IN\"}, {\"from\": 163645, \"to\": 45, \"label\": \"LOCATED_IN\"}, {\"from\": 55659, \"to\": 74283, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 55659, \"to\": 251, \"label\": \"LOCATED_IN\"}, {\"from\": 152187, \"to\": 162988, \"label\": \"IMPACTS_on\"}, {\"from\": 152187, \"to\": 125815, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 152187, \"to\": 101922, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 152187, \"to\": 5, \"label\": \"LOCATED_IN\"}, {\"from\": 130143, \"to\": 1120, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 130143, \"to\": 5, \"label\": \"LOCATED_IN\"}, {\"from\": 66034, \"to\": 169087, \"label\": \"IMPACTS_on\"}, {\"from\": 66034, \"to\": 1973, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 66034, \"to\": 141, \"label\": \"LOCATED_IN\"}, {\"from\": 80650, \"to\": 168194, \"label\": \"IMPACTS_on\"}, {\"from\": 80650, \"to\": 45, \"label\": \"LOCATED_IN\"}, {\"from\": 24331, \"to\": 10653, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 24331, \"to\": 95, \"label\": \"LOCATED_IN\"}, {\"from\": 150583, \"to\": 1692, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 150583, \"to\": 1494, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 150583, \"to\": 12994, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 150583, \"to\": 1311, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 150583, \"to\": 14128, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 150583, \"to\": 95, \"label\": \"LOCATED_IN\"}, {\"from\": 31091, \"to\": 164251, \"label\": \"IMPACTS_on\"}, {\"from\": 31091, \"to\": 550, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 31091, \"to\": 5148, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 31091, \"to\": 3, \"label\": \"LOCATED_IN\"}, {\"from\": 152489, \"to\": 104970, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 152489, \"to\": 413, \"label\": \"LOCATED_IN\"}, {\"from\": 8381, \"to\": 12071, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 8381, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 96176, \"to\": 147203, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 96176, \"to\": 95, \"label\": \"LOCATED_IN\"}, {\"from\": 162907, \"to\": 555, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 162907, \"to\": 13461, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 162907, \"to\": 171, \"label\": \"LOCATED_IN\"}, {\"from\": 151075, \"to\": 103460, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 151075, \"to\": 251, \"label\": \"LOCATED_IN\"}, {\"from\": 57308, \"to\": 124716, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 57308, \"to\": 111410, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 57308, \"to\": 116640, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 57308, \"to\": 59, \"label\": \"LOCATED_IN\"}, {\"from\": 2404, \"to\": 168194, \"label\": \"IMPACTS_on\"}, {\"from\": 2404, \"to\": 251, \"label\": \"LOCATED_IN\"}, {\"from\": 107551, \"to\": 167856, \"label\": \"IMPACTS_on\"}, {\"from\": 107551, \"to\": 164250, \"label\": \"IMPACTS_on\"}, {\"from\": 107551, \"to\": 169056, \"label\": \"IMPACTS_on\"}, {\"from\": 107551, \"to\": 167060, \"label\": \"IMPACTS_on\"}, {\"from\": 107551, \"to\": 5, \"label\": \"LOCATED_IN\"}, {\"from\": 81504, \"to\": 106799, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 81504, \"to\": 45, \"label\": \"LOCATED_IN\"}, {\"from\": 132935, \"to\": 21002, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 132935, \"to\": 233, \"label\": \"LOCATED_IN\"}, {\"from\": 105461, \"to\": 168985, \"label\": \"IMPACTS_on\"}, {\"from\": 105461, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 26964, \"to\": 88682, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 26964, \"to\": 9, \"label\": \"LOCATED_IN\"}, {\"from\": 1411, \"to\": 86723, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 1411, \"to\": 45183, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 1411, \"to\": 59, \"label\": \"LOCATED_IN\"}, {\"from\": 134021, \"to\": 75881, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 134021, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 32935, \"to\": 164258, \"label\": \"IMPACTS_on\"}, {\"from\": 32935, \"to\": 637, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 32935, \"to\": 72, \"label\": \"LOCATED_IN\"}, {\"from\": 70139, \"to\": 107001, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 70139, \"to\": 5, \"label\": \"LOCATED_IN\"}, {\"from\": 38623, \"to\": 96862, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 38623, \"to\": 141, \"label\": \"LOCATED_IN\"}, {\"from\": 39353, \"to\": 168381, \"label\": \"IMPACTS_on\"}, {\"from\": 39353, \"to\": 171, \"label\": \"LOCATED_IN\"}, {\"from\": 136618, \"to\": 90707, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 136618, \"to\": 45, \"label\": \"LOCATED_IN\"}, {\"from\": 162041, \"to\": 68797, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 162041, \"to\": 10614, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 162041, \"to\": 3, \"label\": \"LOCATED_IN\"}, {\"from\": 54064, \"to\": 20488, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 54064, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 154088, \"to\": 168194, \"label\": \"IMPACTS_on\"}, {\"from\": 154088, \"to\": 72, \"label\": \"LOCATED_IN\"}, {\"from\": 48509, \"to\": 105811, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 48509, \"to\": 9, \"label\": \"LOCATED_IN\"}, {\"from\": 92195, \"to\": 168994, \"label\": \"IMPACTS_on\"}, {\"from\": 92195, \"to\": 161849, \"label\": \"IMPACTS_on\"}, {\"from\": 92195, \"to\": 3, \"label\": \"LOCATED_IN\"}, {\"from\": 151648, \"to\": 105103, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 151648, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 84863, \"to\": 169087, \"label\": \"IMPACTS_on\"}, {\"from\": 84863, \"to\": 171, \"label\": \"LOCATED_IN\"}, {\"from\": 59177, \"to\": 1120, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 59177, \"to\": 5, \"label\": \"LOCATED_IN\"}, {\"from\": 24937, \"to\": 10653, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 24937, \"to\": 137, \"label\": \"LOCATED_IN\"}, {\"from\": 118795, \"to\": 161849, \"label\": \"IMPACTS_on\"}, {\"from\": 118795, \"to\": 141, \"label\": \"LOCATED_IN\"}, {\"from\": 54214, \"to\": 10847, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 54214, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 98750, \"to\": 171981, \"label\": \"IMPACTS_on\"}, {\"from\": 98750, \"to\": 49225, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 98750, \"to\": 95, \"label\": \"LOCATED_IN\"}, {\"from\": 133199, \"to\": 31535, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 133199, \"to\": 413, \"label\": \"LOCATED_IN\"}, {\"from\": 153599, \"to\": 104970, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 153599, \"to\": 413, \"label\": \"LOCATED_IN\"}, {\"from\": 48433, \"to\": 166797, \"label\": \"IMPACTS_on\"}, {\"from\": 48433, \"to\": 9, \"label\": \"LOCATED_IN\"}, {\"from\": 37989, \"to\": 147249, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 37989, \"to\": 59, \"label\": \"LOCATED_IN\"}, {\"from\": 11743, \"to\": 66478, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 11743, \"to\": 1061, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 11743, \"to\": 20839, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 11743, \"to\": 5, \"label\": \"LOCATED_IN\"}, {\"from\": 83343, \"to\": 74640, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 83343, \"to\": 97, \"label\": \"LOCATED_IN\"}, {\"from\": 143032, \"to\": 45, \"label\": \"LOCATED_IN\"}, {\"from\": 146653, \"to\": 133564, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 146653, \"to\": 251, \"label\": \"LOCATED_IN\"}, {\"from\": 75082, \"to\": 74640, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 75082, \"to\": 33, \"label\": \"LOCATED_IN\"}, {\"from\": 26420, \"to\": 86870, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 26420, \"to\": 45, \"label\": \"LOCATED_IN\"}, {\"from\": 16533, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 104645, \"to\": 164103, \"label\": \"IMPACTS_on\"}, {\"from\": 104645, \"to\": 97, \"label\": \"LOCATED_IN\"}, {\"from\": 139383, \"to\": 168994, \"label\": \"IMPACTS_on\"}, {\"from\": 139383, \"to\": 45, \"label\": \"LOCATED_IN\"}, {\"from\": 70289, \"to\": 168928, \"label\": \"IMPACTS_on\"}, {\"from\": 70289, \"to\": 137, \"label\": \"LOCATED_IN\"}, {\"from\": 69300, \"to\": 162523, \"label\": \"IMPACTS_on\"}, {\"from\": 69300, \"to\": 33, \"label\": \"LOCATED_IN\"}, {\"from\": 150499, \"to\": 168985, \"label\": \"IMPACTS_on\"}, {\"from\": 150499, \"to\": 167, \"label\": \"LOCATED_IN\"}, {\"from\": 153431, \"to\": 24962, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 153431, \"to\": 51, \"label\": \"LOCATED_IN\"}, {\"from\": 85802, \"to\": 367, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 85802, \"to\": 742, \"label\": \"LOCATED_IN\"}, {\"from\": 117328, \"to\": 164118, \"label\": \"IMPACTS_on\"}, {\"from\": 117328, \"to\": 69, \"label\": \"LOCATED_IN\"}, {\"from\": 58718, \"to\": 14813, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 58718, \"to\": 42, \"label\": \"LOCATED_IN\"}, {\"from\": 158259, \"to\": 138268, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 158259, \"to\": 95, \"label\": \"LOCATED_IN\"}, {\"from\": 25918, \"to\": 31615, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 25918, \"to\": 137, \"label\": \"LOCATED_IN\"}, {\"from\": 111716, \"to\": 161891, \"label\": \"IMPACTS_on\"}, {\"from\": 111716, \"to\": 59, \"label\": \"LOCATED_IN\"}, {\"from\": 80707, \"to\": 162934, \"label\": \"IMPACTS_on\"}, {\"from\": 80707, \"to\": 3, \"label\": \"LOCATED_IN\"}, {\"from\": 62530, \"to\": 130, \"label\": \"LOCATED_IN\"}, {\"from\": 54128, \"to\": 117398, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 54128, \"to\": 141, \"label\": \"LOCATED_IN\"}, {\"from\": 97475, \"to\": 104970, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 97475, \"to\": 413, \"label\": \"LOCATED_IN\"}, {\"from\": 77962, \"to\": 161849, \"label\": \"IMPACTS_on\"}, {\"from\": 77962, \"to\": 9, \"label\": \"LOCATED_IN\"}, {\"from\": 149131, \"to\": 87474, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 149131, \"to\": 153207, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 149131, \"to\": 9, \"label\": \"LOCATED_IN\"}, {\"from\": 67349, \"to\": 59, \"label\": \"LOCATED_IN\"}, {\"from\": 120292, \"to\": 78833, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 120292, \"to\": 59, \"label\": \"LOCATED_IN\"}, {\"from\": 6748, \"to\": 137585, \"label\": \"ASSOCIATED_WITH\"}, {\"from\": 6748, \"to\": 137, \"label\": \"LOCATED_IN\"}];\n",
       "\n",
       "    var container = document.getElementById(\"cc00f517-2427-4bf0-b864-fb7d27a5bdd9\");\n",
       "\n",
       "    var data = {\n",
       "        nodes: nodes,\n",
       "        edges: edges\n",
       "    };\n",
       "\n",
       "    var options = {\n",
       "    nodes: {\n",
       "        shape: 'dot',\n",
       "        size: 25,\n",
       "        font: {\n",
       "            size: 14\n",
       "        }\n",
       "    },\n",
       "    edges: {\n",
       "        font: {\n",
       "            size: 14,\n",
       "            align: 'middle'\n",
       "        },\n",
       "        color: 'gray',\n",
       "        arrows: {\n",
       "            to: {\n",
       "                enabled: true,\n",
       "                scaleFactor: 0.5\n",
       "            }\n",
       "        },\n",
       "        smooth: {\n",
       "            enabled: false\n",
       "        }\n",
       "    },\n",
       "    physics: {\n",
       "        enabled: true\n",
       "        }\n",
       "    };\n",
       "\n",
       "    var network = new vis.Network(container, data, options);\n",
       "\n",
       "</script>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "options = {\"Variant\": \"variantOf\", \"Chromosome\": \"name\", \"Disease\": \"diseaseName\", \"Phenotype\": \"phenotypeName\"}\n",
    "neo4jupyter.draw(graph, options, physics = True, limit = 100)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "HprCsbZW9cEo"
   },
   "source": [
    "Now that we establish structure of our database as expected we can start investigating our data and get some meaningful instights.\n",
    "\n",
    "\n",
    "## Warming up with mitochondrial DNA and Chromosomes Y & X\n",
    "\n",
    "These two chromosomes are very particular. The chromosome Y determine the sex for male individuals. We said before that humans and mammals have *pairs* of chromosomes. But why do we have pairs in the first place. The thing is that humans receive one chromosome from the mom and one from the father. As such they have two versions of the same gene. One from the mom and one from the father. Having two versions of one gene is an advantage as if potent pathological mutation is transmitted from one of the parents the other copy can rescue and prevent the emergence of some diseases. But what about the Y chromosome? Man just have one copy of it and one copy of a X chromosome. This means that if pathological mutations occur in Y or X there will be little possibility for some rescue. This explains why a lot of diseases that affect young boys are X-linked. Let's have a quick look here and see genes with the most pathological variants."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "yXN4GYn39cEp"
   },
   "outputs": [],
   "source": [
    "query_26 = \"\"\"\n",
    "MATCH (c:Chromosome)<-[:LOCATED_IN]-(v:Variant) WHERE c.name = \"X\"\n",
    "WITH COUNT(v) AS x_variants_count, v.variantOf AS x_genes\n",
    "RETURN x_variants_count, x_genes\n",
    "ORDER BY x_variants_count DESC\n",
    "\"\"\"\n",
    "\n",
    "x_genes = graph.run(query_26).to_data_frame()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "to1ZswKK9cEr"
   },
   "source": [
    ">As said before *py2neo* provides an interface that allows to transform data into more readable format. The `to_data_frame()` transforms the data stream generated from Neo4J as a *pandas* dataframe. This is a very nice tool if we want to visualize more precisely our data and perform advanced exploratory data analysis from Neo4J databases."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "CJl2uyf49cEr",
    "outputId": "870a69d1-0daa-4881-9dfe-c6235aedf5d2"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>x_variants_count</th>\n",
       "      <th>x_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>635</td>\n",
       "      <td>COL4A5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>362</td>\n",
       "      <td>MECP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>312</td>\n",
       "      <td>DMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>265</td>\n",
       "      <td>F8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>215</td>\n",
       "      <td>CDKL5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>1</td>\n",
       "      <td>BRCC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>1</td>\n",
       "      <td>MAGEC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>1</td>\n",
       "      <td>SMARCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>1</td>\n",
       "      <td>SNORA11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>1</td>\n",
       "      <td>LOC107985698</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>344 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     x_variants_count       x_genes\n",
       "0                 635        COL4A5\n",
       "1                 362         MECP2\n",
       "2                 312           DMD\n",
       "3                 265            F8\n",
       "4                 215         CDKL5\n",
       "..                ...           ...\n",
       "339                 1         BRCC3\n",
       "340                 1        MAGEC3\n",
       "341                 1       SMARCA1\n",
       "342                 1       SNORA11\n",
       "343                 1  LOC107985698\n",
       "\n",
       "[344 rows x 2 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "NGdEa2iP9cEu"
   },
   "source": [
    "We could modify the previous query and focus on the top 5 of these genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "qyHNsHD-9cEu",
    "outputId": "7941aeb0-c322-42ba-b706-459ab32b2fdd"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>x_variants_count</th>\n",
       "      <th>x_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>635</td>\n",
       "      <td>COL4A5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>362</td>\n",
       "      <td>MECP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>312</td>\n",
       "      <td>DMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>265</td>\n",
       "      <td>F8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>215</td>\n",
       "      <td>CDKL5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   x_variants_count x_genes\n",
       "0               635  COL4A5\n",
       "1               362   MECP2\n",
       "2               312     DMD\n",
       "3               265      F8\n",
       "4               215   CDKL5"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_27 = \"\"\"\n",
    "MATCH (c:Chromosome)<-[:LOCATED_IN]-(v:Variant) WHERE c.name = \"X\"\n",
    "WITH COUNT(v) AS x_variants_count, v.variantOf AS x_genes\n",
    "RETURN x_variants_count, x_genes\n",
    "ORDER BY x_variants_count DESC LIMIT 5\n",
    "\"\"\"\n",
    "\n",
    "x_genes_top5 = graph.run(query_27).to_data_frame()\n",
    "x_genes_top5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "xqvis4Hw9cEy"
   },
   "source": [
    "We can do the same exercise for the Y and mitochondrial chromosomes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "wTSd37Wb9cEy",
    "outputId": "e0e6fb72-f6fb-4f07-a0ee-4309a3fae985"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>y_variants_count</th>\n",
       "      <th>y_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27</td>\n",
       "      <td>SRY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16</td>\n",
       "      <td>SHOX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>CRLF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>CSF2RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>SPRY3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   y_variants_count y_genes\n",
       "0                27     SRY\n",
       "1                16    SHOX\n",
       "2                 1   CRLF2\n",
       "3                 1  CSF2RA\n",
       "4                 1   SPRY3"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_28 = \"\"\"\n",
    "MATCH (c:Chromosome)<-[:LOCATED_IN]-(v:Variant) WHERE c.name = \"Y\"\n",
    "WITH COUNT(v) AS y_variants_count, v.variantOf AS y_genes\n",
    "RETURN y_variants_count, y_genes\n",
    "ORDER BY y_variants_count DESC LIMIT 5\n",
    "\"\"\"\n",
    "\n",
    "y_genes_top5 = graph.run(query_28).to_data_frame()\n",
    "y_genes_top5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "IOTPZIOj9cE1",
    "outputId": "8c5d272b-67b5-4f09-eb01-df965173e8e6"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mt_variants_count</th>\n",
       "      <th>mt_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>62</td>\n",
       "      <td>ND6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>33</td>\n",
       "      <td>ND4L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>25</td>\n",
       "      <td>ND2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>22</td>\n",
       "      <td>COX2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>19</td>\n",
       "      <td>ND5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   mt_variants_count mt_genes\n",
       "0                 62      ND6\n",
       "1                 33     ND4L\n",
       "2                 25      ND2\n",
       "3                 22     COX2\n",
       "4                 19      ND5"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_29 = \"\"\"\n",
    "MATCH (c:Chromosome)<-[:LOCATED_IN]-(v:Variant) WHERE c.name = \"MT\"\n",
    "WITH COUNT(v) AS mt_variants_count, v.variantOf AS mt_genes\n",
    "RETURN mt_variants_count, mt_genes\n",
    "ORDER BY mt_variants_count DESC LIMIT 5\n",
    "\"\"\"\n",
    "\n",
    "mt_genes_top5 = graph.run(query_29).to_data_frame()\n",
    "mt_genes_top5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From the dataframes we can start visualizing the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "0nDXa1vK9cE5",
    "outputId": "78358938-22ce-4247-a622-d50f1370b9e5"
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjAAAAFYCAYAAABNvsbFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjMsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+AADFEAAAgAElEQVR4nOzdeVhUZfsH8O+4seOgImYqpowCRikam6amxBIi7om9qaghkKm4Iaa51BuKuKAhCuZSWC5Ibim+EqSiApYWlUqooVIIL+goY6IB8/vDH/M6sg1yZoaB7+e6vK7mPIdz3yfOzNw85znPI5JKpXIQERER6ZBm2k6AiIiIqK5YwBAREZHOYQFDREREOocFDBEREekcFjBERESkc1jAEBERkc5hAUNEREQ6hwUMERER6RwWMHWUnZ3d6GLxnJqWxvg7aIyxeA1XrzH+DnhOdccChoiIiHQOCxgiIiLSOSxgiIiISOewgCEiIiKd00LbCVDDU1paigcPHqjl2Pr6+rh3755ajl3XOC1atICRkZHacyHtUNd13JCuYSMjI7RowY9xXVHTNdmQritNxqrPNcwrn5SUlpaiuLgYYrEYIpFI8OPr6elBX19f8OM+T5wHDx7g0aNH0NPTU3s+pFnqvI4byjUsl8shlUphYmLCIkYH1HZNNpTrSpOx6nsN8xYSKXnw4IHaipeGxtDQECUlJdpOg9SgKVzHIpEIYrFYbb2lJKymcE3WVX2vYRYwVElTeYM1lfNsqprC77cpnGNjwt9XZfX5f8IChoiIiHQOb5wCEG//sw57GwKpqu0v9Xvx+RIiqiN1XMO8fkmTeA1TXbEHhoiIiHQOe2BIJXX766j+6vqXU3l5Oby8vNC6dWvs3r1bsf3vv//GwIEDMXDgQKxdu1boNEnHNOTr2N/fH5cuXUJycjJatWql2H7y5EmMHj0a3377LRwdHdWRJmlJQ74eKwQGBuLrr7/Gu+++i40bNyq1ffTRR9iwYQPc3d1x6dIl3Lp1q9rj9O/fH99++22d49eEPTDUKDRr1gzR0dE4ffo0vvzyS8X2pUuXorS0FB9//LEWsyOqXXh4OO7evYtVq1Yptt2/fx/vv/8+Zs6cqbXi5fbt2wgICED37t1hYWEBR0dHpKamKtrlcjnCwsJgbW2NDh06wMvLC5cvX9ZKrqQenTp1wjfffKP0tFBpaSn27NmDTp06AQBSUlKQlZWFrKws7N+/HwBw7Ngxxba4uDjB82IBQ41G165d8fHHH2PRokW4desWTp48iW3btiE6OpoT1lGDJxaLERUVhcjISPz4448AgEWLFkEsFmPhwoVayUkqlcLd3R1yuRx79+5Feno6wsPDYW5urtgnMjISUVFRWLVqFZKTk2Fubo6RI0eiuLhYKzmT8Hr16oVu3brhm2++UWw7fvw49PT0MGDAAABAu3btYGFhAQsLC7Rp0wYA0LZtW8U2MzMzwfPiLSRqVKZMmYIjR47ggw8+QG5uLt5//304OztrOy0ilQwePBhTpkxBQEAAFixYgH379lW6paRJGzZsQIcOHbBlyxbFtq5duyr+Wy6XIzo6GrNnz4aPjw8AIDo6GhKJBPHx8fDz89N0yqQm7777Lnbt2oV//etfAIC4uDi88847yMnJ0VpO7IGhRmft2rVIT09Hq1at8OGHH2o7HaI6Wb58OQBg+vTp+PDDD9GrVy+t5fLtt9+ib9++8PPzg5WVFQYMGICYmBjI5XIAwI0bN5Cfn48hQ4YofsbAwAAuLi5IT0/XVtqkBmPHjsXFixdx7do15Ofn47vvvsOECRO0mhMLGGp04uLioK+vj7/++kurfx0QPQ8DAwN88MEH0NPTw4wZM7SaS05ODj7//HN07doV+/fvR0BAAJYvX47Y2FgAQH5+PgAo3VKqeF1QUKDxfEl9xGIxhg0bhri4OHz99dcYMGAAOnfurNWceAuJGpULFy5g/fr12LlzJ7788ksEBQXhP//5D5o3b67t1IhU1rx5c4hEIjRrpt2/McvLy9GnTx8sXboUAPDqq6/i+vXr2Lp1K/z9/RX7PTubqlwur3WG1ezs7Ge2GAqSc80xtHMM4MnChtped61i6ZS6LKFSVlaGsrIylJSUYNy4cZg5cyaMjIwwf/58lJSUKLVXePToUaWYNbl//36VBa9EIqnx57RawNy+fRvLli3DiRMnIJPJ0LVrV6xZs0YxKEgul2PlypXYuXMnpFIp+vbti4iICNjY2CiOIZVKsWDBAiQmJgIAPDw8EB4eDrFYrJVzIu0pKSlBQEAAJkyYgKFDh6Jfv35wcnJCZGQk5syZo+30iHSOhYUFevbsqbStR48eyM3NVbQDQEFBgeJpFAAoLCys1CvzrEpfTipOEFoXtX0B1iY7O7vex6hw7949jS2iWB19fX2UlJTUKY/mzZujefPm0NfXx5tvvolWrVrhzp07GDlyJPT09JTaKzxdqKkSy9TU9Ll6c7RW3gs1un3atGnIzMzEvn37EB8fj8zMTEyfPl0bp0Ratnz5cpSUlODf//43gCcfrhEREVi5ciUuXbqk5eyIdI+TkxOuXr2qtO3q1auKLxtLS0tYWFggJSVF0V5SUoJz585xzppGSCQS4cyZM/j555+13psEaLEHRojR7VlZWUhKSkJiYqLizbJu3Tp4enoKWjlTw3fmzBnExMTgwIEDMDExUXRbjh49GocPH0ZQUBCSkpKea8l2oqYqKCgIbm5uiIiIwKhRo5CZmYmYmBgsWbIEwJMvtMDAQKxZswYSiQRWVlaIiIiAkZERxowZo+XsSR1MTEy0nYKC1j7Nv/32WwwdOhR+fn44ffo0OnTogIkTJ+K9996DSCSqdXS7n58fMjIyYGxsrFTpOzk5wcjICOnp6SxgBCTUmiJ17b5UVf/+/VFUVFRl244dOwSPR7pJiOtYXdfw09555x2MHj1arTFUYW9vj127dmHFihVYvXo1OnXqhEWLFmHatGmKfWbNmoWHDx9i/vz5ilv9CQkJDeqLrqF6+nrUxHX1PKKjo+vc3qdPH0il0jqNtXkeWitgKka3BwUFYfbs2fjll18QEhIC4MmU2jWNbs/LywPw5L5r27ZtlQaLiUQitGvXrsYR8JoYPFZ1HO0coy5xNDHQTN0XdV3iPO/gMVWtXbsWhw8fxtWrV9GqVSv069cPS5cuha2trWKfiqm6n9avXz8kJSUJkgNRfbi7u8Pd3b3adpFIhNDQUISGhmowKyItFjBCjW6vaqR7bSPgNTF4rMo4daSp22BPx1H3QDNN/ZWhapznHTymqtTUVEydOhX29vaQy+X49NNPMWLECKSnpyvNTDl48GCl26namriMiEhXaK2AEWJ0e/v27VFYWKhUsMjlchQVFdU6Ap5IExISEpReb9myBV26dEFaWho8PT0V2/X09BTXPBER1U5rTyEJMbrdwcEBMpkMGRkZin0yMjLw4MEDjoCnBkkmk6G8vLzSY/7nzp2DlZUV+vbti5kzZ+K///2vljIkItINWuuBEWJ0e8+ePeHq6org4GBERkZCLpcjODgY7u7uHMBLDdLChQthZ2cHBwcHxTZXV1d4e3vD0tISN2/exCeffILhw4fj+++/bxCPKhIRNURaK2CEGt0eGxuLkJAQjBo1CgDg6emJ8PBwjZ9PY6LKLJqNQcV6LpqyaNEipKWlITExUWlm4KefNunVqxd69+4NOzs7HD9+HMOHD6/yWE1tFtO6xmrVqhVatWqltuu4oQxEl8vlkEqlah2ITsJpKp+tdVGfz2GtToohxOh2MzMzxMTEqCO9JsnIyAhSqRRisbjRv9H+/vtvjT22GBoaioSEBBw+fFhpvqOqvPDCC+jYsSOuX79e7T5NbRbTusYqLS1FcXGxWq7jhjIQvaJ4efHFFzm/kQ5oSp+tqqq4hp/3kXte9aSkRYsWMDExwf3799Vy/Pv378PU1FQtx65rnBYtWmjkFk1ISAgSEhJw5MgR9OjRo9b9i4qKkJeXx0G99aDO67ghXcMmJiYsXnREbddkQ7quNBmrPtcwr3yqpEWLFmjdurVajl1QUKCRFUw1Fac28+bNw549exAXFwexWKyY38jIyAjGxsaQyWRYuXIlhg8fDgsLC9y8eRMrVqyAubk5hg0bpuXsdZu6ruOmdg2TcGq6JhvjdaXuWCxgiNRo69atAKBYDqNCSEgIQkND0bx5c1y6dAm7d+/GvXv3YGFhgddffx3bt2/nTKZERDVgAUOkRlKptMZ2AwODSnPFEBFR7bQ2DwwRERHR82IBQ0RERDqHBQwRERHpHBYwREREpHNYwBAREZHOYQFDREREOocFDBEREekcFjBERESkc1jAEBERkc5hAUNEREQ6hwUMERER6RwWMERERKRzWMAQERGRzmEBQ0RERDqHBQwRERHpHBYwREREpHNYwBAREZHOYQFDREREOkflAsbc3Bz79++vtv3AgQMwNzcXJCkiImoYwsLCIBaLlf716NFD0S6XyxEWFgZra2t06NABXl5euHz5shYzpqZC5QKmtLQU5eXlz91ORES6SSKRICsrS/Hv7NmzirbIyEhERUVh1apVSE5Ohrm5OUaOHIni4mItZkxNQZ1uIYlEomrbLly4gNatW9c7ISIialhatGgBCwsLxb927doBeNL7Eh0djdmzZ8PHxwe2traIjo6GTCZDfHy8lrOmxq5FTY1btmxBTEyM4vXixYsRFhZWab979+7hzp07GDt2rPAZEhGRVuXk5MDGxgYtW7ZEv3798NFHH6Fr1664ceMG8vPzMWTIEMW+BgYGcHFxQXp6Ovz8/LSYNTV2NRYwrVu3RufOnQEA169fR5s2bdC+fXulfUQiEQwNDdGnTx8EBASoHDgsLAyrVq1S2ta+fXv8/vvvAJ5U9itXrsTOnTshlUrRt29fREREwMbGRrG/VCrFggULkJiYCADw8PBAeHg4xGKxynkQEVH1+vXrh02bNkEikaCwsBCrV6+Gm5sb0tLSkJ+fDwCVxj+am5sjLy+vxuNmZ2c/s8VQyLSriaGdYzS0WLpyThKJpMb2GguY8ePHY/z48QAAT09PhISEYPDgwc+dTFXJHTlyRPG6efPmiv+uuK8aFRUFiUSC8PBwjBw5EufPn4eJiQkAYNq0acjNzcW+ffsgEokwc+ZMTJ8+HXv27BEsRyKipuzNN99Uet2vXz/07t0bX331FV577TUAlYcXyOXyGoccAFV8OaX+Wf9ka4tRR9nZ2fU+RkOL1ZjOqcYC5mnHjh0TPvj/31d91rP3VQEgOjoaEokE8fHx8PPzQ1ZWFpKSkpCYmAhHR0cAwLp16+Dp6anRXxARUVNibGwMa2trXL9+HcOGDQMAFBQUoFOnTop9CgsL+VQqqZ3KBUyFa9euIScnB3fv3oVcLq/UXpdxMPW5r5qRkQFjY2NF8QIATk5OMDIyQnp6OgsYIiI1KCkpQXZ2Nl5//XVYWlrCwsICKSkpsLe3V7SfO3cOK1as0HKm1NipXMDcunULAQEBOHfuXJWFC/CkG1HVAqa+91ULCgrQtm1bpW5KkUiEdu3aoaCgoMbYmrj3WnUc7RyjIcXRZCx13nslaioWL14MDw8PdOrUSfFZ/ffff8PX1xcikQiBgYFYs2YNJBIJrKysEBERASMjI4wZM0bbqVMjp3IBM2fOHFy4cAHLly/HgAED6j1QVoj7qlXdY20o916rjFNHvCfa8OMQNXZ//fUXpk2bhqKiIrRr1w79+vXDiRMn0KVLFwDArFmz8PDhQ8yfP1/xwEVCQoJirCKRuqhcwKSmpmLGjBn44IMP1JJIXe+rtm/fHoWFhUoFi1wuR1FREe+9EhEJZNu2bTW2i0QihIaGIjQ0VEMZET2h8kR2pqamlR6hFlLFfVULCwul+6pPt587d04x5sXBwQEymQwZGRmKfTIyMvDgwQOlcTFERETU+KhcwPj6+uLgwYOCBV68eDFSU1ORk5ODH374AZMmTap0X3X9+vU4dOgQLl26hKCgIKX7qj179oSrqyuCg4Nx/vx5ZGRkIDg4GO7u7rx1QERE1MipfAvprbfewsmTJzFixAj4+fmhc+fOaNascv3Tu3dvlY4nxH3V2NhYhISEYNSoUQCezFUTHh6u6ikRERGRjlK5gHF3d1f896lTpyq1V4xFuXPnjkrHE+K+qpmZmdJSB0RERNQ0qFzAREZGqjMPokZp7dq1OHz4MK5evYpWrVqhX79+WLp0KWxtbRX7qLJsBhERKVO5gJk4caI68yBqlFJTUzF16lTY29tDLpfj008/xYgRI5Ceng4zMzMAqi2bQUREyuo8Ey8RqS4hIUHp9ZYtW9ClSxekpaXB09NTpWUziIioMpULmFmzZtW6j0gkwvr16+uVEFFjJpPJUF5erpgIUpVlM4iIqDKVC5gTJ05UmuG2rKwMBQUFkMvlaNOmDQwMDARPkKgxWbhwIezs7ODg4AAAKi2bURVNLIehS0thNNZYXA6DqHoqFzCXLl2qcntJSQliY2OxY8cOHDhwQLDEiBqbRYsWIS0tDYmJiWjevLlSW23LZjxLE8th6MpSGI01FpfDIKqZyhPZVUdfXx8ffPABXFxcsGDBAiFyImp0QkNDsX//fhw6dAhdu3ZVbLewsACASguQPr1sBhERVVbvAqaCvb09UlNThTocUaMREhKC+Ph4HDp0CD169FBqU2XZDCIiqkywp5AuXLiAFi34UBPR0+bNm4c9e/YgLi4OYrFYMebFyMgIxsbGimUz1qxZA4lEAisrK0RERCgtm0FERJWpXHHs27evyu337t3D6dOncejQIT4xQfSMrVu3AoDiEekKISEhilmmVVk2g4iIlKlcwPj7+1fbJhaLERwczDEwRM+QSqW17qPKshlERKRM5QLmwoULlbaJRCKIxWLFnBZEREREmqByAfPSSy+pMw8iIiIildV51O2NGzeQlJSEmzdvAgC6dOkCV1dXWFpaCp4cERERUVXqVMAsXboUUVFRKCsrU9rerFkzzJgxA8uXLxc0OSIiIqKqqDwPzObNm7Fhwwa4ubnh6NGjuH79Oq5du4ajR4/Cw8MDGzduxJYtW9SZKxERERGAOvTAbN++Ha6urvjqq6+Utjs7O8PZ2RljxozBtm3bMH36dMGTJCIiInqayj0wOTk58PDwqLbdw8MDOTk5QuREREREVCOVCxgzMzNcvXq12vZr167BzMxMkKSIiIiIaqJyAePh4YHPP/8ce/bsqdS2d+9ebNu2DZ6enoImR0RERFQVlcfALF26FBkZGQgMDMTSpUvRvXt3AMD169eRn58Pa2trfPTRR2pLlIiIiKiCygWMmZkZUlJS8Pnnn+M///mPYh6Ynj17YubMmfDz84O+vr7aEiUiIiKqUKd5YPT09BAUFISgoCB15UNERERUq1rHwBw8eBBJSUk17pOUlISDBw8KlhQRETU8a9asgVgsxvz58xXb5HI5wsLCYG1tjQ4dOsDLywuXL1/WYpbUVNRYwBw9ehR+fn4oLy+v8SDl5eXw8/OrtdAhIiLddP78eezcuRO9evVS2h4ZGYmoqCisWrUKycnJMDc3x8iRI1FcXKylTKmpqLGAiYuLg6OjI9zc3Go8iJubG1xcXLBjxw4hcyMiogbg3r17eO+997Bx40aIxWLFdrlcjujoaMyePRs+Pj6wtbVFdHQ0ZDIZ4uPjtZgxNQU1FjA//PBDjZPXPc3NzQ3nz59/7kSet2tSKpXC398fXbp0QZcuXeDv7w+pVPrceRARkbKKAmXQoEFK22/cuIH8/HwMGTJEsc3AwAAuLi5IT0/XdJrUxNQ4iPfu3bto166dSgdq27Yt7t69+1xJ1NY1GRUVBYlEgvDwcIwcORLnz5+HiYkJAGDatGnIzc3Fvn37IBKJMHPmTEyfPr3K+WqIiKhudu7cievXr1e51l1+fj4AwNzcXGm7ubk58vLyajxudnb2M1sM65WnajG0c4yGFktXzkkikdTYXmMBIxaLcfv2bZUC5efno3Xr1qpn9v+e7poMDw9XbH+2axIAoqOjIZFIEB8fDz8/P2RlZSEpKQmJiYlwdHQEAKxbtw6enp7Izs6u9eSJiKh62dnZWLFiBY4dO4ZWrVpVu59IJFJ6LZfLK217VqXP59Q/nztPlWPUkSa/RzQVqzGdU423kOzt7VV+uujgwYPo06dPnROoT9dkRkYGjI2NFcULADg5OcHIyIjdl0RE9ZSRkYGioiI4Ozujbdu2aNu2Lc6cOYOtW7eibdu2aNOmDQCgoKBA6ecKCwsr9coQCa3GAmbixIn45ZdfsHjxYsjl8ir3kcvlWLJkCX755RdMnjy5TsEruiY//PDDSm01dU1WvFkKCgrQtm1bpUpfJBKhXbt2ld5QRERUN15eXjh79ixOnz6t+NenTx+MHj0ap0+fhpWVFSwsLJCSkqL4mZKSEpw7d07pD0sidajxFpKXlxfGjx+PqKgoJCcnY9y4cbC1tYWxsTFkMhkuXbqEvXv34sqVKxg/fjzeeustlQML1TVZVTdlbd2Xmrj3WnUc7RyjIcXRZCx13nslagrEYrHSU0cAYGhoCDMzM9ja2gIAAgMDsWbNGkgkElhZWSEiIgJGRkYYM2aMNlKmJqTWmXg3bdoEiUSCyMhILF++XKkwkMvlMDU1xZIlSzB79uw6BX66a7JCWVkZzp49i23btiEtLQ3Ak16WTp06KfZ5umuyffv2KCwsVCpY5HI5ioqKauy+1MS91yrj1BHviTb8OERN3axZs/Dw4UPMnz8fUqkUffv2RUJCguJBCyJ1qbWAEYlEmDNnDgICAnD27FlkZWWhuLgYJiYm6NmzJ1xcXGBoWPceDC8vr0pjZt5//310794dc+bMUeqatLe3B/C/rskVK1YAABwcHCCTyZCRkaHorszIyMCDBw/YfUlEpAbffvut0muRSITQ0FCEhoZqKSNqqlReC8nQ0BCurq5wdXUVJLAQXZM9e/aEq6srgoODERkZCblcjuDgYLi7u/OvbyIiokasTos5apoqXZOxsbEICQnBqFGjAACenp5Kj2MTERFR49OgCpjn6Zo0MzNDTEyMulMjIiKiBqTW1aiJiIiIGhoWMERERKRzWMAQERGRzqn3GJgLFy5AKpXCycnpuR6nJiLdIt6u6rxJhirPsST1e/H5EyKiJknlHpiIiIhKMytOmDABrq6uGDNmDBwcHHDr1i3BEyQiIiJ6lsoFTEJCAnr06KF4fezYMRw7dgzvv/8+Nm/ejJKSEj6+TERERBqh8i2k3NxcpcnhDh8+jG7duuHjjz8GAFy/fh27du0SPkMiIiKiZ6jcAyOXy5VWpD558iSGDh2qeP3iiy/iv//9r7DZEem4M2fOYPz48bCxsYFYLK5U5AcGBipmpa74J9Rs10REjZnKPTBWVlY4evQopkyZgpSUFOTl5Sl90P7111+VlgagylQfAAmoOgiSAyAbrgcPHsDW1ha+vr4ICAiocp/Bgwdjy5Ytitc1rc5ORERPqFzAzJgxA9OmTUP37t1RXFwMKysrDBkyRNF+6tQp2NnZqSVJIl3l5uYGNzc3AEBQUFCV++jp6cHCwkKTaRER6TyVC5jRo0dDLBbj+PHjMDU1xbRp09CixZMfv3v3LkxMTODr66u2RIkaq3PnzsHKygqtW7dG//79sWTJEpibm2s7LSKiBq1O88AMHTpUadxLBTMzM+zevVuwpIiaCldXV3h7e8PS0hI3b97EJ598guHDh+P777+Hnp5etT+XnZ39zBbh52CqHEMbsTR7jIYWqz5xnn7ogqgxUrmAMTc3x+bNmzF69Ogq2w8cOID33nuPA3mJ6uDp91OvXr3Qu3dv2NnZ4fjx4xg+fHi1P1fpy0nFCePqotovQE3GUlF2drbGvrA1FUuT50Ski1R+Cqm0tBTl5eXP3U5EtXvhhRfQsWNHXL9+XdupEBE1aHVaC0kkElXbduHCBbRu3breCRE1ZUVFRcjLy+OgXiKiWtR4C2nLli2IiYlRvF68eDHCwsIq7Xfv3j3cuXMHY8eOFT5DIh0mk8kUvSnl5eXIzc1FZmYmzMzMYGZmhpUrV2L48OGwsLDAzZs3sWLFCpibm2PYsGFazpyIqGGrsYBp3bo1OnfuDODJTLtt2rRB+/btlfYRiUQwNDREnz59qp3ngqipunjxIry9vRWvw8LCEBYWBl9fX6xduxaXLl3C7t27ce/ePVhYWOD111/H9u3bYWJiosWsiYgavhoLmPHjx2P8+PEAAE9PT4SEhGDw4MGayIuoUXj99dchlUqrbU9ISNBgNkQEcEX1xkLlp5COHTumzjyIiIiIVFaneWAA4Nq1a8jJycHdu3eV1kaqwHEwREREpG4qFzC3bt1CQEAAzp07V2XhAjwZD8MChoiIiNRN5QJmzpw5uHDhApYvX44BAwZw4UYiIiLSGpULmNTUVMyYMQMffPCBOvMhIiIiqpXKE9mZmppWeoSaiIiISBtULmB8fX1x8OBBdeZCREQNTGxsLFxcXNC5c2d07twZb775Jo4fP65ol8vlCAsLg7W1NTp06AAvLy9cvnxZixlTU6HyLaS33noLJ0+exIgRI+Dn54fOnTujWbPK9U/v3r0FTZCIiLSnY8eOWL58Obp3747y8nJ8/fXXeOedd/D999/j5ZdfRmRkJKKiohAVFQWJRILw8HCMHDkS58+f54SMpFYqFzDu7u6K/z516lSldrlcDpFIhDt37giTGRERaZ2Xl5fS6yVLluDzzz/H+fPn0atXL0RHR2P27Nnw8fEBAERHR0MikSA+Ph5+fn7aSJmaCJULmMjISEEDx8bGYvv27bh16xYAwNraGvPmzVMUSnK5HCtXrsTOnTshlUrRt29fREREwMbGRnEMqVSKBQsWIDExEQDg4eGB8PBwPiH1/zjbJBEJqaysDAcOHMCDBw/g4OCAGzduID8/H0OGDFHsY2BgABcXF6Snp7OAIbVSuYCZOHGioIGF6JacNm0acnNzsW/fPohEIsycORPTp0/Hnj17BM2ViKgp++233+Dm5oaSkhIYGRkhLi4OvXr1Qnp6OgDA3NxcaX9zc3Pk5eVpI1VqQuo8E69Q6tstmZWVhaSkJCQmJsLR0YmcQk4AACAASURBVBEAsG7dOnh6eiI7OxsSiUTj50RE1BhJJBKcPn0a9+7dw6FDhxAYGIgjR44o2kUikdL+FUMKapKdnf3MFkOh0q0hhjZiafYYDSlOfWPV9j1epwLm8ePHOHr0KH766Sfcu3cP5eXlSu0ikQjr16+vc5LP0y2ZkZEBY2NjRfECAE5OTjAyMkJ6ejoLGCIigbRq1QrdunUDAPTp0wcXLlzApk2bMG/ePABAQUEBOnXqpNi/sLCwUq/Msyp9Rqt4G7suqv0e0GQsFWnqD29N/oGv7lgqFzB5eXnw8fFBdnY2TExMIJPJYGpqivv370Mul8PMzAyGhnWrauvTLVlQUIC2bdsqVfkikQjt2rVDQUFBjXE1UflXHUc9sfhXhvBxWAATVa+8vByPHz+GpaUlLCwskJKSAnt7ewBASUkJzp07hxUrVmg5S2rsVC5gli5ditu3b+PIkSOwtraGlZUVdu7ciddeew2fffYZdu3aVed5YurbLVlVF6UqXZeaqPyrjKOmWPwrQ3txiBq7ZcuWwc3NDS+++CJkMhni4+ORmpqKvXv3QiQSITAwEGvWrIFEIoGVlRUiIiJgZGSEMWPGaDt1auRULmCSk5MxZcoU9O/fX+lRaUNDQyxYsABXr17FokWL8NVXX6kcvD7dku3bt0dhYaFSwSKXy1FUVFRr1yUREakmPz8f/v7+KCgogKmpKXr16oX4+HgMHToUADBr1iw8fPgQ8+fPVzwxmpCQwDlgSO1ULmBkMpmi2NDT0wMAFBcXK9qdnZ2xbNmyeiVTl25JBwcHyGQyZGRkKMbBZGRk4MGDB0rjYoiI6PlFR0fX2C4SiRAaGorQ0FANZUT0hMoFTIcOHfDXX38BAIyMjGBmZobffvsNw4YNAwD8+eefaN68ucqB69st2bNnT7i6uiI4OBiRkZGQy+UIDg6Gu7s7bx0QERE1cioXMM7OzkhJScHChQsBAMOGDcOGDRvQqlUrlJeXY/PmzXB1dVU5sBDdkrGxsQgJCcGoUaMAAJ6enggPD1c5ByIiItJNKhcwQUFBSE5ORklJCfT19bF8+XLk5OQobuk4OTlh5cqVKgcWolvSzMwMMTExKsckIiKixkHlAsbOzg52dnaK12ZmZjh06BDu3LmDZs2acfp+IiIi0ph6z8Tbpk0bIfIgIiIiUlm1BUzFZHIVT/RUvK4NnwAiIiIidau2gPHw8IBIJMLt27fRqlUrxevqVMzH8vQcMURERETqUG0Bc+DAAQBPJpsDgG+++abWGW6JiIiINKHaAmbQoEFKrwcPHqzuXIiIiIhUotIg3ocPH8LS0hILFy7EnDlz1J0T6Rjx9rqsuWSo0hpNUr8Xnz8hIiJq9JqpspOBgQHEYjHXtiAiIqIGQaUCBgBGjBiBAwcOoLy8XJ35EBEREdVK5XlgRowYgdTUVHh5eWHSpEno2rUr9PX1K+3Xu3dvQRMkIiIiepbKBYyXl5fiv6uaE4aPURMREf0Pxweql8oFTGRkpDrzICIiIlKZygXMxIkT1ZkHERERkcpUHsRLRHV35swZjB8/HjY2NhCLxdi1a5dSu1wuR1hYGKytrdGhQwd4eXnh8uXLWsqWiEh31Gkxx8ePH+Po0aP46aefcO/evUpPJIlEIqxfv17QBIl02YMHD2BrawtfX18EBARUao+MjERUVBSioqIgkUgQHh6OkSNH4vz585y2gIioBioXMHl5efDx8UF2djZMTEwgk8lgamqK+/fvQy6Xw8zMDIaGhurMlUjnuLm5wc3NDQAQFBSk1CaXyxEdHY3Zs2fDx8cHABAdHQ2JRIL4+Hj4+flpPF8iIl2h8i2kpUuX4vbt2zhy5AguXrwIuVyOnTt3Ijc3F6GhoTAxMcGRI0fUmStRo3Ljxg3k5+djyJAhim0GBgZwcXFRefV3IqKmSuUemOTkZEyZMgX9+/dXelTa0NAQCxYswNWrV7Fo0SJ89dVXakmUqLHJz88HAJibmyttNzc3R15eXo0/m52d/cwW4Xs/K8fQRizNHqOhxapPHIlEImAmRA2PygWMTCZDt27dAAB6enoAgOLiYkW7s7Mzli1bJmx2RE3As6u8V8ypVJNKX04qzB9RV9V+AWoyloqys7M19oWtqViaPCciXaTyLaQOHTrgr7/+AgAYGRnBzMwMv/32m6L9zz//RPPmzYXPkKiRsrCwAAAUFBQobS8sLKzUK0NERMpULmCcnZ2RkpKieD1s2DBs2LAB69atw5o1a7B582YMHDhQLUkSNUaWlpawsLBQel+VlJTg3LlzcHR01GJmREQNn8q3kIKCgpCcnIySkhLo6+tj+fLlyMnJwYoVKwAATk5OWLlypdoSJdJFMpkM169fBwCUl5cjNzcXmZmZMDMzQ+fOnREYGIg1a9ZAIpHAysoKERERMDIywpgxY7ScORHpEtWXLVBtyQKg4S9boHIBY2dnBzs7O8VrMzMzHDp0CHfu3EGzZs0gFovVkiCRLrt48SK8vb0Vr8PCwhAWFgZfX19ER0dj1qxZePjwIebPnw+pVIq+ffsiISGBc8AQEdWixgJmyZIlePvtt/Hyyy9Xu0+bNm0ET4qosXj99dchlUqrbReJRAgNDUVoaKgGsyIi0n01joH57LPPMHDgQLi4uCAyMhK5ubmayouIiLRs7dq1eOONN9C5c2d0794db7/9Ni5duqS0D5fDIG2psQfmzJkz2LNnD/bv349ly5ZhxYoVcHFxwbhx4+Dj4wNTU1NN5UkEoGne5yXSltTUVEydOhX29vaQy+X49NNPMWLECKSnp8PMzAwAl8Mg7amxB8bW1hbLly/Hr7/+isOHD+Nf//oXfv31V8ycORM9e/bE5MmTcfToUZSWltY5sFCVvVQqhb+/P7p06YIuXbrA39+/xi57IiJSTUJCAv71r3/B1tYWvXr1wpYtW1BYWIi0tDQAlZfDsLW1RXR0NGQyGeLj47WcPTV2Kj9GPWDAAERGRuL333/Hl19+CTc3Nxw/fhzvvPMOevTogblz5youalVUVPbHjx/HoUOH0KJFC4wYMQJ3795V7FNR2a9atQrJyckwNzfHyJEjlSbQmzZtGjIzM7Fv3z7Ex8cjMzMT06dPVzkPIiJSjUwmQ3l5ueKhDS6HQdpUp9WoAaBly5YYNmwYhg0bhuLiYhw6dAi7du3Ctm3bsGPHDhQVFal0nISEBKXXW7ZsQZcuXZCWlgZPT0+VFrrLyspCUlISEhMTFfNmrFu3Dp6enpzFkohIYAsXLoSdnR0cHBwA1G85DKL6qnMB87Rff/0VP/74I65cuQIA9VqNuq6VvZ+fHzIyMmBsbKw06ZeTkxOMjIyQnp7OAoaISCCLFi1CWloaEhMTK826/jzLYTSN9bwa4zlp7hi1fYfXuYC5dOmS4nbNn3/+iWbNmuGNN97AuHHjMGzYsOdO9Hkq+4KCArRt21bpjSISidCuXbtK07MTEdHzCQ0NRUJCAg4fPoyuXbsqtj+9HEanTp0U21VZDqNJrOfVGM+pDtR9J0SlAiY3Nxfx8fHYt28fLl++DLlcjt69eyMoKAhjxoyp97ot9ansq6rya6v+NVH5Vx1HPbG0X5E3vnNi7x3REyEhIUhISMCRI0fQo0cPpbanl8Owt7cH8L/lMCpmaSdSlxoLmB07dmDv3r1IT09HeXk5unTpgjlz5uDtt98W7AO+PpV9+/btUVhYqFSwyOVyFBUV1VhUaaLyrzKOmmJpvSJvjOdERJg3bx727NmDuLg4iMViRc+4kZERjI2NIRKJuBwGaU2NBUxwcDDEYjHeffddvP3223B2dhY0eH0rewcHB8hkMmRkZCjGwWRkZODBgwdcDI+IqJ62bt0KAIoHKSqEhIQoZo/mchhNi+pzcQHqno+rxgLmiy++gIeHB1q2bFnnA9dGiMq+Z8+ecHV1RXBwMCIjIyGXyxEcHAx3d3f+VU1EVE+qzKnF5TBIW2osYJ5ehE5oQlX2sbGxCAkJwahRowAAnp6eCA8PV1veREREpH31eoy6PoSq7M3MzBATEyNkakRERNTAqTwTLxEREVFDwQKGiIiIdA4LGCIiItI5LGCIiIhI56hcwNy/fx/e3t74+eef1ZkPERERUa1ULmBKS0uRmpqqeHqIBQ0RERFpS42PUb/yyivo168f7O3t0b17dwD/W3vo2YKGiIiISFNqLGBmzJiBH3/8ETt27MC1a9cgEonw4Ycfws3NDXZ2dgCqXkyRiIiISJ1qLGD8/f0V/52Tk4M+ffrgpZdeQmpqKjZu3AiRSISFCxdiwIABcHR0hIODAzp37qz2pImIiKhpq3EMzM8//4yysjIAgKmpKQBg6tSpOH78OC5evAi5XA4bGxtcvXoVc+bMwauvvqr+jImIiKjJq7EHZvDgwTAwMMCrr74KGxsbiEQiyGQyAICBgQEAYOLEiRg0aBDkcjmuXLmi/oyJiIioyauxgLl06RJ++OEHXLhwAefPn4dcLse7776Lbt26wd7eHiKRCPfv3wfwZCyMjY2NRpImIiKipq3GAuaFF16At7c3vL29UVRUBCsrK/z73/9GaWkpTp48CblcjkmTJuGFF17Aa6+9htdeew3vv/++pnInIiKiJkrleWAqnjaytbXFBx98gC1btgAAwsPDMW/ePOjr62P79u3qyZKIiIjoKTX2wCjt2KIF+vfvD7FYDOB/BY1EIsGgQYPg5+enngyJiIiInqFyAWNqaoojR4787wefKWiIiIiINEXlAuZZzxY0RERERJrC1aiJiIhI57CAISIiIp3DAoaIiIh0DgsYIi0LCwuDWCxW+tejRw9tp0VE1KA99yBeIhKORCJRGhTfvHlzLWZDRNTwsYAhagBatGgBCwsLbadBRKQzeAuJqAHIycmBjY0NXnnlFUyZMgU5OTnaTomIqEFjDwyRlvXr1w+bNm2CRCJBYWEhVq9eDTc3N6SlpaFNmzZV/kx2dvYzWwwFz6tyDG3E0uwxGlqs+sSRSCQCZkLU8LCAIdKyN998U+l1v3790Lt3b3z11VeYMWNGlT9T6csp9U/B86r2C1CTsVSUnZ2tsS9sTcXS5DkR6SLeQiJqYIyNjWFtbY3r169rOxUinDlzBuPHj4eNjQ3EYjF27dql1C6XyxEWFgZra2t06NABXl5euHz5spaypaZEqwWMEG8MqVQKf39/dOnSBV26dIG/vz+kUqkmT4NIUCUlJcjOzuagXmoQHjx4AFtbW6xcuRIGBgaV2iMjIxEVFYVVq1YhOTkZ5ubmGDlyJIqLi7WQLTUlWi1ghHhjTJs2DZmZmdi3bx/i4+ORmZmJ6dOna/I0iOpl8eLFSE1NRU5ODn744QdMmjQJf//9N3x9fbWdGhHc3Nzw0UcfwcfHB82aKX9lyOVyREdHY/bs2fDx8YGtrS2io6Mhk8kQHx+vpYypqdBqAVPfN0ZWVhaSkpKwfv16ODo6wsHBAevWrcPx48c1OqCPqD7++usvTJs2Da+99hreffddtGrVCidOnECXLl20nRpRjW7cuIH8/HwMGTJEsc3AwAAuLi5IT0/XYmbUFDTYQby1vTH8/PyQkZEBY2NjODo6KvZxcnKCkZER0tPTOQCOdMK2bdu0nQLRc8nPzwcAmJubK203NzdHXl5ejT/bNJ6k4znVJ1Zt3+ENtoBR5Y1RUFCAtm3bQiQSKdpFIhHatWuHgoKCao+tiTdO1XHUE6sxXtDaPicWv0Sqe/ozGHjSg/7stmc1iSfpeE71i1WLBlvAVKjtjVHVm6S2N48m3jhVxlFTrMZ4QWv9nIioVhUDzQsKCtCpUyfF9sLCwkp/fBIJrcE+Rv30G+NpT78x2rdvj8LCQsjlckW7XC5HUVER3zxERGpmaWkJCwsLpKSkKLaVlJTg3LlzSrf2idShwRYwqrwxHBwcIJPJkJGRodgnIyMDDx484JuHiEgAMpkMmZmZyMzMRHl5OXJzc5GZmYlbt25BJBIhMDAQ69evx6FDh3Dp0iUEBQXByMgIY8aM0Xbq1Mhp9RaSTCZTTNb19BvDzMwMnTt3RmBgINasWQOJRAIrKytEREQovTF69uwJV1dXBAcHIzIyEnK5HMHBwXB3d+dtASIiAVy8eBHe3t6K12FhYQgLC4Ovry+io6Mxa9YsPHz4EPPnz4dUKkXfvn2RkJAAExMTLWZNTYFWCxgh3hixsbEICQnBqFGjAACenp4IDw/X+LkQETVGr7/+eo2Tg4pEIoSGhiI0NFSDWRFpuYAR4o1hZmaGmJgYdaRHREREDVSDHQNDREREVB0WMERERKRzWMAQERGRzmEBQ0RERDqHBQwRERHpHBYwREREpHNYwBAREZHOYQFDREREOocFDBEREekcFjBERESkc1jAEBERkc5hAUNEREQ6hwUMERER6RwWMERERKRzWMAQERGRzmEBQ0RERDqHBQwRERHpHBYwREREpHNYwBAREZHOYQFDREREOocFDBEREekcFjBERESkc1poOwEioqqIt/9Zh70NgVTV9pf6vaiRWFXFqVus+p0TUWPHHhgiIiLSOSxgiIiISOewgCEiIiKd02gKmK1bt+KVV16BhYUFBg0ahLNnz2o7JSKV8folXcdrmDStURQwCQkJWLhwIebOnYtTp07BwcEBY8eOxa1bt7SdGlGteP2SruM1TNrQKAqYqKgoTJgwAZMmTULPnj2xevVqWFhYYNu2bdpOjahWvH5J1/EaJm0QSaVSubaTqI/Hjx/jhRdewOeff44RI0Yots+bNw+XLl3C0aNHtZgdUc14/ZKu4zVM2qLzPTBFRUUoKyuDubm50nZzc3MUFBRoKSsi1fD6JV3Ha5i0RecLmAoikUjptVwur7SNqKHi9Uu6jtcwaZrOFzBt27ZF8+bNK1X6hYWFlf4iIGpoeP2SruM1TNqi8wVMq1at0Lt3b6SkpChtT0lJgaOjo5ayIlINr1/SdbyGSVuaL1y4cJm2k6gvExMThIWFoUOHDtDX18fq1atx9uxZfPbZZ2jdurW20yOqEa9f0nW8hkkbGsVijqNGjcKdO3ewevVq5Ofnw8bGBnv37kWXLl20nRpRrdR1/ZaWlqJFi0bxFqcGjp/BpA06/xg1UVUKCwvRunVrtGzZUtupaM3rr7+O2NhYWFtbqzWOVCqFWCyucZ+ffvoJvXv3VmsempSTk4O4uDgsXrxY26kQNVk6PwZGU6RSKTZt2oR58+Zh9erVyM3N1XZKBGDHjh149OgRgCdPPaxZswaWlpbo0aMHLC0tsWjRIpSXl2s5S+2wsbHBG2+8gY0bN6o1Tv/+/ZGcnFxlW2lpKT755BO4ubkJFi8tLQ1LlizBmjVrKr0PpVIpvL29BYv1tMePHyM+Ph4+Pj6wt7fH3r17BT/+09LS0nDy5MlK25u6Cxcu4ODBg7h48aJG416+fBlWVlYajVlf+/fvx6RJkzBr1iz89NNPSm1FRUV49dVXtZSZMFjAVMPa2hp37twB8OSvLUdHR6xbtw5ZWVn4/PPP4eLigt9//12weH/88QdiY2Oxb98+yGQypbb79+/j/fffFyzWw4cPsXLlSri4uODFF19Ep06d0L9/f6xevRoPHz4UJEZ5eTkiIiIwfvx4bN++HQAQFxeHvn37ok+fPvjoo48E+WCeM2cO7t+/D+BJMbN27VrMnTsXhw8fxpIlSxAXF4etW7fWO44uiomJQWxsLD777DO89dZbuHnzplri+Pr64u2338bcuXPx999/K7b/8ssvGDRoEL766ivExcUJEuvYsWPw8vLC2bNnsXfvXri4uOD48eOK9sePH+PMmTOCxKrw66+/IiQkBD179oS/vz9eeuklnDhxApmZmYIcPy8vD66urujQoQPc3d1x9+5djBkzBp6enhgxYgQcHR2Rl5cnSCxds2LFCiQlJQF40qv65ptvYujQoZg2bRqGDBkCd3d3FBYWaiSX0tJSxXdCfWmisIiLi8P06dPRrFkz/Pnnn3Bzc8OOHTsU7WVlZYIt9ZCcnIzS0lLF63379mHAgAHo2LEj+vTpg82bNwsS51m8hVQNMzMz/P777zA3N8fUqVORn5+PPXv2wMjICCUlJZg0aRL09fWxc+fOesdKS0vD6NGjYWxsjJKSEpiYmCAuLk7R5V5QUKBUUNVHaWkpPD098fPPP2Po0KGwtraGXC7HlStX8N1338He3h7ffvttvcdOfPLJJ9i6dSs8PT1x+vRp+Pr6YuvWrXj//ffRrFkzbNq0CZMnT653F/zTv6chQ4Zg9OjRSsXeF198gS1btgj+paZLioqKMG/ePHz33XcYO3YsmjdvrtQeHh5e7xg//vgjAgMD8c8//2DDhg04e/YsIiIiMHLkSISHh9d6i0lVrq6uePPNNxESEgIAiI2NxbJly7B582Z4e3sL9l4pLi5GfHw8vvjiC2RnZ8Pb2xvjxo3D2LFjkZqaKuhtOX9/f9y4cQOzZ8/Gvn378Oeff6JZs2b4/PPPUVZWhvfeew92dnZYvXq1YDF1RcVYGjs7OwQGBiI7OxtbtmxB9+7d8ccff2DGjBno1KkTtmzZovZcKgry+l5bcXFxmD17Nry9vVFcXIxTp04hPDwckydPBiDc533//v0xdepUTJkyBQCQmJgIf39/fPjhh5g+fbqg3ytt2rRBVlYWzM3NcfDgQUyZMgV+fn7o27cvMjMzsW3bNkRFRWHMmDH1jvU0jvBTwQ8//IANGzbAyMgIAKCvr4/58+dj0qRJghz/3//+N95++22sXbsWjx49wieffILhw4cjPj4eDg4OgsSosGPHDly/fh0nT56EjY2NUtulS5fg7e2NHTt2YNq0afWKs2/fPmzevBkeHh64cuUKXFxcsHnzZowbNw4AIJFI8NFHHwkyhqBisqwbN25g0KBBSm0DBw7EokWL6h1Dl4nFYvTo0QOHDx/GpUuXlAoYoSYa69u3L06dOgV/f3/4+PjA0NAQ27dvx7BhwwQ5foWsrCylHrX33nsP7du3x/Tp01FeXg5nZ2dB4vTq1Qv9+vVDQEAAvL29YWhoKMhxq3Lq1Cl8+eWXeO211+Dk5IRu3brhwIED6NixIwAgNDQUs2bNUlv8huzOnTswNTUFAJw5cwYxMTHo3r07AOCll17Cp59+irFjx2ozxTqLjo5GeHh4pcLi0aNHmD59umBx/vjjDwwdOlTx2sPDA3v37sW4ceNQVlYmaDEhl/+vHyQ6Ohpz585VfO76+vqiY8eO2LRpEwsYTar4cP/nn3+qnCZbqK7LzMxMrF+/HgCgp6eHjz/+GB07dsSYMWOwd+9edOvWTZA4AHDw4EHMnTu3UvECALa2tggODsbBgwfrXcDk5+ejV69eAJ7cjmvevDns7OwU7a+++iry8/PrFaNCYmIiTE1Noa+vjwcPHii1PXz4EM2aNd07pVeuXEFAQACKioqwf//+SgWekLZv344TJ06gf//+OH/+PA4cOIABAwYI1vsCPHl/3L17F127dlVs8/HxAQAEBgZi6dKlgsRp2bIl7t+/j/v37+PRo0dqLWCkUileeOEFAE96FA0NDdG5c2dFe7du3XD79m21xW/ILC0tcfnyZVhaWlb5PhaJRCgpKdFCZs9PU4WFqakp8vPzYWlpqdjm5OSEPXv2YNy4cYJ9/j7r2rVrWLlypdI2T09PtfQgNt1PdhV4eXnBxcUF9+7dQ3Z2tlJbbm4u2rZtK0icZs2aKY0dAJ58GIeGhmLs2LE4d+6cIHGAJ19oAwcOrLZ98ODBuHz5cr3jWFhYKI6TnZ2NsrIyZGVlKeXRrl27escBgA8++ACTJk3CX3/9hdOnTyu1nT9/XunLrinZuHEjBg8ejJ49e+LMmTNqK15u3bqFYcOGITw8HBs3bsThw4eRkpKCa9euwdnZGf/5z38Ei/XKK6/g1KlTlbb7+PggKipKsKeCrly5ghkzZiAxMRE2NjYYP348Dhw4oJap8du1a6f0ZfLee+/BzMxM8frevXuK3t+mZvLkyViyZAmuXr0Kf39/LFmyBH/88QeAJ2MTFy1apFQM1EenTp3QuXPnav+5u7sLEqeisHhaRWERFhaGqKgoQeLY29vjxIkTlbY7Oztj9+7dgo8N/O233/DTTz9BX18fZWVlSm3l5eVqeZiCPTDVqLjHXsHY2FjpdWJiomDd1dbW1khPT1fqoQCeFDEV98CFIpVKaywc2rVrh3v37tU7ztixYxEQEAAPDw+cPn0ac+bMweLFi1FQUIBmzZph/fr1GD58eL3j3L17t8b29u3bC/ZXua6JjIxETEyMIP+fa9K/f384Ozvj3Llz6NChA4AnYxeSkpIQHh6Od955B+PHjxfkaSg/P79qxzONHDkS5eXlikHj9dGyZUuMGDECI0aMwM2bN7Fr1y4sWbIEpaWlCAsLw4QJEzBkyBBBHtO3s7PD+fPn0bdvXwDAsmXLlNrT0tJga2tb7zi6KCgoCLm5uXB2dsZLL72Emzdvom/fvmjRogVKS0vx6quvYtu2bYLEEmIsmCoqCotnhwdUFBYVt9nrKygoCBkZGVW29e/fH7t378bXX38tSCzgyVxAFbeS0tLSFNcz8OQuQ6dOnQSLVYGDeBuAL774AqmpqYiJiamyfcOGDdi6dasgTz20adMGv//+e7VFjFADu8rLy7Fu3TpkZGTA2dkZs2fPxv79+7F06VL8/fff8PDwwOrVq9X+l+WdO3eQkJBQ71tiuqigoADt27evcZ8HDx7U+3fwxRdfYOLEidW2//TTTwgMDBS0J1Eb5HI5kpOT8eWXX+LYsWPQ09MT5Mmuig/96np3zp8/D319/Up/4DQl2dnZOHbsGHJyclBeXg4LCws4OTlh8ODBOrdgZGpqKjIyMjBnzpwq20+fPo2vv/4amzZt0nBmz+/Z94GxsTHatGmjeF1RKPn6+goalwXMczp9+jS+/PLLaouOhsrMzAxvvPEGWrVqVWX748eP+qDBMQAAIABJREFU8f3339e7gPn1119hY2NT6YkXTRPqyYHGpqSkBDExMdiwYQOuXr2q9niPHz+u9pqri82bN2Py5MnQ19cXIKvnV1RUhN27dwsyvcGBAwcwYsQIAbIidfjiiy/g4+Ojc0si1HZdlZSUYPny5QgLC9NgVsLiLaQ6yM/Px65duxAXF4cbN27AyclJsGP/8MMPOHbsGP755x8MHjwYQ4YMEezYT1OlAh4/fny94wwcOFDxWB0AjBs3Dhs2bFDcYiD1e/z4MVatWoXk5GS0bNkSM2fOxLBhw/D1119j+fLlEIlECAoKEixeXl4eDh06pCiIrKysMHz4cLzwwguCFC8AsGjRIowePVpRwHTu3BmnT5/W+DinR48eCVb4BQQE4OjRo1i9erXOfUlqyv379/H999/jxo0bAJ48gTRo0CCYmJioPfbcuXPh6Ogo+O/m2XPq2rUrBg8eLNg5BQQE4MiRI4iIiKg0kD4jIwNBQUEoKysTtIBR9zk9iz0wtSgvL0diYiK+/PJLJCUloaysDCEhIZg8eTIsLCwEiXH48GHFX5XNmzeHTCbDJ598IuiXi6Y9PT8L8GSAXGpqqsa/aJpyD8yKFSuwdetWDB48GOnp6SgqKsKkSZNw8uRJzJkzB2PHjhVsqYXt27cjNDQUjx49UnxYFRcXQ09PD2FhYfDz8xMkjiavqytXruD06dOK8TBisRh3797FqlWrsGPHDlhaWiI9PV2QOIGBgSgoKMCmTZvU+qSYLvrmm28QHBxcaWxe69atsX79esF6r55+8utpMpkMhoaGiqeghJj8TRPndPnyZQQEBKCgoAAbN26Eq6srHj9+jI8//hibN2/Gu+++i48//liw2/ia+j09jT0w1bh+/Tq++OILfP311zAwMMDbb7+NTz75BI6OjvDx8RGseAGAtWvXYsKECVi3bh1atGiB1atXIyIiQqcLGNK+b775BlFRUfD29sYvv/yCgQMHQiqVIi0tTdBFHpOSkjBv3jz4+/tj5syZikeC8/LysH79esyfPx+WlpZq61VUh8TEREycOBH//PMPgCfj0DZu3IjJkyejZ8+e2LFjBzw8PASJZW1tjaSkJKxZswbjxo3DpEmTMH/+/Eq/o6efTGoqfvvtN/j7+2PEiBGYMWOG0sSbGzduhL+/PyQSiWLKhvqQy+Xo37+/4rH8im0zZ87EnDlzFNd1fWnqnGxsbPDdd99h1apVmDBhAkaPHo2LFy9CJpNh7969eOONNwQ5H0Czv6ensQemGubm5hg/fjwmTJig9LRRu3btBJ+Js3PnzkhJSVGss/Ho0SN07NgRv//+u2CPaldYsGCBSvvVd0T+s4OF1fWX8meffVZj++3bt7Fp06Ym2QPTvn17XLx4ES+++CKAJ4+2nzhxAq+88oqgcYYPHw57e/tKT89UWLZsGS5cuIBDhw7VO9az15W6biG9+eabsLe3x5IlS7Bjxw4sWbIEVlZWWL9+Pfr37y9orKelpKRgzJgxShODyeVyiESiJnkNz5w5E//973+rfVrG19cX5ubm2LBhQ71jXbt2DVOnTsXLL7+MVatWKXomhP7M1+Q5AVA8yfrNN9/AyMgIycnJ6NGjhyDHrqDpc6rAHphqSCQSJCUlQSwWw8TEBC+//LLaYslkMqX7q3p6ejAwMEBxcbHgBUxsbCw6deqErl27Kn1IPk2IUf1yuRz+/v6KsQ8lJSWYNWsWDAwMlPbbvXt3veKoMohaHY/v6YJ//vkHenp6itctW7ZUzGoqpJ9++qnGSap8fX0Fe9RVLpejd+/eimtUJpNhwIABla7Z+nbzZ2VlITo6GsbGxggICMDSpUvx6aefqrV4OXz4MObOnQtnZ2fMnz9f6wPgG4Jz587VOEZj6tSpCA0NFSRW9+7dceLECXz00UcYNGgQYmJiYG9vL8ixn6bJc/r9998REBCAvLw8REVFYdeuXYonQEePHi1IDECz5/Q0FjDV+L/27j0upvz/A/hrNtesrXYpl4hyJ6xSPfppKkno7pLb9HWtLUturfs+kHRBdKGitURYohtREXJJvl+XkvJFLi2+2qXSTVKa3x89Ots0TWLOzDT1fj4e+3jszBnnc86OPfM+n/P+vN+pqam4c+cODh8+jEmTJkFdXZ1JbpXEsr3ExESBH5fq6mokJycLVABmo56Hra0tEhISoKGhAUdHR9ja2kpkRUf9ZGG2ahvUx1ZDvZZq8+bNTND48eNH+Pj4CAUx4s62VVVVNfp3qEOHDgKN3sTBVpGvzykpKWFuKtq0aYOOHTtKrBNxUVERfvnlF5w5cwbr169ntXGrvHv9+nWjswX9+/fH//73P9bGa9u2Lby9vWFiYoLZs2djwYIFrF/vpXVOe/bswZYtW2BpaYmoqCgoKytj5syZ2LNnDxYvXozTp09j586dAsudv5a0v6da9AipCcrLy3Hq1CkcOXIEaWlpGD16NKZNmwZra2tWVtU05dk2m1PIhYWFOH78OCIiIvDixQtMmTIFjo6O+PHHH1nZvzTduXPns3dJmzdvbpXF7CwtLT978eVwODh9+rRY43C5XMyfP59pRlffgQMH8PvvvwtVSZaEiooKPHv2TOzpfhUVFURHRzOrNywtLfHbb78J5UHUNlwVx5AhQ6CqqorQ0FBWH023BPWTtutjsyFhfXl5eXB2dsbVq1dx48YN1r4baZ2TlpYW/Pz8GkyerZ2ZefXqlUCF9K8lq++JApgv9PjxYxw6dAjHjx9Hfn4+8vPzZX1IYrl79y4iIiJw6tQp9OzZE+fPn5do3xe2aWlpISEhAf37929wu6enJ3bv3t1qe8k0pKqqCh8+fBCqLv219u7dC29vb4SEhGDixIkC286ePYtFixZh7dq1rDaqE4WtVWcqKirgcDgiH7MC7N1UeHp6Ys2aNawmVrcU9QPJ+goKCjB16lS5yg+S1jl9rpDlp0+fsHPnTvzyyy9ijQPI7nuiAOYrVVVV4dy5c7C2tpb1oYiltLQUJ0+exIEDB/DkyRM8ePBAKrUV2LJo0SJcuXIFSUlJTPfeWl5eXvD398ehQ4dYWzEiT1JSUlBQUAB7e3vmvV27dsHHxwdVVVUwMTHB/v37xW62yOfzsWDBAkRHR6Nfv34YMGAAOBwOHj58iCdPnsDGxgYHDhyQSsVUtgKYplbY7d27t1jjNEV5eTlCQ0OxfPlyiY/V3EgzkKwl6ZozsjgnSZPVOVEA84U8PT3h4uLCWiNCoKao0OrVqxEbGyuUn1BUVAR7e3ts375doLeEuK5du4bDhw/jzJkz0NbWxqxZszB58mTW7sqlpbq6GjweD0+ePEFCQgLzOG7btm3Yvn07Dhw4ACsrKxkfpWzY2trC3NwcixcvBgDcvn0b48aNg6OjIwYMGICgoCA4ODhgy5YtrIwXHR2NyMhIPHnyBEDN7Ni0adMEAihJk9e6P/n5+bh16xbatm0LY2NjKCgooLKyEmFhYdi1axc+ffqEp0+fyvowpU7agaQ0aplI85yqqqoQEhKCkydPIicnBxwOB5qamnBwcICLiwtrs36yCvgpgBGhoSaBfD4fAwcORHx8PPPIgo3aDHPmzMHIkSNF3mEFBATg7t27OHjwoNhj+fn54ciRIygrK8OMGTMwe/Zs1pfUSVtFRQUmT56MDx8+IC4uDqGhofD29sb+/fsFajq0NgMGDMCJEyeYPI1ff/0V//nPf5CQkACgptS4p6cnbt26JcvDZJW0Apjq6mq8evVKZPGzL3Hz5k04ODiguLgYHA4Ho0aNQnBwMHg8HiorK+Hq6goejydXj3bZ8rlmrbXYuA5nZWXBxMREZC2T2i7r4tYykdY5VVRUwN7eHmlpaTA2NsbAgQPB5/Px6NEjpKSkwNDQEFFRUaxUyZbm91QXBTAiiMrMrq3JwGZthhEjRuDYsWMiO85mZ2dj+vTpyMzMFHssFRUVqKurw9zcvNHoW1qdWdlSUlICKysrlJSU4MWLF9i7dy8mT54s68OSKTU1Ndy+fZtZRj5+/HiYm5szz7xzc3NhaGiIV69eyfIwv0h6enqj23NycuDs7Cz2/5cfPnzA2rVrERcXB2VlZSxcuBCurq7MdjaTEm1sbNClSxe4u7sjIiICISEh6NOnD1atWoUZM2bIXbNCNtU+mmgMh8NhJRdRWrVMpHVOvr6+CA8Px7FjxzBixAiBbenp6Zg9ezbmzJnT5NpgjZHm91QXZY2JoKamhuHDh+Pnn39mSkjz+XzY2dkhMDAQGhoarI2Vl5fX6N2VoqIi/v77b1bGMjQ0BIfDwaNHj0R+Rp4umHWLo/3rX//CunXrMGnSJLRp00ZgGxtL0OWNmpoanj17BnV1dVRUVODevXtYv349s720tJSVu6+mXLwAsPJjb2pq2qRn7eLatm0bEhMTsW7dOhQXF2PHjh24e/cuQkNDBa4HbLh//z7i4+MxePBgbNiwAaGhodi4cSM1eAQaXSGXnJyM0NBQ1h6DSKuWibTO6dSpU/D09BQKXoCa1XMeHh7w9fVlJYCR5vdUFwUwIly/fh2LFi3Czp07sXfvXqZ1AIfDgY6ODqvLHbt27YpHjx6JrCZatymiuOLj41nZT3MxZ84coffi4uIEghd5S4hji7m5OTZu3IiNGzfi3LlzUFRUFKgqnZWVBU1NTbHHqf9o08XFBR4eHo2ugPhaGRkZrO+zIVFRUQgICIC5uTkAYOrUqZg6dSoWLlyI3377DQB7gX5hYSGTU6eoqAhFRUXWqyXLqzFjxgi9l56ejk2bNiE1NRVz585l5QcYkF4tE2mdU25uLvT09ERu19fXb3LuyudI83uqiwIYEb7//nv88ccfCAkJgampKbZt2yaxZFATExPs2LED5ubmQhfF6upq+Pn5wcTEhLXxysvLERAQgLi4OOTm5oLD4UBDQ4N59lu/Wm5z1tRnr63RunXrwOPxYGdnh2+//RbBwcECMy4RERGs9EOpn2f0888/w9zcXCINFqWx6geo6Txf98esV69eOH36NGxsbDB//nxs3bqV1fHevXuHNm3aMI+mi4uLhf5ut8ZeSHU9f/4cnp6eiImJgbW1NW7evIm+ffuytv+ysrJGr30dO3ZEeXk5a+MBkj0nRUVFFBQUiKxEXlBQIJFrvaS/p7oogPkMV1dXGBoawsnJCUlJSRIZw93dHVwuF6ampnBzc2MShB89eoTAwEA8e/YMe/fuZWWsqqoq2NjYICMjA2ZmZrCwsGAS1bZt24YLFy4gPj6ealK0AD/88APOnTuHoqIifPvtt0Kl6cPDw1nrRNtcxMbGwsfHBzdu3BBrP7WP3+o+KlZVVUVsbCysra3h4uIi7qEK0NfXZ/6dz+cLNL5szb2QgJofWl9fXxw4cAD6+vo4f/68xIpuZmVlNVrLhC3SOCcDAwOEhYUhKCiowe1hYWECf+/EJc3vqRb9SjXBiBEjcPnyZaxatQo9evRgJW+grj59+iA2NhaLFi0SKF3N5/MxaNAgxMTEsBbBHjx4EE+fPkVKSgoGDx4ssC07OxvW1tY4ePAgFi5cyMp40nLt2jVmRgmo+W9qa2sLQ0NDGR+Z7NXts1WXvN7Rh4eH4+LFi2jbti1cXFygq6uLa9euYe3atcjJycH06dPFHsPIyAiRkZFCM59qamqIi4uDpaWl2GPUErcSckvm5+eHgIAA9O7dG0ePHsW4ceMkOt7kyZMlnl8lrXNauXIlJk2ahHfv3mHp0qXMjfHDhw8RGBiIpKQknD17lpWxpP091aJVSF+poKAAUVFRrP/Q37t3D0+fPgWfz4eWlhbrz8Ktra0xceJELFq0qMHtu3fvRmJiolxdVFevXo19+/ZBWVkZ/fr1A5/PR05ODoqLi+Hs7AwfHx9ZH2KrIqnO4wAQFBQEDw8PDB06lElEX716NQIDA+Hk5AQnJydWGqD++eefePz4MczMzBrcnpeXh4sXL2LWrFlij0VEU1FRQceOHWFkZNRo8CBuU1hAerVMpHlOZ8+ehZubm9DskYqKCgICAlhLi5DmOdVFAcxXknbBrOrqamb1gzj69++P6Ohokd2179+/Dzs7O+Tk5Ig9ljQkJCSAx+PBz88Pjo6OzH+j6upqhIeHY9WqVYiIiICFhYWMj7Tlqp+cd/DgQUyePJn1ppEAoKenhyVLlsDR0RFXr16FjY0NuFwuwsPDxa4oLCt8Ph9BQUGIj49HZWUlTExMsGrVKok0WZU3rq6uTZr1CA4OFnssadUykeY5ATU5jxcuXGAKIWppaWHs2LGs1hWS9jnVogDmK7EZwAwaNAipqalM7RknJyd4eXkxK4/YrDnRtWtXZGZmimxCmZeXB21tbbx580bssaSBx+NBQ0NDZFLl2rVr8eLFC0REREj5yFqPptzFsdE0EgC6d++Of//730wBOVVVVZw9exa6urpi77suLpfL1IABgJMnT2LixIkSyRnauXMnPD09YWxsjA4dOuDixYuYMWMGAgICWB+LiCarWiaScv78eaxYsQLXrl0TeoxcVFSEMWPGICAgQCDfSt5QDkwz8Ndff+HTp0/M63PnzmH9+vUCS6fZqjnx6dOnRhN0v/nmG4Fjae7S09Ph5uYmcru9vT3mz58vxSNqfc6cOdPg+2w3jQRqCszVnZlo164dq209amVmZqKyspJ5vXz5cujq6kokgDl69Ci2bdvGPI6+cOECZs+eDX9/f7mqySTvZFXLRFLCwsLg5ubWYA6ckpISli1bhtDQUApgiOSxdSHj8/lwdnYWmYj88eNHVsaRlrdv36Jnz54it/fs2RNv376V4hG1PtJqGlnr999/ZwKJqqoqHD58WGhav7b/E1vYuoFoyIsXLwQecZqZmYHP5+P169dCDUqJ5MiqlomkZGVlNbrcn8vlws/PT4pHxD4KYETYvXt3o9vz8vKkdCTsmjlz5mc/M2PGDCkcCTsqKioaXRXWtm1buQvK5M3OnTuZgm9ATdNIDw8PgaaRfn5+rDSNVFdXx5EjR5jXqqqqiIyMFPgMh8NhPYCRpI8fPwrU4+BwOGjXrh39vZUhadYykZS3b982mjfZEpbmUwAjwr59+z77GVEFgr4Uh8OR2lQx20lUzUHdO/L6SktLpXw0rc+DBw+wefNm5nVMTAz09fWZnjHq6urw9PRkJYBhox9YUyUmJjKJyNXV1UhOThaqiM1Wi4rNmzcLBDEfP36Ej4+PQCK0vPUnk0eyqGUiKT169MD9+/ehpaXV4PasrCx0795dykfFLgpgRLh3757UxuLz+bC0tGQKjZWXl2PatGlo27YtAMhVToq01b8jF/UZIjlFRUUCeSg3b94UmJH58ccf8fr1a1bGkmZi4pIlSwReu7u7C7xm6w7W0NAQz549E3hPT08PL1++FBiLSJasaplIyvjx4+Hl5YXx48cLVdx9//49s02e0SqkZsDb27tJF6g1a9ZI4WgI+TLDhw/Hnj17YGRkhIqKCmhoaOD48eMwNjYGUHOnZ2VlJfQj/TWmTZuG8ePHw8nJqcHt+/fvR2JiIk6cOCH2WLImiSRoIpqsaplIyps3b8DlcsHhcODs7CxQ4T0sLAx8Ph8pKSkS6VkmLTQD04iqqiqEhITg5MmTyMnJAYfDgaamJhwcHODi4sJaRjobHU5bq9awVLC5k1bTSKCmWrSXl5fI7fKYmCjtJGjSsBkzZrSoma6uXbsiMTERK1euhIeHB5OIzuFwYGZmhh07dsh18ALQDIxIFRUVsLe3R1paGoyNjTFw4EDw+Xw8evQIKSkpMDQ0RFRUFCttBZqSNMvhcHDs2DGxx2ppWtMdeXOVn58PHo+HtLQ0pmmktbU1s93GxgZ6enrYsGGD2GOpqakhNTVV5HP9nJwcjBkzRuwk+5cvX6KkpESg3caVK1ewfft2lJWVwdraGsuXLxdrjFq2trYwNzdnEo9v376NcePGCSRBOzg4sJJDRFqnd+/eCVR4bynBMM3AiODv74/nz5/j0qVLGDFihMC29PR0pk4DG8vqEhMT0atXrwaX8ZHGtcQ7cnkjzaaR0kpMXL9+Pfr27YtNmzYBqFnqPHPmTPTu3Rt9+vSBl5cXFBUV8dNPP4k9ljSToEnrpKysjFGjRsn6MFhHAYwIp06dgqenp1DwAgAjR46Eh4cHfH19WQlglixZghMnTiA1NRWzZ8/GrFmzGq1tQv7RGpYKygtpNI2UVmLi7du3BYKTyMhIqKmp4erVq2jTpg2CgoJw5MgRVgIYaSZBE9KSiN9cp4XKzc2Fnp6eyO36+vpNbv71OR4eHsjKyoKXlxfu3r0LHR0dTJ06FbGxsQLVQImw2jtyUVrCUkHyD3d3dxQXF0NHRwf+/v6Ij49HfHw8du3aBV1dXRQXF2PlypVij/P27VumXQFQ0+3c0tKSyXubOHEi0/lcXGpqakyCc0VFBe7duydw7SktLWXlUTUhLQ0FMCIoKio2eudeUFAgdAcoDgUFBUyaNAlHjx5FRkYGjIyM4OnpicGDB1Mtk0bU3pGXl5cLbWspSwXJP2oTE4cOHQoPDw/weDzweDxs2bIFQ4cORUJCAiuJiUpKSgL//9+5c0eg3xKHw2GtvEFtEnRKSgo2btwo0SRoQloSeoQkgoGBAcLCwhAUFNTg9rCwMOjr60tk7Pfv36OoqAhlZWXo1KlTi8qMZ5u7uzvi4uKgo6MjcqkgG3fkpPno3bs3IiMjJZqYqKOjg+DgYOzZswcxMTF4//49sywcqEkWZusx77p168Dj8WBnZ8ckQdedcYmIiICpqSkrYxHSktAqJBFu3bqFSZMmwcLCAkuXLmV+GB8+fIjAwEAkJSWx2gW3vLwc0dHROHz4MNLT02FlZQUejydw0SQN+/PPP7Fy5UokJyc3uFRQQ0NDxkdI5E1mZibs7OxQVFSE6upqrFixQmAVlbOzMzp37sxqgrioJOjCwkJ06tSJHiMRUg8FMI04e/Ys3NzchB4lqaioICAgAFZWVqyMs3TpUkRHR0NTUxOOjo6YMmVKi1nmJk0tdakgkY38/HykpaVBTU1N6EYlKSkJAwcOpOCYEBmiAOYzysvLceHCBTx9+hQAoKWlhbFjx0JRUZG1MVRUVKCuro4hQ4a0iAqQhMg7KpBISPNHOTAi1L2A1S3KBdRcwLS1tVm7gLW0CpCEyLt9+/bBzc2twaXhSkpKWLZsGUJDQymAIUSGaAZGBAcHB5ibm1OFV0JaoaFDhyImJobJfavv8ePHsLW1RXZ2tpSPjBBSi5ZRi5CVlQUTExOR27lcbqP1Rwgh8osKJBLS/FEAIwJdwAhpvahAIiHNHwUwItAFjJDWiwokEtL8UQ6MCKtXr8bly5dx+fLlBnuumJqawsTEBL6+vjI6QkKIpLx58wZcLhccDkdkgcSUlBRWqv4SQr4OBTAi0AWMkNaNCiQS0rxRANMIuoARQqhAIiHNEwUwTUAXMEIIIaR5oQCGEEIIIXKHViERQgghRO5QAEMIIYQQuUMBDCGtVGFhIbZu3QojIyP06tULqqqqGDZsGObMmYPTp08zievNlbe3N5SVlaGlpYWysjKh7aNHj4alpaUMjowQIg0UwBDSCmVkZMDAwAABAQEYPHgwfv31V/j5+YHH4+Hly5dwdHTE/v37ZX2YTZKfn4+wsDBZHwYhRMqoGzUhrUxRURFmzZoFPp+Py5cvY8iQIQLb16xZgytXrqCoqEhGR/hltLW1ERQUBCcnJ3Tq1EnWh0MIkRKagSGklTl48CBevXqFrVu3CgUvtbhcLqytrQXeKy4uxoYNG6Ctrc08btq0aRMqKioEPqesrIzly5fj/PnzMDIygpqaGkaNGoWTJ08KjdPUfTZm9erVTZ6FCQoKgoWFBTQ1NaGmpgZDQ0McOnRI6HPa2tqYMmUKbty4ATMzM3Tr1g0GBga4dOkSAODChQvgcrnMPm7evCm0j7y8PCxduhSDBg2CqqoqRo0ahYCAAKFHc9HR0TA1NUWvXr3Qu3dvGBoaUoVvQpqAllET0spMmDABGRkZyM3NRbt27Zr0Z8rLy2FhYYHc3FzMnTsXffv2RWZmJsLDwzF+/HgcPXqU+ayysjKGDRuGv//+G/PmzcMPP/yAQ4cOITs7Gzdu3MDAgQO/eJ8N8fb2hq+vLx4+fAhXV1dkZGTg3r17zCzM6NGjoaqqivj4eObPDB48GOPGjcOQIUPA4XBw5swZXLt2Df7+/pg7dy7zOW1tbbRv3x6lpaVwdHTE999/j+DgYLx58wbBwcHYsGED5s+fj/bt2yMgIADffPMNMjMz0b59ewA1lbxNTU1RVVWFOXPmoFu3brhx4wZOnDgBFxcX+Pj4AAAuX74Me3t7cLlcWFlZQUFBAY8fP0Z2djZiY2Ob9N0Q0lrRIyRCWpn//ve/0NLSEgpeysrK8OHDB+Z1mzZtoKSkBAAIDg7G48ePcfnyZSYAAWoCAnd3d6SmpsLQ0FBgjOvXrzOftbOzw7BhwxAREYEtW7Z81T4bs3btWpibmyMsLAzLli0T+bnbt29DUVGRee3i4gI7OzsEBgYKBDAAkJOTg/j4ePzf//0fAGD48OGwtLTETz/9hNTUVKa9iLq6OubNm4eEhATY2toCADw9PVFRUYHr168z7UbmzZuHbt26Yffu3XB1dYWGhgYSExPRuXNnREVFQUFBoUnnSgipQY+QCGllSkpK0LlzZ6H3N23aBC0tLeafWbNmMduio6Ohr6+PLl26ID8/n/nHxMQEAHDlyhWBfRkZGQkEJaqqqujfvz+eP3/+1ftszOjRozF27FgEBQU1uCKpVm3wUllZicLCQuTn54PL5eLp06dCOT/9+vVjghcA0NXVBQDo6ekxwQsA6OjoAABzbnw+H7GxsbCwsICCgoLAuZmZmaG6uhrXr18HAHTu3BllZWW4ePFik8+VEFKDZmAIaWU6d+6MkpISofednZ2ZZceLFy8W2PbkyRPcv38fWlpaDe4l3QIrAAAD0UlEQVTz7du3Aq979eol9BllZWUUFhZ+9T4/pymzMPHx8di+fTsyMzPx6dMngW3FxcXMjBNQM7NSV/v27dG+fXv07NlT4P3vvvsOQE3LkdrjfvfuHSIiIhAREdHouS1YsAAxMTGYNm0aunfvDmNjY1hbW2PSpEngcDhfcPaEtD4UwBDSygwaNAjp6en4+PGjwGOk/v37MzMLHTt2FPgz1dXV4HK5WLFiRYP77NGjh8BrUY9D6iawfuk+P6fuLIyTk5PQ9rS0NPB4PBgYGGDXrl3o1q0b2rVrh6SkJAQHB6O6urpJ5/C5c6vdz9SpU8Hj8Rr8rKamJgBATU0N165dw6VLl3DhwgUkJyfjjz/+gLm5OU6cOEFBDCGNoACGkFZm4sSJSEtLQ0xMDBwcHJr0Z/r27YvS0lLm8Q4bJLHPurMw9cXExKBDhw6Ijo5Ghw4dmPevXr3K2vgA0KVLF3z33Xeoqqpq0rm1a9cOFhYWsLCwAJ/Px+bNm+Hv74+bN2/CwMCA1WMjpCWhHBhCWpm5c+eiZ8+eWL9+PR48eNDgZ+ov9Z08eTLu3LmDs2fPCn22vLwcpaWlX3wckthn3VmY9+/fC2xTUFAAh8MRmGmpfdTDJgUFBdjY2ODMmTNIT08X2l5UVITKykoAQEFBgcA2DoeD4cOHM8dGCBGNZmAIaWWUlJRw5MgRTJ8+HcbGxrCzs4Ouri46duyIvLw8JCYmIicnh0laBYAlS5YgKSkJjo6OcHBwgI6ODioqKpCTk4Po6GhERkZi9OjRX3Qcktgn8M8sDAD06dOHeX/ChAnYs2cP7O3tMX36dBQWFiI8PByqqqr466+/vnicxmzatAnXr1/HhAkT4OjoiCFDhqCkpATZ2dk4ffo07ty5AzU1NSxZsgQFBQXgcrno2bMnXr9+jbCwMHTr1k0ggZgQIowCGEJaoZEjRyI1NRUhISE4d+4c4uPjUVlZCVVVVejq6mLFihWYOHEi8/mOHTsiLi4OAQEBiIqKwqlTp9CpUyf06dMHrq6uAqtymkoS+wRqZmHMzMyQnJws8L6RkRFCQkKwa9curF27Fj169ICzszOUlZWFkpbF1aVLFyQnJ2P79u2Ij4/HwYMHoaSkhH79+mHNmjVQUVEBADg4OODQoUM4cOAA3r17B1VVVZibm2P16tUNrhQjhPyDCtkRQgghRO5QDgwhhBBC5A4FMIQQQgiROxTAEEIIIUTuUABDCCGEELlDAQwhhBBC5A4FMIQQQgiROxTAEEIIIUTuUABDCCGEELlDAQwhhBBC5A4FMIQQQgiRO/8P2r5oF3J25FUAAAAASUVORK5CYII=\n",
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Created with matplotlib (https://matplotlib.org/) -->\n",
       "<svg height=\"342.463625pt\" version=\"1.1\" viewBox=\"0 0 558.76675 342.463625\" width=\"558.76675pt\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       " <defs>\n",
       "  <style type=\"text/css\">\n",
       "*{stroke-linecap:butt;stroke-linejoin:round;}\n",
       "  </style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 342.463625 \n",
       "L 558.76675 342.463625 \n",
       "L 558.76675 0 \n",
       "L 0 0 \n",
       "z\n",
       "\" style=\"fill:#f0f0f0;\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 57.68175 256.585 \n",
       "L 198.063417 256.585 \n",
       "L 198.063417 7.2 \n",
       "L 57.68175 7.2 \n",
       "z\n",
       "\" style=\"fill:#f0f0f0;\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\">\n",
       "    <g id=\"xtick_1\">\n",
       "     <g id=\"line2d_1\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 74.69771 256.585 \n",
       "L 74.69771 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_2\"/>\n",
       "     <g id=\"text_1\">\n",
       "      <!-- COL4A5 -->\n",
       "      <defs>\n",
       "       <path d=\"M 64.40625 67.28125 \n",
       "L 64.40625 56.890625 \n",
       "Q 59.421875 61.53125 53.78125 63.8125 \n",
       "Q 48.140625 66.109375 41.796875 66.109375 \n",
       "Q 29.296875 66.109375 22.65625 58.46875 \n",
       "Q 16.015625 50.828125 16.015625 36.375 \n",
       "Q 16.015625 21.96875 22.65625 14.328125 \n",
       "Q 29.296875 6.6875 41.796875 6.6875 \n",
       "Q 48.140625 6.6875 53.78125 8.984375 \n",
       "Q 59.421875 11.28125 64.40625 15.921875 \n",
       "L 64.40625 5.609375 \n",
       "Q 59.234375 2.09375 53.4375 0.328125 \n",
       "Q 47.65625 -1.421875 41.21875 -1.421875 \n",
       "Q 24.65625 -1.421875 15.125 8.703125 \n",
       "Q 5.609375 18.84375 5.609375 36.375 \n",
       "Q 5.609375 53.953125 15.125 64.078125 \n",
       "Q 24.65625 74.21875 41.21875 74.21875 \n",
       "Q 47.75 74.21875 53.53125 72.484375 \n",
       "Q 59.328125 70.75 64.40625 67.28125 \n",
       "z\n",
       "\" id=\"DejaVuSans-67\"/>\n",
       "       <path d=\"M 39.40625 66.21875 \n",
       "Q 28.65625 66.21875 22.328125 58.203125 \n",
       "Q 16.015625 50.203125 16.015625 36.375 \n",
       "Q 16.015625 22.609375 22.328125 14.59375 \n",
       "Q 28.65625 6.59375 39.40625 6.59375 \n",
       "Q 50.140625 6.59375 56.421875 14.59375 \n",
       "Q 62.703125 22.609375 62.703125 36.375 \n",
       "Q 62.703125 50.203125 56.421875 58.203125 \n",
       "Q 50.140625 66.21875 39.40625 66.21875 \n",
       "z\n",
       "M 39.40625 74.21875 \n",
       "Q 54.734375 74.21875 63.90625 63.9375 \n",
       "Q 73.09375 53.65625 73.09375 36.375 \n",
       "Q 73.09375 19.140625 63.90625 8.859375 \n",
       "Q 54.734375 -1.421875 39.40625 -1.421875 \n",
       "Q 24.03125 -1.421875 14.8125 8.828125 \n",
       "Q 5.609375 19.09375 5.609375 36.375 \n",
       "Q 5.609375 53.65625 14.8125 63.9375 \n",
       "Q 24.03125 74.21875 39.40625 74.21875 \n",
       "z\n",
       "\" id=\"DejaVuSans-79\"/>\n",
       "       <path d=\"M 9.8125 72.90625 \n",
       "L 19.671875 72.90625 \n",
       "L 19.671875 8.296875 \n",
       "L 55.171875 8.296875 \n",
       "L 55.171875 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-76\"/>\n",
       "       <path d=\"M 37.796875 64.3125 \n",
       "L 12.890625 25.390625 \n",
       "L 37.796875 25.390625 \n",
       "z\n",
       "M 35.203125 72.90625 \n",
       "L 47.609375 72.90625 \n",
       "L 47.609375 25.390625 \n",
       "L 58.015625 25.390625 \n",
       "L 58.015625 17.1875 \n",
       "L 47.609375 17.1875 \n",
       "L 47.609375 0 \n",
       "L 37.796875 0 \n",
       "L 37.796875 17.1875 \n",
       "L 4.890625 17.1875 \n",
       "L 4.890625 26.703125 \n",
       "z\n",
       "\" id=\"DejaVuSans-52\"/>\n",
       "       <path d=\"M 34.1875 63.1875 \n",
       "L 20.796875 26.90625 \n",
       "L 47.609375 26.90625 \n",
       "z\n",
       "M 28.609375 72.90625 \n",
       "L 39.796875 72.90625 \n",
       "L 67.578125 0 \n",
       "L 57.328125 0 \n",
       "L 50.6875 18.703125 \n",
       "L 17.828125 18.703125 \n",
       "L 11.1875 0 \n",
       "L 0.78125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-65\"/>\n",
       "       <path d=\"M 10.796875 72.90625 \n",
       "L 49.515625 72.90625 \n",
       "L 49.515625 64.59375 \n",
       "L 19.828125 64.59375 \n",
       "L 19.828125 46.734375 \n",
       "Q 21.96875 47.46875 24.109375 47.828125 \n",
       "Q 26.265625 48.1875 28.421875 48.1875 \n",
       "Q 40.625 48.1875 47.75 41.5 \n",
       "Q 54.890625 34.8125 54.890625 23.390625 \n",
       "Q 54.890625 11.625 47.5625 5.09375 \n",
       "Q 40.234375 -1.421875 26.90625 -1.421875 \n",
       "Q 22.3125 -1.421875 17.546875 -0.640625 \n",
       "Q 12.796875 0.140625 7.71875 1.703125 \n",
       "L 7.71875 11.625 \n",
       "Q 12.109375 9.234375 16.796875 8.0625 \n",
       "Q 21.484375 6.890625 26.703125 6.890625 \n",
       "Q 35.15625 6.890625 40.078125 11.328125 \n",
       "Q 45.015625 15.765625 45.015625 23.390625 \n",
       "Q 45.015625 31 40.078125 35.4375 \n",
       "Q 35.15625 39.890625 26.703125 39.890625 \n",
       "Q 22.75 39.890625 18.8125 39.015625 \n",
       "Q 14.890625 38.140625 10.796875 36.28125 \n",
       "z\n",
       "\" id=\"DejaVuSans-53\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(78.560835 316.074062)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"69.824219\" xlink:href=\"#DejaVuSans-79\"/>\n",
       "       <use x=\"148.535156\" xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"204.248047\" xlink:href=\"#DejaVuSans-52\"/>\n",
       "       <use x=\"267.871094\" xlink:href=\"#DejaVuSans-65\"/>\n",
       "       <use x=\"336.279297\" xlink:href=\"#DejaVuSans-53\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_2\">\n",
       "     <g id=\"line2d_3\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 101.285146 256.585 \n",
       "L 101.285146 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_4\"/>\n",
       "     <g id=\"text_2\">\n",
       "      <!-- MECP2 -->\n",
       "      <defs>\n",
       "       <path d=\"M 9.8125 72.90625 \n",
       "L 24.515625 72.90625 \n",
       "L 43.109375 23.296875 \n",
       "L 61.8125 72.90625 \n",
       "L 76.515625 72.90625 \n",
       "L 76.515625 0 \n",
       "L 66.890625 0 \n",
       "L 66.890625 64.015625 \n",
       "L 48.09375 14.015625 \n",
       "L 38.1875 14.015625 \n",
       "L 19.390625 64.015625 \n",
       "L 19.390625 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-77\"/>\n",
       "       <path d=\"M 9.8125 72.90625 \n",
       "L 55.90625 72.90625 \n",
       "L 55.90625 64.59375 \n",
       "L 19.671875 64.59375 \n",
       "L 19.671875 43.015625 \n",
       "L 54.390625 43.015625 \n",
       "L 54.390625 34.71875 \n",
       "L 19.671875 34.71875 \n",
       "L 19.671875 8.296875 \n",
       "L 56.78125 8.296875 \n",
       "L 56.78125 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-69\"/>\n",
       "       <path d=\"M 19.671875 64.796875 \n",
       "L 19.671875 37.40625 \n",
       "L 32.078125 37.40625 \n",
       "Q 38.96875 37.40625 42.71875 40.96875 \n",
       "Q 46.484375 44.53125 46.484375 51.125 \n",
       "Q 46.484375 57.671875 42.71875 61.234375 \n",
       "Q 38.96875 64.796875 32.078125 64.796875 \n",
       "z\n",
       "M 9.8125 72.90625 \n",
       "L 32.078125 72.90625 \n",
       "Q 44.34375 72.90625 50.609375 67.359375 \n",
       "Q 56.890625 61.8125 56.890625 51.125 \n",
       "Q 56.890625 40.328125 50.609375 34.8125 \n",
       "Q 44.34375 29.296875 32.078125 29.296875 \n",
       "L 19.671875 29.296875 \n",
       "L 19.671875 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-80\"/>\n",
       "       <path d=\"M 19.1875 8.296875 \n",
       "L 53.609375 8.296875 \n",
       "L 53.609375 0 \n",
       "L 7.328125 0 \n",
       "L 7.328125 8.296875 \n",
       "Q 12.9375 14.109375 22.625 23.890625 \n",
       "Q 32.328125 33.6875 34.8125 36.53125 \n",
       "Q 39.546875 41.84375 41.421875 45.53125 \n",
       "Q 43.3125 49.21875 43.3125 52.78125 \n",
       "Q 43.3125 58.59375 39.234375 62.25 \n",
       "Q 35.15625 65.921875 28.609375 65.921875 \n",
       "Q 23.96875 65.921875 18.8125 64.3125 \n",
       "Q 13.671875 62.703125 7.8125 59.421875 \n",
       "L 7.8125 69.390625 \n",
       "Q 13.765625 71.78125 18.9375 73 \n",
       "Q 24.125 74.21875 28.421875 74.21875 \n",
       "Q 39.75 74.21875 46.484375 68.546875 \n",
       "Q 53.21875 62.890625 53.21875 53.421875 \n",
       "Q 53.21875 48.921875 51.53125 44.890625 \n",
       "Q 49.859375 40.875 45.40625 35.40625 \n",
       "Q 44.1875 33.984375 37.640625 27.21875 \n",
       "Q 31.109375 20.453125 19.1875 8.296875 \n",
       "z\n",
       "\" id=\"DejaVuSans-50\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(105.148271 308.135625)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-77\"/>\n",
       "       <use x=\"86.279297\" xlink:href=\"#DejaVuSans-69\"/>\n",
       "       <use x=\"149.462891\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"219.287109\" xlink:href=\"#DejaVuSans-80\"/>\n",
       "       <use x=\"279.589844\" xlink:href=\"#DejaVuSans-50\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_3\">\n",
       "     <g id=\"line2d_5\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 127.872583 256.585 \n",
       "L 127.872583 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_6\"/>\n",
       "     <g id=\"text_3\">\n",
       "      <!-- DMD -->\n",
       "      <defs>\n",
       "       <path d=\"M 19.671875 64.796875 \n",
       "L 19.671875 8.109375 \n",
       "L 31.59375 8.109375 \n",
       "Q 46.6875 8.109375 53.6875 14.9375 \n",
       "Q 60.6875 21.78125 60.6875 36.53125 \n",
       "Q 60.6875 51.171875 53.6875 57.984375 \n",
       "Q 46.6875 64.796875 31.59375 64.796875 \n",
       "z\n",
       "M 9.8125 72.90625 \n",
       "L 30.078125 72.90625 \n",
       "Q 51.265625 72.90625 61.171875 64.09375 \n",
       "Q 71.09375 55.28125 71.09375 36.53125 \n",
       "Q 71.09375 17.671875 61.125 8.828125 \n",
       "Q 51.171875 0 30.078125 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-68\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(131.735708 293.724375)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-68\"/>\n",
       "       <use x=\"77.001953\" xlink:href=\"#DejaVuSans-77\"/>\n",
       "       <use x=\"163.28125\" xlink:href=\"#DejaVuSans-68\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_4\">\n",
       "     <g id=\"line2d_7\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 154.46002 256.585 \n",
       "L 154.46002 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_8\"/>\n",
       "     <g id=\"text_4\">\n",
       "      <!-- F8 -->\n",
       "      <defs>\n",
       "       <path d=\"M 9.8125 72.90625 \n",
       "L 51.703125 72.90625 \n",
       "L 51.703125 64.59375 \n",
       "L 19.671875 64.59375 \n",
       "L 19.671875 43.109375 \n",
       "L 48.578125 43.109375 \n",
       "L 48.578125 34.8125 \n",
       "L 19.671875 34.8125 \n",
       "L 19.671875 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-70\"/>\n",
       "       <path d=\"M 31.78125 34.625 \n",
       "Q 24.75 34.625 20.71875 30.859375 \n",
       "Q 16.703125 27.09375 16.703125 20.515625 \n",
       "Q 16.703125 13.921875 20.71875 10.15625 \n",
       "Q 24.75 6.390625 31.78125 6.390625 \n",
       "Q 38.8125 6.390625 42.859375 10.171875 \n",
       "Q 46.921875 13.96875 46.921875 20.515625 \n",
       "Q 46.921875 27.09375 42.890625 30.859375 \n",
       "Q 38.875 34.625 31.78125 34.625 \n",
       "z\n",
       "M 21.921875 38.8125 \n",
       "Q 15.578125 40.375 12.03125 44.71875 \n",
       "Q 8.5 49.078125 8.5 55.328125 \n",
       "Q 8.5 64.0625 14.71875 69.140625 \n",
       "Q 20.953125 74.21875 31.78125 74.21875 \n",
       "Q 42.671875 74.21875 48.875 69.140625 \n",
       "Q 55.078125 64.0625 55.078125 55.328125 \n",
       "Q 55.078125 49.078125 51.53125 44.71875 \n",
       "Q 48 40.375 41.703125 38.8125 \n",
       "Q 48.828125 37.15625 52.796875 32.3125 \n",
       "Q 56.78125 27.484375 56.78125 20.515625 \n",
       "Q 56.78125 9.90625 50.3125 4.234375 \n",
       "Q 43.84375 -1.421875 31.78125 -1.421875 \n",
       "Q 19.734375 -1.421875 13.25 4.234375 \n",
       "Q 6.78125 9.90625 6.78125 20.515625 \n",
       "Q 6.78125 27.484375 10.78125 32.3125 \n",
       "Q 14.796875 37.15625 21.921875 38.8125 \n",
       "z\n",
       "M 18.3125 54.390625 \n",
       "Q 18.3125 48.734375 21.84375 45.5625 \n",
       "Q 25.390625 42.390625 31.78125 42.390625 \n",
       "Q 38.140625 42.390625 41.71875 45.5625 \n",
       "Q 45.3125 48.734375 45.3125 54.390625 \n",
       "Q 45.3125 60.0625 41.71875 63.234375 \n",
       "Q 38.140625 66.40625 31.78125 66.40625 \n",
       "Q 25.390625 66.40625 21.84375 63.234375 \n",
       "Q 18.3125 60.0625 18.3125 54.390625 \n",
       "z\n",
       "\" id=\"DejaVuSans-56\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(158.323145 277.044688)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-70\"/>\n",
       "       <use x=\"57.519531\" xlink:href=\"#DejaVuSans-56\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_5\">\n",
       "     <g id=\"line2d_9\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 181.047457 256.585 \n",
       "L 181.047457 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_10\"/>\n",
       "     <g id=\"text_5\">\n",
       "      <!-- CDKL5 -->\n",
       "      <defs>\n",
       "       <path d=\"M 9.8125 72.90625 \n",
       "L 19.671875 72.90625 \n",
       "L 19.671875 42.09375 \n",
       "L 52.390625 72.90625 \n",
       "L 65.09375 72.90625 \n",
       "L 28.90625 38.921875 \n",
       "L 67.671875 0 \n",
       "L 54.6875 0 \n",
       "L 19.671875 35.109375 \n",
       "L 19.671875 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-75\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(184.910582 306.53)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"69.824219\" xlink:href=\"#DejaVuSans-68\"/>\n",
       "       <use x=\"146.826172\" xlink:href=\"#DejaVuSans-75\"/>\n",
       "       <use x=\"212.402344\" xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"268.115234\" xlink:href=\"#DejaVuSans-53\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"line2d_11\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 57.68175 256.585 \n",
       "L 198.063417 256.585 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_12\"/>\n",
       "     <g id=\"text_6\">\n",
       "      <!-- 0 -->\n",
       "      <defs>\n",
       "       <path d=\"M 31.78125 66.40625 \n",
       "Q 24.171875 66.40625 20.328125 58.90625 \n",
       "Q 16.5 51.421875 16.5 36.375 \n",
       "Q 16.5 21.390625 20.328125 13.890625 \n",
       "Q 24.171875 6.390625 31.78125 6.390625 \n",
       "Q 39.453125 6.390625 43.28125 13.890625 \n",
       "Q 47.125 21.390625 47.125 36.375 \n",
       "Q 47.125 51.421875 43.28125 58.90625 \n",
       "Q 39.453125 66.40625 31.78125 66.40625 \n",
       "z\n",
       "M 31.78125 74.21875 \n",
       "Q 44.046875 74.21875 50.515625 64.515625 \n",
       "Q 56.984375 54.828125 56.984375 36.375 \n",
       "Q 56.984375 17.96875 50.515625 8.265625 \n",
       "Q 44.046875 -1.421875 31.78125 -1.421875 \n",
       "Q 19.53125 -1.421875 13.0625 8.265625 \n",
       "Q 6.59375 17.96875 6.59375 36.375 \n",
       "Q 6.59375 54.828125 13.0625 64.515625 \n",
       "Q 19.53125 74.21875 31.78125 74.21875 \n",
       "z\n",
       "\" id=\"DejaVuSans-48\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(45.27425 261.903906)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"line2d_13\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 57.68175 219.181925 \n",
       "L 198.063417 219.181925 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_14\"/>\n",
       "     <g id=\"text_7\">\n",
       "      <!-- 100 -->\n",
       "      <defs>\n",
       "       <path d=\"M 12.40625 8.296875 \n",
       "L 28.515625 8.296875 \n",
       "L 28.515625 63.921875 \n",
       "L 10.984375 60.40625 \n",
       "L 10.984375 69.390625 \n",
       "L 28.421875 72.90625 \n",
       "L 38.28125 72.90625 \n",
       "L 38.28125 8.296875 \n",
       "L 54.390625 8.296875 \n",
       "L 54.390625 0 \n",
       "L 12.40625 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-49\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(27.45925 224.500832)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "       <use x=\"127.246094\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"line2d_15\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 57.68175 181.778851 \n",
       "L 198.063417 181.778851 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_16\"/>\n",
       "     <g id=\"text_8\">\n",
       "      <!-- 200 -->\n",
       "      <g transform=\"translate(27.45925 187.097757)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-50\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "       <use x=\"127.246094\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"line2d_17\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 57.68175 144.375776 \n",
       "L 198.063417 144.375776 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_18\"/>\n",
       "     <g id=\"text_9\">\n",
       "      <!-- 300 -->\n",
       "      <defs>\n",
       "       <path d=\"M 40.578125 39.3125 \n",
       "Q 47.65625 37.796875 51.625 33 \n",
       "Q 55.609375 28.21875 55.609375 21.1875 \n",
       "Q 55.609375 10.40625 48.1875 4.484375 \n",
       "Q 40.765625 -1.421875 27.09375 -1.421875 \n",
       "Q 22.515625 -1.421875 17.65625 -0.515625 \n",
       "Q 12.796875 0.390625 7.625 2.203125 \n",
       "L 7.625 11.71875 \n",
       "Q 11.71875 9.328125 16.59375 8.109375 \n",
       "Q 21.484375 6.890625 26.8125 6.890625 \n",
       "Q 36.078125 6.890625 40.9375 10.546875 \n",
       "Q 45.796875 14.203125 45.796875 21.1875 \n",
       "Q 45.796875 27.640625 41.28125 31.265625 \n",
       "Q 36.765625 34.90625 28.71875 34.90625 \n",
       "L 20.21875 34.90625 \n",
       "L 20.21875 43.015625 \n",
       "L 29.109375 43.015625 \n",
       "Q 36.375 43.015625 40.234375 45.921875 \n",
       "Q 44.09375 48.828125 44.09375 54.296875 \n",
       "Q 44.09375 59.90625 40.109375 62.90625 \n",
       "Q 36.140625 65.921875 28.71875 65.921875 \n",
       "Q 24.65625 65.921875 20.015625 65.03125 \n",
       "Q 15.375 64.15625 9.8125 62.3125 \n",
       "L 9.8125 71.09375 \n",
       "Q 15.4375 72.65625 20.34375 73.4375 \n",
       "Q 25.25 74.21875 29.59375 74.21875 \n",
       "Q 40.828125 74.21875 47.359375 69.109375 \n",
       "Q 53.90625 64.015625 53.90625 55.328125 \n",
       "Q 53.90625 49.265625 50.4375 45.09375 \n",
       "Q 46.96875 40.921875 40.578125 39.3125 \n",
       "z\n",
       "\" id=\"DejaVuSans-51\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(27.45925 149.694682)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-51\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "       <use x=\"127.246094\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"line2d_19\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 57.68175 106.972702 \n",
       "L 198.063417 106.972702 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_20\"/>\n",
       "     <g id=\"text_10\">\n",
       "      <!-- 400 -->\n",
       "      <g transform=\"translate(27.45925 112.291608)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-52\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "       <use x=\"127.246094\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"line2d_21\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 57.68175 69.569627 \n",
       "L 198.063417 69.569627 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_22\"/>\n",
       "     <g id=\"text_11\">\n",
       "      <!-- 500 -->\n",
       "      <g transform=\"translate(27.45925 74.888533)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-53\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "       <use x=\"127.246094\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_7\">\n",
       "     <g id=\"line2d_23\">\n",
       "      <path clip-path=\"url(#p73750b554c)\" d=\"M 57.68175 32.166552 \n",
       "L 198.063417 32.166552 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_24\"/>\n",
       "     <g id=\"text_12\">\n",
       "      <!-- 600 -->\n",
       "      <defs>\n",
       "       <path d=\"M 33.015625 40.375 \n",
       "Q 26.375 40.375 22.484375 35.828125 \n",
       "Q 18.609375 31.296875 18.609375 23.390625 \n",
       "Q 18.609375 15.53125 22.484375 10.953125 \n",
       "Q 26.375 6.390625 33.015625 6.390625 \n",
       "Q 39.65625 6.390625 43.53125 10.953125 \n",
       "Q 47.40625 15.53125 47.40625 23.390625 \n",
       "Q 47.40625 31.296875 43.53125 35.828125 \n",
       "Q 39.65625 40.375 33.015625 40.375 \n",
       "z\n",
       "M 52.59375 71.296875 \n",
       "L 52.59375 62.3125 \n",
       "Q 48.875 64.0625 45.09375 64.984375 \n",
       "Q 41.3125 65.921875 37.59375 65.921875 \n",
       "Q 27.828125 65.921875 22.671875 59.328125 \n",
       "Q 17.53125 52.734375 16.796875 39.40625 \n",
       "Q 19.671875 43.65625 24.015625 45.921875 \n",
       "Q 28.375 48.1875 33.59375 48.1875 \n",
       "Q 44.578125 48.1875 50.953125 41.515625 \n",
       "Q 57.328125 34.859375 57.328125 23.390625 \n",
       "Q 57.328125 12.15625 50.6875 5.359375 \n",
       "Q 44.046875 -1.421875 33.015625 -1.421875 \n",
       "Q 20.359375 -1.421875 13.671875 8.265625 \n",
       "Q 6.984375 17.96875 6.984375 36.375 \n",
       "Q 6.984375 53.65625 15.1875 63.9375 \n",
       "Q 23.390625 74.21875 37.203125 74.21875 \n",
       "Q 40.921875 74.21875 44.703125 73.484375 \n",
       "Q 48.484375 72.75 52.59375 71.296875 \n",
       "z\n",
       "\" id=\"DejaVuSans-54\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(27.45925 37.485459)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-54\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "       <use x=\"127.246094\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_13\">\n",
       "     <!-- # Variants Count -->\n",
       "     <defs>\n",
       "      <path d=\"M 51.125 44 \n",
       "L 36.921875 44 \n",
       "L 32.8125 27.6875 \n",
       "L 47.125 27.6875 \n",
       "z\n",
       "M 43.796875 71.78125 \n",
       "L 38.71875 51.515625 \n",
       "L 52.984375 51.515625 \n",
       "L 58.109375 71.78125 \n",
       "L 65.921875 71.78125 \n",
       "L 60.890625 51.515625 \n",
       "L 76.125 51.515625 \n",
       "L 76.125 44 \n",
       "L 58.984375 44 \n",
       "L 54.984375 27.6875 \n",
       "L 70.515625 27.6875 \n",
       "L 70.515625 20.21875 \n",
       "L 53.078125 20.21875 \n",
       "L 48 0 \n",
       "L 40.1875 0 \n",
       "L 45.21875 20.21875 \n",
       "L 30.90625 20.21875 \n",
       "L 25.875 0 \n",
       "L 18.015625 0 \n",
       "L 23.09375 20.21875 \n",
       "L 7.71875 20.21875 \n",
       "L 7.71875 27.6875 \n",
       "L 24.90625 27.6875 \n",
       "L 29 44 \n",
       "L 13.28125 44 \n",
       "L 13.28125 51.515625 \n",
       "L 30.90625 51.515625 \n",
       "L 35.890625 71.78125 \n",
       "z\n",
       "\" id=\"DejaVuSans-35\"/>\n",
       "      <path id=\"DejaVuSans-32\"/>\n",
       "      <path d=\"M 28.609375 0 \n",
       "L 0.78125 72.90625 \n",
       "L 11.078125 72.90625 \n",
       "L 34.1875 11.53125 \n",
       "L 57.328125 72.90625 \n",
       "L 67.578125 72.90625 \n",
       "L 39.796875 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-86\"/>\n",
       "      <path d=\"M 34.28125 27.484375 \n",
       "Q 23.390625 27.484375 19.1875 25 \n",
       "Q 14.984375 22.515625 14.984375 16.5 \n",
       "Q 14.984375 11.71875 18.140625 8.90625 \n",
       "Q 21.296875 6.109375 26.703125 6.109375 \n",
       "Q 34.1875 6.109375 38.703125 11.40625 \n",
       "Q 43.21875 16.703125 43.21875 25.484375 \n",
       "L 43.21875 27.484375 \n",
       "z\n",
       "M 52.203125 31.203125 \n",
       "L 52.203125 0 \n",
       "L 43.21875 0 \n",
       "L 43.21875 8.296875 \n",
       "Q 40.140625 3.328125 35.546875 0.953125 \n",
       "Q 30.953125 -1.421875 24.3125 -1.421875 \n",
       "Q 15.921875 -1.421875 10.953125 3.296875 \n",
       "Q 6 8.015625 6 15.921875 \n",
       "Q 6 25.140625 12.171875 29.828125 \n",
       "Q 18.359375 34.515625 30.609375 34.515625 \n",
       "L 43.21875 34.515625 \n",
       "L 43.21875 35.40625 \n",
       "Q 43.21875 41.609375 39.140625 45 \n",
       "Q 35.0625 48.390625 27.6875 48.390625 \n",
       "Q 23 48.390625 18.546875 47.265625 \n",
       "Q 14.109375 46.140625 10.015625 43.890625 \n",
       "L 10.015625 52.203125 \n",
       "Q 14.9375 54.109375 19.578125 55.046875 \n",
       "Q 24.21875 56 28.609375 56 \n",
       "Q 40.484375 56 46.34375 49.84375 \n",
       "Q 52.203125 43.703125 52.203125 31.203125 \n",
       "z\n",
       "\" id=\"DejaVuSans-97\"/>\n",
       "      <path d=\"M 41.109375 46.296875 \n",
       "Q 39.59375 47.171875 37.8125 47.578125 \n",
       "Q 36.03125 48 33.890625 48 \n",
       "Q 26.265625 48 22.1875 43.046875 \n",
       "Q 18.109375 38.09375 18.109375 28.8125 \n",
       "L 18.109375 0 \n",
       "L 9.078125 0 \n",
       "L 9.078125 54.6875 \n",
       "L 18.109375 54.6875 \n",
       "L 18.109375 46.1875 \n",
       "Q 20.953125 51.171875 25.484375 53.578125 \n",
       "Q 30.03125 56 36.53125 56 \n",
       "Q 37.453125 56 38.578125 55.875 \n",
       "Q 39.703125 55.765625 41.0625 55.515625 \n",
       "z\n",
       "\" id=\"DejaVuSans-114\"/>\n",
       "      <path d=\"M 9.421875 54.6875 \n",
       "L 18.40625 54.6875 \n",
       "L 18.40625 0 \n",
       "L 9.421875 0 \n",
       "z\n",
       "M 9.421875 75.984375 \n",
       "L 18.40625 75.984375 \n",
       "L 18.40625 64.59375 \n",
       "L 9.421875 64.59375 \n",
       "z\n",
       "\" id=\"DejaVuSans-105\"/>\n",
       "      <path d=\"M 54.890625 33.015625 \n",
       "L 54.890625 0 \n",
       "L 45.90625 0 \n",
       "L 45.90625 32.71875 \n",
       "Q 45.90625 40.484375 42.875 44.328125 \n",
       "Q 39.84375 48.1875 33.796875 48.1875 \n",
       "Q 26.515625 48.1875 22.3125 43.546875 \n",
       "Q 18.109375 38.921875 18.109375 30.90625 \n",
       "L 18.109375 0 \n",
       "L 9.078125 0 \n",
       "L 9.078125 54.6875 \n",
       "L 18.109375 54.6875 \n",
       "L 18.109375 46.1875 \n",
       "Q 21.34375 51.125 25.703125 53.5625 \n",
       "Q 30.078125 56 35.796875 56 \n",
       "Q 45.21875 56 50.046875 50.171875 \n",
       "Q 54.890625 44.34375 54.890625 33.015625 \n",
       "z\n",
       "\" id=\"DejaVuSans-110\"/>\n",
       "      <path d=\"M 18.3125 70.21875 \n",
       "L 18.3125 54.6875 \n",
       "L 36.8125 54.6875 \n",
       "L 36.8125 47.703125 \n",
       "L 18.3125 47.703125 \n",
       "L 18.3125 18.015625 \n",
       "Q 18.3125 11.328125 20.140625 9.421875 \n",
       "Q 21.96875 7.515625 27.59375 7.515625 \n",
       "L 36.8125 7.515625 \n",
       "L 36.8125 0 \n",
       "L 27.59375 0 \n",
       "Q 17.1875 0 13.234375 3.875 \n",
       "Q 9.28125 7.765625 9.28125 18.015625 \n",
       "L 9.28125 47.703125 \n",
       "L 2.6875 47.703125 \n",
       "L 2.6875 54.6875 \n",
       "L 9.28125 54.6875 \n",
       "L 9.28125 70.21875 \n",
       "z\n",
       "\" id=\"DejaVuSans-116\"/>\n",
       "      <path d=\"M 44.28125 53.078125 \n",
       "L 44.28125 44.578125 \n",
       "Q 40.484375 46.53125 36.375 47.5 \n",
       "Q 32.28125 48.484375 27.875 48.484375 \n",
       "Q 21.1875 48.484375 17.84375 46.4375 \n",
       "Q 14.5 44.390625 14.5 40.28125 \n",
       "Q 14.5 37.15625 16.890625 35.375 \n",
       "Q 19.28125 33.59375 26.515625 31.984375 \n",
       "L 29.59375 31.296875 \n",
       "Q 39.15625 29.25 43.1875 25.515625 \n",
       "Q 47.21875 21.78125 47.21875 15.09375 \n",
       "Q 47.21875 7.46875 41.1875 3.015625 \n",
       "Q 35.15625 -1.421875 24.609375 -1.421875 \n",
       "Q 20.21875 -1.421875 15.453125 -0.5625 \n",
       "Q 10.6875 0.296875 5.421875 2 \n",
       "L 5.421875 11.28125 \n",
       "Q 10.40625 8.6875 15.234375 7.390625 \n",
       "Q 20.0625 6.109375 24.8125 6.109375 \n",
       "Q 31.15625 6.109375 34.5625 8.28125 \n",
       "Q 37.984375 10.453125 37.984375 14.40625 \n",
       "Q 37.984375 18.0625 35.515625 20.015625 \n",
       "Q 33.0625 21.96875 24.703125 23.78125 \n",
       "L 21.578125 24.515625 \n",
       "Q 13.234375 26.265625 9.515625 29.90625 \n",
       "Q 5.8125 33.546875 5.8125 39.890625 \n",
       "Q 5.8125 47.609375 11.28125 51.796875 \n",
       "Q 16.75 56 26.8125 56 \n",
       "Q 31.78125 56 36.171875 55.265625 \n",
       "Q 40.578125 54.546875 44.28125 53.078125 \n",
       "z\n",
       "\" id=\"DejaVuSans-115\"/>\n",
       "      <path d=\"M 30.609375 48.390625 \n",
       "Q 23.390625 48.390625 19.1875 42.75 \n",
       "Q 14.984375 37.109375 14.984375 27.296875 \n",
       "Q 14.984375 17.484375 19.15625 11.84375 \n",
       "Q 23.34375 6.203125 30.609375 6.203125 \n",
       "Q 37.796875 6.203125 41.984375 11.859375 \n",
       "Q 46.1875 17.53125 46.1875 27.296875 \n",
       "Q 46.1875 37.015625 41.984375 42.703125 \n",
       "Q 37.796875 48.390625 30.609375 48.390625 \n",
       "z\n",
       "M 30.609375 56 \n",
       "Q 42.328125 56 49.015625 48.375 \n",
       "Q 55.71875 40.765625 55.71875 27.296875 \n",
       "Q 55.71875 13.875 49.015625 6.21875 \n",
       "Q 42.328125 -1.421875 30.609375 -1.421875 \n",
       "Q 18.84375 -1.421875 12.171875 6.21875 \n",
       "Q 5.515625 13.875 5.515625 27.296875 \n",
       "Q 5.515625 40.765625 12.171875 48.375 \n",
       "Q 18.84375 56 30.609375 56 \n",
       "z\n",
       "\" id=\"DejaVuSans-111\"/>\n",
       "      <path d=\"M 8.5 21.578125 \n",
       "L 8.5 54.6875 \n",
       "L 17.484375 54.6875 \n",
       "L 17.484375 21.921875 \n",
       "Q 17.484375 14.15625 20.5 10.265625 \n",
       "Q 23.53125 6.390625 29.59375 6.390625 \n",
       "Q 36.859375 6.390625 41.078125 11.03125 \n",
       "Q 45.3125 15.671875 45.3125 23.6875 \n",
       "L 45.3125 54.6875 \n",
       "L 54.296875 54.6875 \n",
       "L 54.296875 0 \n",
       "L 45.3125 0 \n",
       "L 45.3125 8.40625 \n",
       "Q 42.046875 3.421875 37.71875 1 \n",
       "Q 33.40625 -1.421875 27.6875 -1.421875 \n",
       "Q 18.265625 -1.421875 13.375 4.4375 \n",
       "Q 8.5 10.296875 8.5 21.578125 \n",
       "z\n",
       "M 31.109375 56 \n",
       "z\n",
       "\" id=\"DejaVuSans-117\"/>\n",
       "     </defs>\n",
       "     <g transform=\"translate(19.965375 204.0275)rotate(-90)scale(0.168 -0.168)\">\n",
       "      <use xlink:href=\"#DejaVuSans-35\"/>\n",
       "      <use x=\"83.789062\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"115.576172\" xlink:href=\"#DejaVuSans-86\"/>\n",
       "      <use x=\"183.875\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"245.154297\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"286.267578\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "      <use x=\"314.050781\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"375.330078\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "      <use x=\"438.708984\" xlink:href=\"#DejaVuSans-116\"/>\n",
       "      <use x=\"477.917969\" xlink:href=\"#DejaVuSans-115\"/>\n",
       "      <use x=\"530.017578\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"561.804688\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "      <use x=\"631.628906\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "      <use x=\"692.810547\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "      <use x=\"756.189453\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "      <use x=\"819.568359\" xlink:href=\"#DejaVuSans-116\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_3\">\n",
       "    <path clip-path=\"url(#p73750b554c)\" d=\"M 64.062735 256.585 \n",
       "L 85.332684 256.585 \n",
       "L 85.332684 19.075476 \n",
       "L 64.062735 19.075476 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_4\">\n",
       "    <path clip-path=\"url(#p73750b554c)\" d=\"M 90.650172 256.585 \n",
       "L 111.920121 256.585 \n",
       "L 111.920121 121.18587 \n",
       "L 90.650172 121.18587 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_5\">\n",
       "    <path clip-path=\"url(#p73750b554c)\" d=\"M 117.237609 256.585 \n",
       "L 138.507558 256.585 \n",
       "L 138.507558 139.887407 \n",
       "L 117.237609 139.887407 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_6\">\n",
       "    <path clip-path=\"url(#p73750b554c)\" d=\"M 143.825045 256.585 \n",
       "L 165.094995 256.585 \n",
       "L 165.094995 157.466852 \n",
       "L 143.825045 157.466852 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_7\">\n",
       "    <path clip-path=\"url(#p73750b554c)\" d=\"M 170.412482 256.585 \n",
       "L 191.682432 256.585 \n",
       "L 191.682432 176.16839 \n",
       "L 170.412482 176.16839 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_8\">\n",
       "    <path d=\"M 57.68175 256.585 \n",
       "L 57.68175 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_9\">\n",
       "    <path d=\"M 198.063417 256.585 \n",
       "L 198.063417 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_10\">\n",
       "    <path d=\"M 57.68175 256.585 \n",
       "L 198.063417 256.585 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_11\">\n",
       "    <path d=\"M 57.68175 7.2 \n",
       "L 198.063417 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"legend_1\">\n",
       "    <g id=\"patch_12\">\n",
       "     <path d=\"M 133.873417 38.949375 \n",
       "L 188.263417 38.949375 \n",
       "Q 191.063417 38.949375 191.063417 36.149375 \n",
       "L 191.063417 17 \n",
       "Q 191.063417 14.2 188.263417 14.2 \n",
       "L 133.873417 14.2 \n",
       "Q 131.073417 14.2 131.073417 17 \n",
       "L 131.073417 36.149375 \n",
       "Q 131.073417 38.949375 133.873417 38.949375 \n",
       "z\n",
       "\" style=\"fill:#f0f0f0;opacity:0.8;stroke:#cccccc;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "    </g>\n",
       "    <g id=\"patch_13\">\n",
       "     <path d=\"M 136.673417 30.437813 \n",
       "L 164.673417 30.437813 \n",
       "L 164.673417 20.637813 \n",
       "L 136.673417 20.637813 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "    </g>\n",
       "    <g id=\"text_14\">\n",
       "     <!-- X -->\n",
       "     <defs>\n",
       "      <path d=\"M 6.296875 72.90625 \n",
       "L 16.890625 72.90625 \n",
       "L 35.015625 45.796875 \n",
       "L 53.21875 72.90625 \n",
       "L 63.8125 72.90625 \n",
       "L 40.375 37.890625 \n",
       "L 65.375 0 \n",
       "L 54.78125 0 \n",
       "L 34.28125 31 \n",
       "L 13.625 0 \n",
       "L 2.984375 0 \n",
       "L 29 38.921875 \n",
       "z\n",
       "\" id=\"DejaVuSans-88\"/>\n",
       "     </defs>\n",
       "     <g transform=\"translate(175.873417 30.437813)scale(0.14 -0.14)\">\n",
       "      <use xlink:href=\"#DejaVuSans-88\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       "  <g id=\"axes_2\">\n",
       "   <g id=\"patch_14\">\n",
       "    <path d=\"M 234.433417 256.585 \n",
       "L 374.815083 256.585 \n",
       "L 374.815083 7.2 \n",
       "L 234.433417 7.2 \n",
       "z\n",
       "\" style=\"fill:#f0f0f0;\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_3\">\n",
       "    <g id=\"xtick_6\">\n",
       "     <g id=\"line2d_25\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 251.449376 256.585 \n",
       "L 251.449376 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_26\"/>\n",
       "     <g id=\"text_15\">\n",
       "      <!-- SRY -->\n",
       "      <defs>\n",
       "       <path d=\"M 53.515625 70.515625 \n",
       "L 53.515625 60.890625 \n",
       "Q 47.90625 63.578125 42.921875 64.890625 \n",
       "Q 37.9375 66.21875 33.296875 66.21875 \n",
       "Q 25.25 66.21875 20.875 63.09375 \n",
       "Q 16.5 59.96875 16.5 54.203125 \n",
       "Q 16.5 49.359375 19.40625 46.890625 \n",
       "Q 22.3125 44.4375 30.421875 42.921875 \n",
       "L 36.375 41.703125 \n",
       "Q 47.40625 39.59375 52.65625 34.296875 \n",
       "Q 57.90625 29 57.90625 20.125 \n",
       "Q 57.90625 9.515625 50.796875 4.046875 \n",
       "Q 43.703125 -1.421875 29.984375 -1.421875 \n",
       "Q 24.8125 -1.421875 18.96875 -0.25 \n",
       "Q 13.140625 0.921875 6.890625 3.21875 \n",
       "L 6.890625 13.375 \n",
       "Q 12.890625 10.015625 18.65625 8.296875 \n",
       "Q 24.421875 6.59375 29.984375 6.59375 \n",
       "Q 38.421875 6.59375 43.015625 9.90625 \n",
       "Q 47.609375 13.234375 47.609375 19.390625 \n",
       "Q 47.609375 24.75 44.3125 27.78125 \n",
       "Q 41.015625 30.8125 33.5 32.328125 \n",
       "L 27.484375 33.5 \n",
       "Q 16.453125 35.6875 11.515625 40.375 \n",
       "Q 6.59375 45.0625 6.59375 53.421875 \n",
       "Q 6.59375 63.09375 13.40625 68.65625 \n",
       "Q 20.21875 74.21875 32.171875 74.21875 \n",
       "Q 37.3125 74.21875 42.625 73.28125 \n",
       "Q 47.953125 72.359375 53.515625 70.515625 \n",
       "z\n",
       "\" id=\"DejaVuSans-83\"/>\n",
       "       <path d=\"M 44.390625 34.1875 \n",
       "Q 47.5625 33.109375 50.5625 29.59375 \n",
       "Q 53.5625 26.078125 56.59375 19.921875 \n",
       "L 66.609375 0 \n",
       "L 56 0 \n",
       "L 46.6875 18.703125 \n",
       "Q 43.0625 26.03125 39.671875 28.421875 \n",
       "Q 36.28125 30.8125 30.421875 30.8125 \n",
       "L 19.671875 30.8125 \n",
       "L 19.671875 0 \n",
       "L 9.8125 0 \n",
       "L 9.8125 72.90625 \n",
       "L 32.078125 72.90625 \n",
       "Q 44.578125 72.90625 50.734375 67.671875 \n",
       "Q 56.890625 62.453125 56.890625 51.90625 \n",
       "Q 56.890625 45.015625 53.6875 40.46875 \n",
       "Q 50.484375 35.9375 44.390625 34.1875 \n",
       "z\n",
       "M 19.671875 64.796875 \n",
       "L 19.671875 38.921875 \n",
       "L 32.078125 38.921875 \n",
       "Q 39.203125 38.921875 42.84375 42.21875 \n",
       "Q 46.484375 45.515625 46.484375 51.90625 \n",
       "Q 46.484375 58.296875 42.84375 61.546875 \n",
       "Q 39.203125 64.796875 32.078125 64.796875 \n",
       "z\n",
       "\" id=\"DejaVuSans-82\"/>\n",
       "       <path d=\"M -0.203125 72.90625 \n",
       "L 10.40625 72.90625 \n",
       "L 30.609375 42.921875 \n",
       "L 50.6875 72.90625 \n",
       "L 61.28125 72.90625 \n",
       "L 35.5 34.71875 \n",
       "L 35.5 0 \n",
       "L 25.59375 0 \n",
       "L 25.59375 34.71875 \n",
       "z\n",
       "\" id=\"DejaVuSans-89\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(255.312501 287.238437)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-83\"/>\n",
       "       <use x=\"63.476562\" xlink:href=\"#DejaVuSans-82\"/>\n",
       "       <use x=\"132.865234\" xlink:href=\"#DejaVuSans-89\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_7\">\n",
       "     <g id=\"line2d_27\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 278.036813 256.585 \n",
       "L 278.036813 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_28\"/>\n",
       "     <g id=\"text_16\">\n",
       "      <!-- SHOX -->\n",
       "      <defs>\n",
       "       <path d=\"M 9.8125 72.90625 \n",
       "L 19.671875 72.90625 \n",
       "L 19.671875 43.015625 \n",
       "L 55.515625 43.015625 \n",
       "L 55.515625 72.90625 \n",
       "L 65.375 72.90625 \n",
       "L 65.375 0 \n",
       "L 55.515625 0 \n",
       "L 55.515625 34.71875 \n",
       "L 19.671875 34.71875 \n",
       "L 19.671875 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-72\"/>\n",
       "      </defs>\n",
       "      <g transform=\"translate(281.899938 300.09875)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-83\"/>\n",
       "       <use x=\"63.476562\" xlink:href=\"#DejaVuSans-72\"/>\n",
       "       <use x=\"138.671875\" xlink:href=\"#DejaVuSans-79\"/>\n",
       "       <use x=\"217.289062\" xlink:href=\"#DejaVuSans-88\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_8\">\n",
       "     <g id=\"line2d_29\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 304.62425 256.585 \n",
       "L 304.62425 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_30\"/>\n",
       "     <g id=\"text_17\">\n",
       "      <!-- CRLF2 -->\n",
       "      <g transform=\"translate(308.487375 304.349063)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"69.824219\" xlink:href=\"#DejaVuSans-82\"/>\n",
       "       <use x=\"139.306641\" xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"195.019531\" xlink:href=\"#DejaVuSans-70\"/>\n",
       "       <use x=\"252.539062\" xlink:href=\"#DejaVuSans-50\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_9\">\n",
       "     <g id=\"line2d_31\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 331.211687 256.585 \n",
       "L 331.211687 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_32\"/>\n",
       "     <g id=\"text_18\">\n",
       "      <!-- CSF2RA -->\n",
       "      <g transform=\"translate(335.074812 315.004375)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"69.824219\" xlink:href=\"#DejaVuSans-83\"/>\n",
       "       <use x=\"133.300781\" xlink:href=\"#DejaVuSans-70\"/>\n",
       "       <use x=\"190.820312\" xlink:href=\"#DejaVuSans-50\"/>\n",
       "       <use x=\"254.443359\" xlink:href=\"#DejaVuSans-82\"/>\n",
       "       <use x=\"323.863281\" xlink:href=\"#DejaVuSans-65\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_10\">\n",
       "     <g id=\"line2d_33\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 357.799124 256.585 \n",
       "L 357.799124 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_34\"/>\n",
       "     <g id=\"text_19\">\n",
       "      <!-- SPRY3 -->\n",
       "      <g transform=\"translate(361.662249 304.5875)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-83\"/>\n",
       "       <use x=\"63.476562\" xlink:href=\"#DejaVuSans-80\"/>\n",
       "       <use x=\"123.779297\" xlink:href=\"#DejaVuSans-82\"/>\n",
       "       <use x=\"193.167969\" xlink:href=\"#DejaVuSans-89\"/>\n",
       "       <use x=\"254.251953\" xlink:href=\"#DejaVuSans-51\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_20\">\n",
       "     <!-- Gene Names -->\n",
       "     <defs>\n",
       "      <path d=\"M 59.515625 10.40625 \n",
       "L 59.515625 29.984375 \n",
       "L 43.40625 29.984375 \n",
       "L 43.40625 38.09375 \n",
       "L 69.28125 38.09375 \n",
       "L 69.28125 6.78125 \n",
       "Q 63.578125 2.734375 56.6875 0.65625 \n",
       "Q 49.8125 -1.421875 42 -1.421875 \n",
       "Q 24.90625 -1.421875 15.25 8.5625 \n",
       "Q 5.609375 18.5625 5.609375 36.375 \n",
       "Q 5.609375 54.25 15.25 64.234375 \n",
       "Q 24.90625 74.21875 42 74.21875 \n",
       "Q 49.125 74.21875 55.546875 72.453125 \n",
       "Q 61.96875 70.703125 67.390625 67.28125 \n",
       "L 67.390625 56.78125 \n",
       "Q 61.921875 61.421875 55.765625 63.765625 \n",
       "Q 49.609375 66.109375 42.828125 66.109375 \n",
       "Q 29.4375 66.109375 22.71875 58.640625 \n",
       "Q 16.015625 51.171875 16.015625 36.375 \n",
       "Q 16.015625 21.625 22.71875 14.15625 \n",
       "Q 29.4375 6.6875 42.828125 6.6875 \n",
       "Q 48.046875 6.6875 52.140625 7.59375 \n",
       "Q 56.25 8.5 59.515625 10.40625 \n",
       "z\n",
       "\" id=\"DejaVuSans-71\"/>\n",
       "      <path d=\"M 56.203125 29.59375 \n",
       "L 56.203125 25.203125 \n",
       "L 14.890625 25.203125 \n",
       "Q 15.484375 15.921875 20.484375 11.0625 \n",
       "Q 25.484375 6.203125 34.421875 6.203125 \n",
       "Q 39.59375 6.203125 44.453125 7.46875 \n",
       "Q 49.3125 8.734375 54.109375 11.28125 \n",
       "L 54.109375 2.78125 \n",
       "Q 49.265625 0.734375 44.1875 -0.34375 \n",
       "Q 39.109375 -1.421875 33.890625 -1.421875 \n",
       "Q 20.796875 -1.421875 13.15625 6.1875 \n",
       "Q 5.515625 13.8125 5.515625 26.8125 \n",
       "Q 5.515625 40.234375 12.765625 48.109375 \n",
       "Q 20.015625 56 32.328125 56 \n",
       "Q 43.359375 56 49.78125 48.890625 \n",
       "Q 56.203125 41.796875 56.203125 29.59375 \n",
       "z\n",
       "M 47.21875 32.234375 \n",
       "Q 47.125 39.59375 43.09375 43.984375 \n",
       "Q 39.0625 48.390625 32.421875 48.390625 \n",
       "Q 24.90625 48.390625 20.390625 44.140625 \n",
       "Q 15.875 39.890625 15.1875 32.171875 \n",
       "z\n",
       "\" id=\"DejaVuSans-101\"/>\n",
       "      <path d=\"M 9.8125 72.90625 \n",
       "L 23.09375 72.90625 \n",
       "L 55.421875 11.921875 \n",
       "L 55.421875 72.90625 \n",
       "L 64.984375 72.90625 \n",
       "L 64.984375 0 \n",
       "L 51.703125 0 \n",
       "L 19.390625 60.984375 \n",
       "L 19.390625 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-78\"/>\n",
       "      <path d=\"M 52 44.1875 \n",
       "Q 55.375 50.25 60.0625 53.125 \n",
       "Q 64.75 56 71.09375 56 \n",
       "Q 79.640625 56 84.28125 50.015625 \n",
       "Q 88.921875 44.046875 88.921875 33.015625 \n",
       "L 88.921875 0 \n",
       "L 79.890625 0 \n",
       "L 79.890625 32.71875 \n",
       "Q 79.890625 40.578125 77.09375 44.375 \n",
       "Q 74.3125 48.1875 68.609375 48.1875 \n",
       "Q 61.625 48.1875 57.5625 43.546875 \n",
       "Q 53.515625 38.921875 53.515625 30.90625 \n",
       "L 53.515625 0 \n",
       "L 44.484375 0 \n",
       "L 44.484375 32.71875 \n",
       "Q 44.484375 40.625 41.703125 44.40625 \n",
       "Q 38.921875 48.1875 33.109375 48.1875 \n",
       "Q 26.21875 48.1875 22.15625 43.53125 \n",
       "Q 18.109375 38.875 18.109375 30.90625 \n",
       "L 18.109375 0 \n",
       "L 9.078125 0 \n",
       "L 9.078125 54.6875 \n",
       "L 18.109375 54.6875 \n",
       "L 18.109375 46.1875 \n",
       "Q 21.1875 51.21875 25.484375 53.609375 \n",
       "Q 29.78125 56 35.6875 56 \n",
       "Q 41.65625 56 45.828125 52.96875 \n",
       "Q 50 49.953125 52 44.1875 \n",
       "z\n",
       "\" id=\"DejaVuSans-109\"/>\n",
       "     </defs>\n",
       "     <g transform=\"translate(250.626688 331.76975)scale(0.168 -0.168)\">\n",
       "      <use xlink:href=\"#DejaVuSans-71\"/>\n",
       "      <use x=\"77.490234\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"139.013672\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "      <use x=\"202.392578\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"263.916016\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"295.703125\" xlink:href=\"#DejaVuSans-78\"/>\n",
       "      <use x=\"370.507812\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"431.787109\" xlink:href=\"#DejaVuSans-109\"/>\n",
       "      <use x=\"529.199219\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"590.722656\" xlink:href=\"#DejaVuSans-115\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_4\">\n",
       "    <g id=\"ytick_8\">\n",
       "     <g id=\"line2d_35\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 234.433417 256.585 \n",
       "L 374.815083 256.585 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_36\"/>\n",
       "     <g id=\"text_21\">\n",
       "      <!-- 0 -->\n",
       "      <g transform=\"translate(222.025917 261.903906)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_9\">\n",
       "     <g id=\"line2d_37\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 234.433417 212.601755 \n",
       "L 374.815083 212.601755 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_38\"/>\n",
       "     <g id=\"text_22\">\n",
       "      <!-- 5 -->\n",
       "      <g transform=\"translate(222.025917 217.920661)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-53\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_10\">\n",
       "     <g id=\"line2d_39\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 234.433417 168.61851 \n",
       "L 374.815083 168.61851 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_40\"/>\n",
       "     <g id=\"text_23\">\n",
       "      <!-- 10 -->\n",
       "      <g transform=\"translate(213.118417 173.937416)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_11\">\n",
       "     <g id=\"line2d_41\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 234.433417 124.635265 \n",
       "L 374.815083 124.635265 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_42\"/>\n",
       "     <g id=\"text_24\">\n",
       "      <!-- 15 -->\n",
       "      <g transform=\"translate(213.118417 129.954171)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-53\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_12\">\n",
       "     <g id=\"line2d_43\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 234.433417 80.652019 \n",
       "L 374.815083 80.652019 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_44\"/>\n",
       "     <g id=\"text_25\">\n",
       "      <!-- 20 -->\n",
       "      <g transform=\"translate(213.118417 85.970926)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-50\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_13\">\n",
       "     <g id=\"line2d_45\">\n",
       "      <path clip-path=\"url(#p98c49c4735)\" d=\"M 234.433417 36.668774 \n",
       "L 374.815083 36.668774 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_46\"/>\n",
       "     <g id=\"text_26\">\n",
       "      <!-- 25 -->\n",
       "      <g transform=\"translate(213.118417 41.987681)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-50\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-53\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_15\">\n",
       "    <path clip-path=\"url(#p98c49c4735)\" d=\"M 240.814402 256.585 \n",
       "L 262.084351 256.585 \n",
       "L 262.084351 19.075476 \n",
       "L 240.814402 19.075476 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_16\">\n",
       "    <path clip-path=\"url(#p98c49c4735)\" d=\"M 267.401838 256.585 \n",
       "L 288.671788 256.585 \n",
       "L 288.671788 115.838616 \n",
       "L 267.401838 115.838616 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_17\">\n",
       "    <path clip-path=\"url(#p98c49c4735)\" d=\"M 293.989275 256.585 \n",
       "L 315.259225 256.585 \n",
       "L 315.259225 247.788351 \n",
       "L 293.989275 247.788351 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_18\">\n",
       "    <path clip-path=\"url(#p98c49c4735)\" d=\"M 320.576712 256.585 \n",
       "L 341.846662 256.585 \n",
       "L 341.846662 247.788351 \n",
       "L 320.576712 247.788351 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_19\">\n",
       "    <path clip-path=\"url(#p98c49c4735)\" d=\"M 347.164149 256.585 \n",
       "L 368.434098 256.585 \n",
       "L 368.434098 247.788351 \n",
       "L 347.164149 247.788351 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_20\">\n",
       "    <path d=\"M 234.433417 256.585 \n",
       "L 234.433417 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_21\">\n",
       "    <path d=\"M 374.815083 256.585 \n",
       "L 374.815083 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_22\">\n",
       "    <path d=\"M 234.433417 256.585 \n",
       "L 374.815083 256.585 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_23\">\n",
       "    <path d=\"M 234.433417 7.2 \n",
       "L 374.815083 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"legend_2\">\n",
       "    <g id=\"patch_24\">\n",
       "     <path d=\"M 311.664146 38.949375 \n",
       "L 365.015083 38.949375 \n",
       "Q 367.815083 38.949375 367.815083 36.149375 \n",
       "L 367.815083 17 \n",
       "Q 367.815083 14.2 365.015083 14.2 \n",
       "L 311.664146 14.2 \n",
       "Q 308.864146 14.2 308.864146 17 \n",
       "L 308.864146 36.149375 \n",
       "Q 308.864146 38.949375 311.664146 38.949375 \n",
       "z\n",
       "\" style=\"fill:#f0f0f0;opacity:0.8;stroke:#cccccc;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "    </g>\n",
       "    <g id=\"patch_25\">\n",
       "     <path d=\"M 314.464146 30.437813 \n",
       "L 342.464146 30.437813 \n",
       "L 342.464146 20.637813 \n",
       "L 314.464146 20.637813 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "    </g>\n",
       "    <g id=\"text_27\">\n",
       "     <!-- Y -->\n",
       "     <g transform=\"translate(353.664146 30.437813)scale(0.14 -0.14)\">\n",
       "      <use xlink:href=\"#DejaVuSans-89\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       "  <g id=\"axes_3\">\n",
       "   <g id=\"patch_26\">\n",
       "    <path d=\"M 411.185083 256.585 \n",
       "L 551.56675 256.585 \n",
       "L 551.56675 7.2 \n",
       "L 411.185083 7.2 \n",
       "z\n",
       "\" style=\"fill:#f0f0f0;\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_5\">\n",
       "    <g id=\"xtick_11\">\n",
       "     <g id=\"line2d_47\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 428.201043 256.585 \n",
       "L 428.201043 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_48\"/>\n",
       "     <g id=\"text_28\">\n",
       "      <!-- ND6 -->\n",
       "      <g transform=\"translate(432.064168 290.24625)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-78\"/>\n",
       "       <use x=\"74.804688\" xlink:href=\"#DejaVuSans-68\"/>\n",
       "       <use x=\"151.806641\" xlink:href=\"#DejaVuSans-54\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_12\">\n",
       "     <g id=\"line2d_49\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 454.78848 256.585 \n",
       "L 454.78848 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_50\"/>\n",
       "     <g id=\"text_29\">\n",
       "      <!-- ND4L -->\n",
       "      <g transform=\"translate(458.651605 298.046875)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-78\"/>\n",
       "       <use x=\"74.804688\" xlink:href=\"#DejaVuSans-68\"/>\n",
       "       <use x=\"151.806641\" xlink:href=\"#DejaVuSans-52\"/>\n",
       "       <use x=\"215.429688\" xlink:href=\"#DejaVuSans-76\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_13\">\n",
       "     <g id=\"line2d_51\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 481.375917 256.585 \n",
       "L 481.375917 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_52\"/>\n",
       "     <g id=\"text_30\">\n",
       "      <!-- ND2 -->\n",
       "      <g transform=\"translate(485.239042 290.24625)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-78\"/>\n",
       "       <use x=\"74.804688\" xlink:href=\"#DejaVuSans-68\"/>\n",
       "       <use x=\"151.806641\" xlink:href=\"#DejaVuSans-50\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_14\">\n",
       "     <g id=\"line2d_53\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 507.963354 256.585 \n",
       "L 507.963354 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_54\"/>\n",
       "     <g id=\"text_31\">\n",
       "      <!-- COX2 -->\n",
       "      <g transform=\"translate(511.826479 299.365937)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"69.824219\" xlink:href=\"#DejaVuSans-79\"/>\n",
       "       <use x=\"148.441406\" xlink:href=\"#DejaVuSans-88\"/>\n",
       "       <use x=\"216.947266\" xlink:href=\"#DejaVuSans-50\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_15\">\n",
       "     <g id=\"line2d_55\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 534.55079 256.585 \n",
       "L 534.55079 7.2 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_56\"/>\n",
       "     <g id=\"text_32\">\n",
       "      <!-- ND5 -->\n",
       "      <g transform=\"translate(538.413915 290.24625)rotate(-90)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-78\"/>\n",
       "       <use x=\"74.804688\" xlink:href=\"#DejaVuSans-68\"/>\n",
       "       <use x=\"151.806641\" xlink:href=\"#DejaVuSans-53\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_6\">\n",
       "    <g id=\"ytick_14\">\n",
       "     <g id=\"line2d_57\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 411.185083 256.585 \n",
       "L 551.56675 256.585 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_58\"/>\n",
       "     <g id=\"text_33\">\n",
       "      <!-- 0 -->\n",
       "      <g transform=\"translate(398.777583 261.903906)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_15\">\n",
       "     <g id=\"line2d_59\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 411.185083 218.277012 \n",
       "L 551.56675 218.277012 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_60\"/>\n",
       "     <g id=\"text_34\">\n",
       "      <!-- 10 -->\n",
       "      <g transform=\"translate(389.870083 223.595919)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_16\">\n",
       "     <g id=\"line2d_61\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 411.185083 179.969025 \n",
       "L 551.56675 179.969025 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_62\"/>\n",
       "     <g id=\"text_35\">\n",
       "      <!-- 20 -->\n",
       "      <g transform=\"translate(389.870083 185.287931)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-50\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_17\">\n",
       "     <g id=\"line2d_63\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 411.185083 141.661037 \n",
       "L 551.56675 141.661037 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_64\"/>\n",
       "     <g id=\"text_36\">\n",
       "      <!-- 30 -->\n",
       "      <g transform=\"translate(389.870083 146.979943)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-51\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_18\">\n",
       "     <g id=\"line2d_65\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 411.185083 103.353049 \n",
       "L 551.56675 103.353049 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_66\"/>\n",
       "     <g id=\"text_37\">\n",
       "      <!-- 40 -->\n",
       "      <g transform=\"translate(389.870083 108.671955)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-52\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_19\">\n",
       "     <g id=\"line2d_67\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 411.185083 65.045061 \n",
       "L 551.56675 65.045061 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_68\"/>\n",
       "     <g id=\"text_38\">\n",
       "      <!-- 50 -->\n",
       "      <g transform=\"translate(389.870083 70.363968)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-53\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_20\">\n",
       "     <g id=\"line2d_69\">\n",
       "      <path clip-path=\"url(#p0b31e1114f)\" d=\"M 411.185083 26.737074 \n",
       "L 551.56675 26.737074 \n",
       "\" style=\"fill:none;stroke:#cbcbcb;\"/>\n",
       "     </g>\n",
       "     <g id=\"line2d_70\"/>\n",
       "     <g id=\"text_39\">\n",
       "      <!-- 60 -->\n",
       "      <g transform=\"translate(389.870083 32.05598)scale(0.14 -0.14)\">\n",
       "       <use xlink:href=\"#DejaVuSans-54\"/>\n",
       "       <use x=\"63.623047\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_27\">\n",
       "    <path clip-path=\"url(#p0b31e1114f)\" d=\"M 417.566068 256.585 \n",
       "L 438.836018 256.585 \n",
       "L 438.836018 19.075476 \n",
       "L 417.566068 19.075476 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_28\">\n",
       "    <path clip-path=\"url(#p0b31e1114f)\" d=\"M 444.153505 256.585 \n",
       "L 465.423455 256.585 \n",
       "L 465.423455 130.168641 \n",
       "L 444.153505 130.168641 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_29\">\n",
       "    <path clip-path=\"url(#p0b31e1114f)\" d=\"M 470.740942 256.585 \n",
       "L 492.010891 256.585 \n",
       "L 492.010891 160.815031 \n",
       "L 470.740942 160.815031 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_30\">\n",
       "    <path clip-path=\"url(#p0b31e1114f)\" d=\"M 497.328379 256.585 \n",
       "L 518.598328 256.585 \n",
       "L 518.598328 172.307427 \n",
       "L 497.328379 172.307427 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_31\">\n",
       "    <path clip-path=\"url(#p0b31e1114f)\" d=\"M 523.915816 256.585 \n",
       "L 545.185765 256.585 \n",
       "L 545.185765 183.799823 \n",
       "L 523.915816 183.799823 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_32\">\n",
       "    <path d=\"M 411.185083 256.585 \n",
       "L 411.185083 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_33\">\n",
       "    <path d=\"M 551.56675 256.585 \n",
       "L 551.56675 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_34\">\n",
       "    <path d=\"M 411.185083 256.585 \n",
       "L 551.56675 256.585 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_35\">\n",
       "    <path d=\"M 411.185083 7.2 \n",
       "L 551.56675 7.2 \n",
       "\" style=\"fill:none;stroke:#f0f0f0;stroke-linecap:square;stroke-linejoin:miter;stroke-width:3;\"/>\n",
       "   </g>\n",
       "   <g id=\"legend_3\">\n",
       "    <g id=\"patch_36\">\n",
       "     <path d=\"M 476.336438 38.949375 \n",
       "L 541.76675 38.949375 \n",
       "Q 544.56675 38.949375 544.56675 36.149375 \n",
       "L 544.56675 17 \n",
       "Q 544.56675 14.2 541.76675 14.2 \n",
       "L 476.336438 14.2 \n",
       "Q 473.536438 14.2 473.536438 17 \n",
       "L 473.536438 36.149375 \n",
       "Q 473.536438 38.949375 476.336438 38.949375 \n",
       "z\n",
       "\" style=\"fill:#f0f0f0;opacity:0.8;stroke:#cccccc;stroke-linejoin:miter;stroke-width:0.5;\"/>\n",
       "    </g>\n",
       "    <g id=\"patch_37\">\n",
       "     <path d=\"M 479.136438 30.437813 \n",
       "L 507.136438 30.437813 \n",
       "L 507.136438 20.637813 \n",
       "L 479.136438 20.637813 \n",
       "z\n",
       "\" style=\"fill:#008fd5;\"/>\n",
       "    </g>\n",
       "    <g id=\"text_40\">\n",
       "     <!-- MT -->\n",
       "     <defs>\n",
       "      <path d=\"M -0.296875 72.90625 \n",
       "L 61.375 72.90625 \n",
       "L 61.375 64.59375 \n",
       "L 35.5 64.59375 \n",
       "L 35.5 0 \n",
       "L 25.59375 0 \n",
       "L 25.59375 64.59375 \n",
       "L -0.296875 64.59375 \n",
       "z\n",
       "\" id=\"DejaVuSans-84\"/>\n",
       "     </defs>\n",
       "     <g transform=\"translate(518.336438 30.437813)scale(0.14 -0.14)\">\n",
       "      <use xlink:href=\"#DejaVuSans-77\"/>\n",
       "      <use x=\"86.279297\" xlink:href=\"#DejaVuSans-84\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"p73750b554c\">\n",
       "   <rect height=\"249.385\" width=\"140.381667\" x=\"57.68175\" y=\"7.2\"/>\n",
       "  </clipPath>\n",
       "  <clipPath id=\"p98c49c4735\">\n",
       "   <rect height=\"249.385\" width=\"140.381667\" x=\"234.433417\" y=\"7.2\"/>\n",
       "  </clipPath>\n",
       "  <clipPath id=\"p0b31e1114f\">\n",
       "   <rect height=\"249.385\" width=\"140.381667\" x=\"411.185083\" y=\"7.2\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<Figure size 576x360 with 3 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Let's put this on some graphs\n",
    "def plot_variant_numbers(subplot, df, genes, variant_count, chromosome):\n",
    "    subplot.bar(df[genes], height = df[variant_count], label = f\"{chromosome}\")\n",
    "    subplot.tick_params(axis = 'x', rotation = 90)\n",
    "    subplot.legend()\n",
    "    \n",
    "\n",
    "fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize = (8, 5))\n",
    "plot_variant_numbers(ax1, x_genes_top5, \"x_genes\", \"x_variants_count\", \"X\")\n",
    "plot_variant_numbers(ax2, y_genes_top5, \"y_genes\", \"y_variants_count\", \"Y\")\n",
    "plot_variant_numbers(ax3, mt_genes_top5, \"mt_genes\", \"mt_variants_count\", \"MT\")\n",
    "ax1.set_ylabel('# Variants Count')\n",
    "ax2.set_xlabel('Gene Names')\n",
    "\n",
    "fig.tight_layout()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "uo7xK2wj9cE7"
   },
   "source": [
    "From this graph we can start ask other questions. For each chromosome we are going to select one gene (DMD, SRY, ND6) and determine most common diseases associated to them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "xixTrB3K9cE9"
   },
   "outputs": [],
   "source": [
    "query_30 = \"\"\"\n",
    "MATCH (v:Variant {variantOf: \"DMD\"})-[:ASSOCIATED_WITH]->(d:Disease)\n",
    "RETURN COUNT(d.diseaseName) AS numberOfDmdVariants, d.diseaseName AS dmdLinkedDiseases\n",
    "ORDER BY numberOfDmdVariants DESC LIMIT 3\n",
    "\"\"\"\n",
    "\n",
    "query_31 = \"\"\"\n",
    "MATCH (v:Variant {variantOf: \"SRY\"})-[:ASSOCIATED_WITH]->(d:Disease)\n",
    "RETURN COUNT(d.diseaseName) AS numberOfSryVariants, d.diseaseName AS sryLinkedDiseases\n",
    "ORDER BY numberOfSryVariants DESC LIMIT 3\n",
    "\"\"\"\n",
    "\n",
    "query_32 = \"\"\"\n",
    "MATCH (v:Variant {variantOf: \"ND6\"})-[:ASSOCIATED_WITH]->(d:Disease)\n",
    "RETURN COUNT(d.diseaseName) AS numberOfNd6Variants, d.diseaseName AS nd6LinkedDiseases\n",
    "ORDER BY numberOfNd6Variants DESC LIMIT 3\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "iGI03jwr9cFA"
   },
   "outputs": [],
   "source": [
    "dmd_diseases = graph.run(query_30).to_data_frame()\n",
    "sry_diseases = graph.run(query_31).to_data_frame()\n",
    "nd6_diseases = graph.run(query_32).to_data_frame()\n",
    "\n",
    "temp_df = pd.concat([dmd_diseases, sry_diseases], axis = 1)\n",
    "gene_and_diseases = pd.concat([temp_df, nd6_diseases], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "Eo2NChvH9cFC"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>numberOfDmdVariants</th>\n",
       "      <th>dmdLinkedDiseases</th>\n",
       "      <th>numberOfSryVariants</th>\n",
       "      <th>sryLinkedDiseases</th>\n",
       "      <th>numberOfNd6Variants</th>\n",
       "      <th>nd6LinkedDiseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>164</td>\n",
       "      <td>Dmd-Associated Dilated Cardiomyopathy</td>\n",
       "      <td>27</td>\n",
       "      <td>46,Xy Gonadal Dysgenesis, Complete, Sry-Related</td>\n",
       "      <td>24</td>\n",
       "      <td>Malignant neoplasm of breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>151</td>\n",
       "      <td>Muscular Dystrophy, Duchenne</td>\n",
       "      <td>1</td>\n",
       "      <td>Ovotesticular Disorders of Sex Development</td>\n",
       "      <td>12</td>\n",
       "      <td>ovarian neoplasm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>29</td>\n",
       "      <td>Becker Muscular Dystrophy</td>\n",
       "      <td>1</td>\n",
       "      <td>46,Xy True Hermaphroditism, Sry-Related</td>\n",
       "      <td>11</td>\n",
       "      <td>Optic Atrophy, Hereditary, Leber</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   numberOfDmdVariants                      dmdLinkedDiseases  \\\n",
       "0                  164  Dmd-Associated Dilated Cardiomyopathy   \n",
       "1                  151           Muscular Dystrophy, Duchenne   \n",
       "2                   29              Becker Muscular Dystrophy   \n",
       "\n",
       "   numberOfSryVariants                                sryLinkedDiseases  \\\n",
       "0                   27  46,Xy Gonadal Dysgenesis, Complete, Sry-Related   \n",
       "1                    1       Ovotesticular Disorders of Sex Development   \n",
       "2                    1          46,Xy True Hermaphroditism, Sry-Related   \n",
       "\n",
       "   numberOfNd6Variants                 nd6LinkedDiseases  \n",
       "0                   24      Malignant neoplasm of breast  \n",
       "1                   12                  ovarian neoplasm  \n",
       "2                   11  Optic Atrophy, Hereditary, Leber  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_and_diseases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "wCwfrHVN9cFF"
   },
   "source": [
    "With very simple query we can quickly get the diseases with the most numbers of associated genetic variants. Let's focus on the gene `DMD`. The acronym is not really verbose but this gene codes for a protein called the **dystrophin**. Mutations in these genes are associated with X-linked dilated cardiomyopathy and Becker / Duchenne dystrophy. This is a famous disease sadly, as it affects young male individuals mostly. This is characterized by progressive muscle degeneration and weakness due to the alterations of the dystrophin. Life expectancy is strongly affected but improvement in critical care management enhanced survival into early 30. Several investigations are ongoing with notably many attempts to directly correct the mutation via gene therapy. France is one of the leader in the field and focus all hope and attention. The mitochondrial gene`ND6` which codes for the **NADH dehydrogenase 6**. A complicated name for a critical enzyme that is at the core of cellular energy production through a process called oxidative phosphorylation. We can see from our table that the two first diseases are cancers. While in the case of the dystrophin it is known that mutation is the cause of the diseases, here it is more complicated. Presence of these variants will not promote cancer development. It just says that these avriants are associated with the appearance of these cancers."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Diving more into the dataset\n",
    "\n",
    "In this part we are going to dive more into the dataset and will try to answer the following questions:\n",
    "- *Q1*: What are the top10 genes with the most genetic variants and their associated diseases or phenotypes\n",
    "- *Q2*: What are the most and less pleiotropic (associated with multiple diseases) genes?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Q1: What are the top10 genes with the most genetic variants and their associated diseases or phenotypes\n",
    "\n",
    "#### For diseases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_35 = \"\"\"\n",
    "MATCH (v:Variant)-[:ASSOCIATED_WITH]->(d:Disease)\n",
    "RETURN COUNT(v) AS number_of_variants, v.variantOf AS genes, d.diseaseName AS disease_name\n",
    "ORDER BY number_of_variants DESC LIMIT 10\n",
    "\"\"\"\n",
    "\n",
    "variants_in_diseases = graph.run(query_35).to_data_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number_of_variants</th>\n",
       "      <th>genes</th>\n",
       "      <th>disease_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2632</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILIT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2212</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILIT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1160</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>Hereditary Breast and Ovarian Cancer Syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1026</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>Hypercholesterolemia, Familial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>980</td>\n",
       "      <td>FBN1</td>\n",
       "      <td>Marfan Syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>860</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>Hereditary Breast and Ovarian Cancer Syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>791</td>\n",
       "      <td>NF1</td>\n",
       "      <td>Neurofibromatosis 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>542</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>Cystic Fibrosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>503</td>\n",
       "      <td>ULK4</td>\n",
       "      <td>Multiple Myeloma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>459</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>Hereditary Nonpolyposis Colorectal Cancer</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   number_of_variants  genes  \\\n",
       "0                2632  BRCA2   \n",
       "1                2212  BRCA1   \n",
       "2                1160  BRCA2   \n",
       "3                1026   LDLR   \n",
       "4                 980   FBN1   \n",
       "5                 860  BRCA1   \n",
       "6                 791    NF1   \n",
       "7                 542   CFTR   \n",
       "8                 503   ULK4   \n",
       "9                 459   MSH2   \n",
       "\n",
       "                                        disease_name  \n",
       "0  BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILIT...  \n",
       "1  BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILIT...  \n",
       "2      Hereditary Breast and Ovarian Cancer Syndrome  \n",
       "3                     Hypercholesterolemia, Familial  \n",
       "4                                    Marfan Syndrome  \n",
       "5      Hereditary Breast and Ovarian Cancer Syndrome  \n",
       "6                                Neurofibromatosis 1  \n",
       "7                                    Cystic Fibrosis  \n",
       "8                                   Multiple Myeloma  \n",
       "9          Hereditary Nonpolyposis Colorectal Cancer  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_in_diseases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For this query we can see that cancer-related diseases are the one with the highest number of associated variants. Notably we can see some key genes here, `BRCA1` and `BRCA2`, `BRCA` being an abbreviation for **BR**east **CA**ncer. These genes are involved in DNA repair. Mutations in these two genes are stronly associated with apperance of cancers (notably breast and ovarian tumors). Another sadly famous disease is Cystic fibrosis and its gene `CFTR`. We can also observed in this top10 that a lot of mutations have been identified and associated with a metabolic disease called `familial Hypercholesterolemia` that is assocaited with pathological levels of cholesterol in blood of affected individuals.\n",
    "\n",
    "Let's do now the same exercises but with our `Phenotype` node.\n",
    "\n",
    "#### For Phenotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "nF5GO-SU9cFI"
   },
   "outputs": [],
   "source": [
    "query_36 = \"\"\"\n",
    "MATCH (v:Variant)-[:IMPACTS_on]->(p:Phenotype)\n",
    "RETURN COUNT(v) AS number_of_variants, v.variantOf AS genes, p.phenotypeName AS observed_phentoype\n",
    "ORDER BY number_of_variants DESC LIMIT 10\n",
    "\"\"\"\n",
    "\n",
    "variants_in_phenotypes = graph.run(query_36).to_data_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "19BmKn1m9cFL"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number_of_variants</th>\n",
       "      <th>genes</th>\n",
       "      <th>observed_phentoype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>319</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>Hypercholesterolemia result</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>277</td>\n",
       "      <td>SLC2A9</td>\n",
       "      <td>Uric acid measurement (procedure)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>248</td>\n",
       "      <td>LDLR</td>\n",
       "      <td>Low density lipoprotein increased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>240</td>\n",
       "      <td>KANSL1</td>\n",
       "      <td>Corpuscular Hemoglobin Concentration Mean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>188</td>\n",
       "      <td>ALDH1A2</td>\n",
       "      <td>Serum albumin measurement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>163</td>\n",
       "      <td>DOCK7</td>\n",
       "      <td>Serum albumin measurement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>163</td>\n",
       "      <td>LINC02210-CRHR1</td>\n",
       "      <td>Corpuscular Hemoglobin Concentration Mean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>161</td>\n",
       "      <td>LOC105377462</td>\n",
       "      <td>Forced expiratory volume function</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>156</td>\n",
       "      <td>CCDC162P</td>\n",
       "      <td>Corpuscular Hemoglobin Concentration Mean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>138</td>\n",
       "      <td>LOC105377462</td>\n",
       "      <td>response to bronchodilator</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   number_of_variants            genes  \\\n",
       "0                 319             LDLR   \n",
       "1                 277           SLC2A9   \n",
       "2                 248             LDLR   \n",
       "3                 240           KANSL1   \n",
       "4                 188          ALDH1A2   \n",
       "5                 163            DOCK7   \n",
       "6                 163  LINC02210-CRHR1   \n",
       "7                 161     LOC105377462   \n",
       "8                 156         CCDC162P   \n",
       "9                 138     LOC105377462   \n",
       "\n",
       "                          observed_phentoype  \n",
       "0                Hypercholesterolemia result  \n",
       "1          Uric acid measurement (procedure)  \n",
       "2          Low density lipoprotein increased  \n",
       "3  Corpuscular Hemoglobin Concentration Mean  \n",
       "4                  Serum albumin measurement  \n",
       "5                  Serum albumin measurement  \n",
       "6  Corpuscular Hemoglobin Concentration Mean  \n",
       "7          Forced expiratory volume function  \n",
       "8  Corpuscular Hemoglobin Concentration Mean  \n",
       "9                 response to bronchodilator  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_in_phenotypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "WrKvHYnf9cFP"
   },
   "source": [
    "Herein we can see that the gene `LDLR` previously identified as polymorphic gene and associated with familial Hypercholesterolemia is here associated with two clinical observations / phenotypes: \n",
    "- `Hypercholesterolemia result` & `Low density lipoprotein increased`.\n",
    "\n",
    "Both observation relates to an increase of cholesterol in the blood of patients which clearly highlights here a link between the gene the phenotype and the disease.\n",
    "\n",
    "### Q2: What are the most and less pleiotropic genes?\n",
    "\n",
    "Let's imagine you are a scientist looking for its next move. You want your research to have a wide impact. You decide then to focus on genes that are described to be involved in several diseases. Our imaginary scientist could fastly find its next target among this top20:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number_of_disease</th>\n",
       "      <th>genes</th>\n",
       "      <th>disease_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>22</td>\n",
       "      <td>MECP2</td>\n",
       "      <td>Nervous System Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>21</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>Cardiovascular Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>21</td>\n",
       "      <td>FBN1</td>\n",
       "      <td>Cardiovascular Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>21</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>20</td>\n",
       "      <td>APC</td>\n",
       "      <td>Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>20</td>\n",
       "      <td>PTCH1</td>\n",
       "      <td>Not determined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>20</td>\n",
       "      <td>PTPN11</td>\n",
       "      <td>Cardiovascular Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>19</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>18</td>\n",
       "      <td>FBN1</td>\n",
       "      <td>Not determined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>17</td>\n",
       "      <td>FBN1</td>\n",
       "      <td>Musculoskeletal Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>17</td>\n",
       "      <td>GJB2</td>\n",
       "      <td>Otorhinolaryngologic Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>16</td>\n",
       "      <td>TTN-AS1</td>\n",
       "      <td>Cardiovascular Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>16</td>\n",
       "      <td>SCN1A-AS1</td>\n",
       "      <td>Nervous System Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>CDKN2B-AS1</td>\n",
       "      <td>Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>15</td>\n",
       "      <td>DSP</td>\n",
       "      <td>Cardiovascular Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>15</td>\n",
       "      <td>MTOR</td>\n",
       "      <td>Nervous System Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>15</td>\n",
       "      <td>COL7A1</td>\n",
       "      <td>Skin and Connective Tissue Diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>15</td>\n",
       "      <td>HBB</td>\n",
       "      <td>Hemic and Lymphatic Diseases</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    number_of_disease       genes                        disease_class\n",
       "0                  29        TP53                            Neoplasms\n",
       "1                  22       MECP2              Nervous System Diseases\n",
       "2                  21       SCN5A              Cardiovascular Diseases\n",
       "3                  21        FBN1              Cardiovascular Diseases\n",
       "4                  21        PTEN                            Neoplasms\n",
       "5                  20         APC                            Neoplasms\n",
       "6                  20       PTCH1                       Not determined\n",
       "7                  20      PTPN11              Cardiovascular Diseases\n",
       "8                  19      PIK3CA                            Neoplasms\n",
       "9                  18        FBN1                       Not determined\n",
       "10                 17        FBN1             Musculoskeletal Diseases\n",
       "11                 17        GJB2        Otorhinolaryngologic Diseases\n",
       "12                 16     TTN-AS1              Cardiovascular Diseases\n",
       "13                 16   SCN1A-AS1              Nervous System Diseases\n",
       "14                 15  CDKN2B-AS1                            Neoplasms\n",
       "15                 15       BRCA2                            Neoplasms\n",
       "16                 15         DSP              Cardiovascular Diseases\n",
       "17                 15        MTOR              Nervous System Diseases\n",
       "18                 15      COL7A1  Skin and Connective Tissue Diseases\n",
       "19                 15         HBB         Hemic and Lymphatic Diseases"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_37 = \"\"\"\n",
    "MATCH (v:Variant )-[:ASSOCIATED_WITH]->(d:Disease)\n",
    "RETURN COUNT(DISTINCT d) AS number_of_disease, v.variantOf AS genes, d.diseaseGroup AS disease_class\n",
    "ORDER BY number_of_disease DESC LIMIT 20\n",
    "\"\"\"\n",
    "\n",
    "top20_pleiotropic_genes = graph.run(query_37)\n",
    "top20_pleiotropic_genes.to_data_frame()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The query we sent to our database returns the top20 genes based the number of diseases they are associated with and group by disease class. As such our scientist can quickly infer the most pleiotropic genes and the type of disease it is involved in."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Conclusion\n",
    "\n",
    "Herein we demonstrated the utility of Neo4J as a graph database to build, investigate and study biological network data. Most importantly the Cypher language behind Neo4J and the software environment driven by the both the company and the Open Source World makes the use of Neo4J very accessible. You can interact with Neo4J databases through any languages including Python, Ruby, Java and R. As such it allows scientist to quickly include this tool in their current pipeline. Our analysis here could have been much more thorough from a biological point of view. We chose to tackle very simple questions to avoid to much redundancy in our queries with as a main objective: be able to give a concise overview of Neo4J. Indeed Neo4J could be also used for more advanced analysis as it permits to harness advanced machine learning algorithms thanks to the use of different plugins (APOC, Neo4J Data Science Plugin). Additionnaly we could have fed our data with other data repositories going from pathway analysis to protein expression but this would definitively implies much more work and could actually lead to a scientific publication if well done. Asa final note we would like to underline a very important point when dealing with graph databases which is data modeling. Indeed we are convinced that thies the most critical step. Being able to modelize the network of interactions between different data sources, choosing the right labels, properties and relationships are critical steps to pave the way toward a good and well rounded graph database in Neo4J."
   ]
  }
 ],
 "metadata": {
  "colab": {
   "name": "20200624_testing_py2neo.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  },
  "nteract": {
   "version": "0.24.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
